diff --git a/dist-cjs/is-crt-available.js b/dist-cjs/is-crt-available.js
index 4d794148f7cf0eb8f359b1a9199967b74aad8168..c451bbc3e0d866942c4a90b42458c219bc498807 100644
--- a/dist-cjs/is-crt-available.js
+++ b/dist-cjs/is-crt-available.js
@@ -2,14 +2,8 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.isCrtAvailable = void 0;
 const isCrtAvailable = () => {
-    try {
-        if (typeof require === "function" && typeof module !== "undefined" && module.require && require("aws-crt")) {
-            return ["md/crt-avail"];
-        }
-        return null;
-    }
-    catch (e) {
-        return null;
-    }
+    // NOTE(calebmer): `aws-crt` is a native module that doesn't play well with
+    // our Next.js server bundling. Always disable it.
+    return null;
 };
 exports.isCrtAvailable = isCrtAvailable;
diff --git a/dist-es/is-crt-available.js b/dist-es/is-crt-available.js
index b060369f453fa6cbe93e9894d87b4e4848a5d8a2..c53a8b36d47830e168016943f20ea6da3cad7655 100644
--- a/dist-es/is-crt-available.js
+++ b/dist-es/is-crt-available.js
@@ -1,11 +1,5 @@
 export const isCrtAvailable = () => {
-    try {
-        if (typeof require === "function" && typeof module !== "undefined" && module.require && require("aws-crt")) {
-            return ["md/crt-avail"];
-        }
-        return null;
-    }
-    catch (e) {
-        return null;
-    }
+    // NOTE(calebmer): `aws-crt` is a native module that doesn't play well with
+    // our Next.js server bundling. Always disable it.
+    return null;
 };