diff --git a/lib/providers/ec2.d.ts b/lib/providers/ec2.d.ts
index 5ed1001f3d3abd6312c3d172cab2163486333c6c..635fa215633aa9b39cca0494e450852fdb7abc50 100644
--- a/lib/providers/ec2.d.ts
+++ b/lib/providers/ec2.d.ts
@@ -87,6 +87,12 @@ export interface Ec2RunnerProviderProps extends RunnerProviderProps {
      * @default no max price (you will pay current spot price)
      */
     readonly spotMaxPrice?: string;
+    // NOTE(calebmer): Extra data to pass into the EC2 GitHub runner instance
+    // when it launches. Can be any string, by convention we typically pass in
+    // a JSON string.
+    readonly userDataExtra?: string;
+    // NOTE(rmtobin): Extra tags to add to EC2 instances launched by this runner.
+    readonly extraTags?: Array<{key: string; value: string}>;
 }
 /**
  * GitHub Actions runner provider using EC2 to execute jobs.
diff --git a/lib/providers/ec2.js b/lib/providers/ec2.js
index d1a6f14ad2a72fc3abd53cec9c197ed9d28eb019..fde524ec479f28df03a6cc828833bc40dcf95bb5 100644
--- a/lib/providers/ec2.js
+++ b/lib/providers/ec2.js
@@ -22,6 +22,7 @@ repoPath="{}"
 runnerTokenPath="{}"
 labels="{}"
 registrationURL="{}"
+export USER_DATA_EXTRA="$(echo "{}" | base64 --decode)"
 
 heartbeat () {
   while true; do
@@ -65,7 +66,7 @@ action () {
   sudo -Hu runner /home/runner/config.sh --unattended --url "$registrationURL" --token "$runnerTokenPath" --ephemeral --work _work --labels "$labelsTemplate" $RUNNER_FLAGS --name "$runnerNamePath" || exit 1
 
   # Execute the run command
-  sudo --preserve-env=AWS_REGION -Hu runner /home/runner/run.sh || exit 2
+  sudo --preserve-env=AWS_REGION,USER_DATA_EXTRA -Hu runner /home/runner/run.sh || exit 2
 
   # Retrieve the status
   STATUS=$(grep -Phors "finish job request for job [0-9a-f\\-]+ with result: \K.*" /home/runner/_diag/ | tail -n1)
@@ -94,6 +95,7 @@ $repoPath="{}"
 $runnerTokenPath="{}"
 $labels="{}"
 $registrationURL="{}"
+$USER_DATA_EXTRA="{}"
 
 # EC2Launch only starts ssm agent after user data is done, so we need to start it ourselves (it is disabled by default)
 Set-Service -StartupType Manual AmazonSSMAgent
@@ -236,6 +238,8 @@ class Ec2RunnerProvider extends common_1.BaseProvider {
             removalPolicy: aws_cdk_lib_1.RemovalPolicy.DESTROY,
         });
         this.logGroup.grantWrite(this);
+        this.userDataExtra = props.userDataExtra || "";
+        this.extraTags = props.extraTags || [];
     }
     /**
      * Generate step function task(s) to start a new runner.
@@ -256,6 +260,7 @@ class Ec2RunnerProvider extends common_1.BaseProvider {
             parameters.runnerTokenPath,
             this.labels.join(','),
             parameters.registrationUrl,
+            aws_cdk_lib_1.aws_stepfunctions.JsonPath.base64Encode(this.userDataExtra),
         ];
         const passUserData = new aws_cdk_lib_1.aws_stepfunctions.Pass(this, `${this.labels.join(', ')} data`, {
             parameters: {
@@ -316,10 +321,13 @@ class Ec2RunnerProvider extends common_1.BaseProvider {
                     TagSpecifications: [
                         {
                             ResourceType: 'instance',
-                            Tags: [{
+                            Tags: [
+                                {
                                     Key: 'GitHubRunners:Provider',
                                     Value: this.node.path,
-                                }],
+                                },
+                                ...this.extraTags.map(t => ({Key: t.key, Value: t.value})),
+                            ],
                         },
                         {
                             ResourceType: 'volume',
