diff --git a/dist/src/index.js b/dist/src/index.js
index fb79bb02047bb0b978286ab645cc5298da8e6341..47d061c1ff435f98d1ee1b72a2625e1d1c61cef1 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -151,7 +151,22 @@ ${suggestion}`);
     this.#referencedPathSizes.set(identifier, data.byteLength);
     switch (rule.type) {
       case "ESModule":
-        module2 = new import_vm2.default.SourceTextModule(data.toString("utf8"), moduleOptions);
+        module2 = new import_vm2.default.SourceTextModule(data.toString("utf8"), {
+          ...moduleOptions,
+          // NOTE(calebmer): Use the linker to support dynamic `import()`
+          // expressions.
+          importModuleDynamically: async (spec, referencing) => {
+            const module = await this.linker(spec, referencing);
+            // Defend against concurrent execution by only calling `link()` and
+            // `evaluate()` once and stashing the resulting promise so concurrent
+            // executions can await it.
+            if (module.status === "unlinked") module._linkPromise = module.link(this.linker);
+            await module._linkPromise;
+            if (module.status === "linked") module._evaluatePromise = module.evaluate();
+            await module._evaluatePromise;
+            return module;
+          },
+        });
         break;
       case "CommonJS":
         const exports = this.loadCommonJSModule(errorBase, identifier, spec, referencing.context);
@@ -256,7 +271,20 @@ var VMScriptRunner = class {
   async runAsModule(context, blueprint, linker) {
     const module2 = new import_vm3.default.SourceTextModule(blueprint.code, {
       identifier: blueprint.filePath,
-      context
+      context,
+      // NOTE(calebmer): Use the linker to support dynamic `import()`
+      // expressions.
+      importModuleDynamically: async (spec, referencing) => {
+        const module = await linker(spec, referencing);
+        // Defend against concurrent execution by only calling `link()` and
+        // `evaluate()` once and stashing the resulting promise so concurrent
+        // executions can await it.
+        if (module.status === "unlinked") module._linkPromise = module.link(linker);
+        await module._linkPromise;
+        if (module.status === "linked") module._evaluatePromise = module.evaluate();
+        await module._evaluatePromise;
+        return module;
+      },
     });
     await module2.link(linker);
     await module2.evaluate();