diff --git a/dist/src/index.js b/dist/src/index.js
index 1cfffad73fdb40e4f903388fad35b99a635701bc..f0ba673c84fd70d4a47276d8c24f71ed32a45056 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -102,7 +102,7 @@ var FileStorage = class extends import_storage_memory.LocalStorage {
   root;
   sqliteDB;
   keyPath(key) {
-    const sanitisedKey = this.sanitise ? (0, import_shared2.sanitisePath)(key) : key;
+    const sanitisedKey = encodePath(key);
     const filePath = import_path2.default.join(this.root, sanitisedKey);
     return [
       filePath.startsWith(this.root) ? filePath : void 0,
@@ -220,7 +220,7 @@ This is a limitation of Miniflare's file-system storage. Please use in-memory/Re
         continue;
       const name = filePath.substring(this.root.length + 1);
       const meta = await this.meta(filePath);
-      const realName = meta?.key ?? (this.sanitise ? name : name.split(import_path2.default.sep).join(import_path2.default.posix.sep));
+      const realName = meta?.key ?? name;
       keys.push({
         name: realName,
         expiration: meta.expiration,
@@ -230,6 +230,50 @@ This is a limitation of Miniflare's file-system storage. Please use in-memory/Re
     return keys;
   }
 };
+
+// NOTE(calebmer): MacOS file systems are case-insensitive by default. Whereas
+// Durable Object storage (which this class is used for) is case-sensitive. So
+// use percent encoding for all non-path safe characters. Including uppercase
+// characters.
+//
+// A bug was filed here: https://github.com/cloudflare/miniflare/issues/247
+//
+// But the fix was never implemented since Miniflare v2 was archived and now
+// it's recommend you run `workerd` directly locally. We've chosen not to
+// upgrade to `workerd` and instead maintain Miniflare v2 ourselves since the
+// Node.js runtime is really handy for our dev + test environments.
+function encodePath(path) {
+  const encodedPath = path.replace(/./g, char => {
+    const code = char.charCodeAt(0);
+      
+    // Keep lowercase letters and numbers unencoded
+    if (
+      (code >= 0x61 && code <= 0x7A) || // a-z
+      (code >= 0x30 && code <= 0x39) || // 0-9
+      code === 0x2D ||                  // `-`
+      code === 0x5F ||                  // `_`
+      code === 0x7E                     // `~`
+    ) {
+        return char;
+    }
+      
+    // Encode everything else (uppercase, special chars, etc.)
+    return `%${code.toString(16).toUpperCase().padStart(2, "0")}`;
+  });
+
+  // Paths on MacOS can't be more than 255 bytes. So split the string into 255
+  // character chunks.
+  return splitStringIntoChunks(encodedPath, 255).join("/");
+}
+
+function splitStringIntoChunks(string, chunkLength) {
+  const chunks = [];
+  for (let i = 0; i < string.length; i += chunkLength) {
+      chunks.push(string.slice(i, i + chunkLength));
+  }
+  return chunks;
+}
+
 // Annotate the CommonJS export names for ESM import in node:
 0 && (module.exports = {
   FileStorage,
