diff --git a/dist/src/index.js b/dist/src/index.js
index 08378e7404d3de090a884c878b0ed1dd93795b81..b79b0ee4b0378d8c66abf55295dbc7141e81fd58 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -39,7 +39,10 @@ __export(exports, {
   createRequestListener: () => createRequestListener,
   createServer: () => createServer,
   getAccessibleHosts: () => getAccessibleHosts,
-  startServer: () => startServer
+  startServer: () => startServer,
+  // NOTE(calebmer): We manually connect the HTTP and WebSocket servers so we
+  // need this function.
+  writeResponse: () => writeResponse
 });
 var import_assert = __toModule(require("assert"));
 var import_http = __toModule(require("http"));
@@ -681,6 +684,9 @@ async function startServer(mf, options) {
   createRequestListener,
   createServer,
   getAccessibleHosts,
-  startServer
+  startServer,
+  // NOTE(calebmer): We manually connect the HTTP and WebSocket servers so we
+  // need this function.
+  writeResponse
 });
 //# sourceMappingURL=index.js.map