diff --git a/dist/src/index.js b/dist/src/index.js
index 9880107a961a732e7162aaf73e01573e28cb1ff9..19f061bd40582390d18d908886d3ee56dce628e8 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -61,6 +61,7 @@ var QueuesPlugin = class extends import_shared.Plugin {
     }
   }
   async setup(_storageFactory) {
+    const consumerQueueNames = new Set();
     for (const entry of this.queueConsumers ?? []) {
       let opts;
       if (typeof entry === "string") {
@@ -78,6 +79,16 @@ var QueuesPlugin = class extends import_shared.Plugin {
         deadLetterQueue: opts.deadLetterQueue,
         dispatcher: this.ctx.queueEventDispatcher
       };
+
+      // NOTE(calebmer): Moving multiple consumer error here to preserve the
+      // "only one consumer per queue in a plugin's configuration" behavior
+      // while fixing plugin reloads which needs to swap out one consumer for
+      // another.
+      if (consumerQueueNames.has(consumer.queueName)) {
+        throw new QueueError("ERR_CONSUMER_ALREADY_SET");
+      }
+      consumerQueueNames.add(consumer.queueName);
+
       const queue = this.ctx.queueBroker.getOrCreateQueue(opts.queueName);
       this.ctx.queueBroker.setConsumer(queue, consumer);
     }
@@ -212,9 +223,17 @@ var WorkerQueue = class {
     }
   }
   [import_shared2.kSetConsumer](consumer) {
-    if (this.#consumer) {
-      throw new QueueError("ERR_CONSUMER_ALREADY_SET");
-    }
+    // NOTE(calebmer): Clear existing pending flush when setting a new consumer.
+    // We will schedule a new flush if there are messages. Unclear to me whether
+    // this is actually the correct behavior. If there was a 10s batch timer
+    // then the timer is reset on reload! (But this is the correct behavior,
+    // say, if the batch timeout changed.) This works for us and feels like the
+    // most defensive reload behavior. Since we're patching, better to be
+    // defensive.
+    this.#pendingFlush = 0;
+    clearTimeout(this.#timeout);
+    this.#timeout = void 0;
+
     this.#consumer = consumer;
     if (this.#messages.length) {
       this.#ensurePendingFlush();