diff --git a/dist/import.mjs b/dist/import.mjs
index 4c3bacad099f18c8a2c3d7f108fa083bd83b0d9a..19a91d218596bc797ee21b4eb3cf0f9efa1e628d 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -808,8 +808,13 @@ function $a1ea59d68270f0dd$export$f8168d8dd8fd66e6(props) {
 // See https://github.com/facebook/react/tree/cc7c1aece46a6b69b41958d731e0fd27c94bfc6c/packages/react-interactions
 
 
-
-let $507fabe10e71c6fb$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $507fabe10e71c6fb$var$currentModality = "pointer";
 let $507fabe10e71c6fb$var$changeHandlers = new Set();
 let $507fabe10e71c6fb$var$hasSetupGlobalListeners = false;
 let $507fabe10e71c6fb$var$hasEventBeforeFocus = false;
@@ -1282,7 +1287,10 @@ function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
             },
             continuePropagation () {
                 shouldStopPropagation = false;
diff --git a/dist/main.js b/dist/main.js
index b4115b501a23f92a4cbe3fcbfe0b3495053590dd..e8ba90fbb4d45e6f8fd360db7ea391c8f744c1f6 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -833,7 +833,13 @@ function $5cb73d0ce355b0dc$export$f8168d8dd8fd66e6(props) {
 
 
 
-let $e77252a287ef94ab$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $e77252a287ef94ab$var$currentModality = "pointer";
 let $e77252a287ef94ab$var$changeHandlers = new Set();
 let $e77252a287ef94ab$var$hasSetupGlobalListeners = false;
 let $e77252a287ef94ab$var$hasEventBeforeFocus = false;
@@ -1306,7 +1312,10 @@ function $edcfa848c42f94f4$var$isValidEvent(event, ref) {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
             },
             continuePropagation () {
                 shouldStopPropagation = false;
diff --git a/dist/module.js b/dist/module.js
index 4c3bacad099f18c8a2c3d7f108fa083bd83b0d9a..20b0f6a9211ad3bb4df32c89b34645dd6be49065 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -809,7 +809,13 @@ function $a1ea59d68270f0dd$export$f8168d8dd8fd66e6(props) {
 
 
 
-let $507fabe10e71c6fb$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $507fabe10e71c6fb$var$currentModality = "pointer";
 let $507fabe10e71c6fb$var$changeHandlers = new Set();
 let $507fabe10e71c6fb$var$hasSetupGlobalListeners = false;
 let $507fabe10e71c6fb$var$hasEventBeforeFocus = false;
@@ -1282,7 +1288,10 @@ function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
             },
             continuePropagation () {
                 shouldStopPropagation = false;
diff --git a/src/createEventHandler.ts b/src/createEventHandler.ts
index 5733c7e4221092879a6409ca983e919edff0da87..d748e49c5cda985cc5ae9f07edd7d2c7c7d6c874 100644
--- a/src/createEventHandler.ts
+++ b/src/createEventHandler.ts
@@ -32,7 +32,10 @@ export function createEventHandler<T extends SyntheticEvent>(handler: (e: BaseEv
         return e.isDefaultPrevented();
       },
       stopPropagation() {
-        console.error('stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.');
+        // NOTE(calebmer): Our style guide recommends calling both
+        // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+        // Don't warn when we do this. Treat repeat calls a noop like regular
+        // DOM events.
       },
       continuePropagation() {
         shouldStopPropagation = false;
diff --git a/src/useFocusVisible.ts b/src/useFocusVisible.ts
index 60fd6b5a77f5d9f26ffdb6e750e11a0cab6923e4..db1066703ef1c941ab6ee4c422b98931bf93c1f9 100644
--- a/src/useFocusVisible.ts
+++ b/src/useFocusVisible.ts
@@ -35,7 +35,13 @@ export interface FocusVisibleResult {
   isFocusVisible: boolean
 }
 
-let currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let currentModality = 'pointer';
 let changeHandlers = new Set<Handler>();
 let hasSetupGlobalListeners = false;
 let hasEventBeforeFocus = false;