diff --git a/dist/import.mjs b/dist/import.mjs
index a6e7e525bb20d202316f8feda95969bc1df5599a..b4dec30ea4d474666dcb95dd657d77e3d75bd208 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -130,7 +130,13 @@ function $b5e257d569688ac6$var$useModernSSRSafeId(defaultId) {
     let prefix = didSSR || process.env.NODE_ENV === "test" ? "react-aria" : `react-aria${$b5e257d569688ac6$var$defaultContext.prefix}`;
     return defaultId || `${prefix}-${id}`;
 }
-const $b5e257d569688ac6$export$619500959fc48b26 = typeof (0, $73SJx$react)["useId"] === "function" ? $b5e257d569688ac6$var$useModernSSRSafeId : $b5e257d569688ac6$var$useLegacySSRSafeId;
+function $b5e257d569688ac6$export$619500959fc48b26(defaultId) {
+    // NOTE(calebmer): Use the React hook for generating unique IDs which is SSR safe
+    // without needing context. (`react-aria` has added support for `React.useId()`
+    // but we still use this patch so we can remove the `react-aria` prefix.)
+    const id = (0, $73SJx$react).useId();
+    return defaultId || id;
+}
 function $b5e257d569688ac6$var$getSnapshot() {
     return false;
 }
@@ -142,15 +148,26 @@ function $b5e257d569688ac6$var$subscribe(onStoreChange) {
     // noop
     return ()=>{};
 }
+// NOTE(calebmer): We want `useIsSSR()` to have the same implementation as
+// `useIsInitialAppRender()`. So declare our context in `@react-aria/ssr`
+// and copy the implementation of `useIsInitialAppRender()` into `useIsSSR()`.
+// We setup the context provider in our code.
+const AppInitialRenderContext = /*#__PURE__*/ (0, $73SJx$react).createContext(null);
 function $b5e257d569688ac6$export$535bd6ca7f90a273() {
-    // In React 18, we can use useSyncExternalStore to detect if we're server rendering or hydrating.
-    if (typeof (0, $73SJx$react)["useSyncExternalStore"] === "function") return (0, $73SJx$react)["useSyncExternalStore"]($b5e257d569688ac6$var$subscribe, $b5e257d569688ac6$var$getSnapshot, $b5e257d569688ac6$var$getServerSnapshot);
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    return (0, $73SJx$useContext)($b5e257d569688ac6$var$IsSSRContext);
-}
+    const initialAppRenderId = (0, $73SJx$useContext)(AppInitialRenderContext);
+
+    if (initialAppRenderId === null) {
+        // In Jest tests, act like we are not in the initial render unless an
+        // `useAppInitialRenderContextProvider()` is explicitly used.
+        if (import.meta.jest) return false;
 
+        throw new Error("Must be rendered in an `useAppInitialRenderContextProvider()`");
+    }
+
+    return initialAppRenderId !== false;
+}
 
 
 
-export {$b5e257d569688ac6$export$9f8ac96af4b1b2ae as SSRProvider, $b5e257d569688ac6$export$619500959fc48b26 as useSSRSafeId, $b5e257d569688ac6$export$535bd6ca7f90a273 as useIsSSR};
+export {$b5e257d569688ac6$export$9f8ac96af4b1b2ae as SSRProvider, $b5e257d569688ac6$export$619500959fc48b26 as useSSRSafeId, $b5e257d569688ac6$export$535bd6ca7f90a273 as useIsSSR, AppInitialRenderContext};
 //# sourceMappingURL=module.js.map
diff --git a/dist/main.js b/dist/main.js
index 655e666962e75ba4c1840e679e23bcbfa9503b31..b6ad8ff01f031ad0f44a0b3cf8c29dbd91a8c1e7 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -10,6 +10,7 @@ function $parcel$interopDefault(a) {
 $parcel$export(module.exports, "SSRProvider", () => $97d95f6660b1bb14$export$9f8ac96af4b1b2ae);
 $parcel$export(module.exports, "useSSRSafeId", () => $97d95f6660b1bb14$export$619500959fc48b26);
 $parcel$export(module.exports, "useIsSSR", () => $97d95f6660b1bb14$export$535bd6ca7f90a273);
+$parcel$export(module.exports, "AppInitialRenderContext", () => AppInitialRenderContext);
 /*
  * Copyright 2020 Adobe. All rights reserved.
  * This file is licensed to you under the Apache License, Version 2.0 (the "License");
@@ -140,7 +141,13 @@ function $97d95f6660b1bb14$var$useModernSSRSafeId(defaultId) {
     let prefix = didSSR || process.env.NODE_ENV === "test" ? "react-aria" : `react-aria${$97d95f6660b1bb14$var$defaultContext.prefix}`;
     return defaultId || `${prefix}-${id}`;
 }
-const $97d95f6660b1bb14$export$619500959fc48b26 = typeof (0, ($parcel$interopDefault($32tCg$react)))["useId"] === "function" ? $97d95f6660b1bb14$var$useModernSSRSafeId : $97d95f6660b1bb14$var$useLegacySSRSafeId;
+function $97d95f6660b1bb14$export$619500959fc48b26(defaultId) {
+    // NOTE(calebmer): Use the React hook for generating unique IDs which is SSR safe
+    // without needing context. (`react-aria` has added support for `React.useId()`
+    // but we still use this patch so we can remove the `react-aria` prefix.)
+    const id = (0, ($parcel$interopDefault($32tCg$react))).useId();
+    return defaultId || id;
+}
 function $97d95f6660b1bb14$var$getSnapshot() {
     return false;
 }
@@ -152,11 +159,19 @@ function $97d95f6660b1bb14$var$subscribe(onStoreChange) {
     // noop
     return ()=>{};
 }
+// NOTE(calebmer): We want `useIsSSR()` to have the same implementation as
+// `useIsInitialAppRender()`. So declare our context in `@react-aria/ssr`
+// and copy the implementation of `useIsInitialAppRender()` into `useIsSSR()`.
+// We setup the context provider in our code.
+const AppInitialRenderContext = /*#__PURE__*/ (0, ($parcel$interopDefault($32tCg$react))).createContext(null);
 function $97d95f6660b1bb14$export$535bd6ca7f90a273() {
-    // In React 18, we can use useSyncExternalStore to detect if we're server rendering or hydrating.
-    if (typeof (0, ($parcel$interopDefault($32tCg$react)))["useSyncExternalStore"] === "function") return (0, ($parcel$interopDefault($32tCg$react)))["useSyncExternalStore"]($97d95f6660b1bb14$var$subscribe, $97d95f6660b1bb14$var$getSnapshot, $97d95f6660b1bb14$var$getServerSnapshot);
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    return (0, $32tCg$react.useContext)($97d95f6660b1bb14$var$IsSSRContext);
+    const initialAppRenderId = (0, $32tCg$react.useContext)(AppInitialRenderContext);
+
+    if (initialAppRenderId === null) {
+        throw new Error("Must be rendered in an `useAppInitialRenderContextProvider()`");
+    }
+
+    return initialAppRenderId !== false;
 }
 
 
diff --git a/dist/module.js b/dist/module.js
index a6e7e525bb20d202316f8feda95969bc1df5599a..f7c4fe92e883d6e51733f75af4bcbf829787b840 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -130,7 +130,13 @@ function $b5e257d569688ac6$var$useModernSSRSafeId(defaultId) {
     let prefix = didSSR || process.env.NODE_ENV === "test" ? "react-aria" : `react-aria${$b5e257d569688ac6$var$defaultContext.prefix}`;
     return defaultId || `${prefix}-${id}`;
 }
-const $b5e257d569688ac6$export$619500959fc48b26 = typeof (0, $73SJx$react)["useId"] === "function" ? $b5e257d569688ac6$var$useModernSSRSafeId : $b5e257d569688ac6$var$useLegacySSRSafeId;
+function $b5e257d569688ac6$export$619500959fc48b26(defaultId) {
+    // NOTE(calebmer): Use the React hook for generating unique IDs which is SSR safe
+    // without needing context. (`react-aria` has added support for `React.useId()`
+    // but we still use this patch so we can remove the `react-aria` prefix.)
+    const id = (0, $73SJx$react).useId();
+    return defaultId || id;
+}
 function $b5e257d569688ac6$var$getSnapshot() {
     return false;
 }
@@ -142,15 +148,27 @@ function $b5e257d569688ac6$var$subscribe(onStoreChange) {
     // noop
     return ()=>{};
 }
+// NOTE(calebmer): We want `useIsSSR()` to have the same implementation as
+// `useIsInitialAppRender()`. So declare our context in `@react-aria/ssr`
+// and copy the implementation of `useIsInitialAppRender()` into `useIsSSR()`.
+// We setup the context provider in our code.
+const AppInitialRenderContext = /*#__PURE__*/ (0, $73SJx$react).createContext(null);
 function $b5e257d569688ac6$export$535bd6ca7f90a273() {
-    // In React 18, we can use useSyncExternalStore to detect if we're server rendering or hydrating.
-    if (typeof (0, $73SJx$react)["useSyncExternalStore"] === "function") return (0, $73SJx$react)["useSyncExternalStore"]($b5e257d569688ac6$var$subscribe, $b5e257d569688ac6$var$getSnapshot, $b5e257d569688ac6$var$getServerSnapshot);
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    return (0, $73SJx$useContext)($b5e257d569688ac6$var$IsSSRContext);
+    const initialAppRenderId = (0, $73SJx$useContext)(AppInitialRenderContext);
+
+    if (initialAppRenderId === null) {
+        // In Jest tests, act like we are not in the initial render unless an
+        // `useAppInitialRenderContextProvider()` is explicitly used.
+        if (import.meta.jest) return false;
+
+        throw new Error("Must be rendered in an `useAppInitialRenderContextProvider()`");
+    }
+
+    return initialAppRenderId !== false;
 }
 
 
 
 
-export {$b5e257d569688ac6$export$9f8ac96af4b1b2ae as SSRProvider, $b5e257d569688ac6$export$619500959fc48b26 as useSSRSafeId, $b5e257d569688ac6$export$535bd6ca7f90a273 as useIsSSR};
+export {$b5e257d569688ac6$export$9f8ac96af4b1b2ae as SSRProvider, $b5e257d569688ac6$export$619500959fc48b26 as useSSRSafeId, $b5e257d569688ac6$export$535bd6ca7f90a273 as useIsSSR, AppInitialRenderContext};
 //# sourceMappingURL=module.js.map
diff --git a/dist/types.d.ts b/dist/types.d.ts
index adbad36402df8376a017ff3d70216d8768ed7096..24a22d000327239355d94e7d9d43f2cd0b71f41f 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -1,4 +1,4 @@
-import { ReactNode } from "react";
+import { ReactNode, Context } from "react";
 export interface SSRProviderProps {
     /** Your application here. */
     children: ReactNode;
@@ -18,4 +18,10 @@ export const useSSRSafeId: typeof useModernSSRSafeId;
  */
 export function useIsSSR(): boolean;
 
+// NOTE(calebmer): We want `useIsSSR()` to have the same implementation as
+// `useIsInitialAppRender()`. So declare our context in `@react-aria/ssr`
+// and copy the implementation of `useIsInitialAppRender()` into `useIsSSR()`.
+// We setup the context provider in our code.
+export const AppInitialRenderContext: Context<{time: Date; id: string} | false | null>;
+
 //# sourceMappingURL=types.d.ts.map
diff --git a/src/SSRProvider.tsx b/src/SSRProvider.tsx
index 7be1abf3c6fc11e10d69fb52318029a771d75b86..52cffed9bdee7902755151f99928aac15427348a 100644
--- a/src/SSRProvider.tsx
+++ b/src/SSRProvider.tsx
@@ -166,7 +166,13 @@ function useModernSSRSafeId(defaultId?: string): string {
 
 // Use React.useId in React 18 if available, otherwise fall back to our old implementation.
 /** @private */
-export const useSSRSafeId = typeof React['useId'] === 'function' ? useModernSSRSafeId : useLegacySSRSafeId;
+export function useSSRSafeId(defaultId?: string) {
+  // NOTE(calebmer): Use the React hook for generating unique IDs which is SSR safe
+  // without needing context. (`react-aria` has added support for `React.useId()`
+  // but we still use this patch so we can remove the `react-aria` prefix.)
+  const id = React.useId();
+  return defaultId || id;
+}
 
 function getSnapshot() {
   return false;
@@ -182,17 +188,27 @@ function subscribe(onStoreChange: () => void): () => void {
   return () => {};
 }
 
+// NOTE(calebmer): We want `useIsSSR()` to have the same implementation as
+// `useIsInitialAppRender()`. So declare our context in `@react-aria/ssr`
+// and copy the implementation of `useIsInitialAppRender()` into `useIsSSR()`.
+// We setup the context provider in our code.
+export const AppInitialRenderContext = React.createContext<string | false | null>(null);
+
 /**
  * Returns whether the component is currently being server side rendered or
  * hydrated on the client. Can be used to delay browser-specific rendering
  * until after hydration.
  */
 export function useIsSSR(): boolean {
-  // In React 18, we can use useSyncExternalStore to detect if we're server rendering or hydrating.
-  if (typeof React['useSyncExternalStore'] === 'function') {
-    return React['useSyncExternalStore'](subscribe, getSnapshot, getServerSnapshot);
+  const initialAppRenderId = useContext(AppInitialRenderContext);
+
+  if (initialAppRenderId === null) {
+      // In Jest tests, act like we are not in the initial render unless an
+      // `useAppInitialRenderContextProvider()` is explicitly used.
+      if (import.meta.jest) return false;
+
+      throw new Error("Must be rendered in an `useAppInitialRenderContextProvider()`");
   }
 
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  return useContext(IsSSRContext);
+  return initialAppRenderId !== false;
 }
diff --git a/src/index.ts b/src/index.ts
index ee5c98be7279063a09ef8b4e27ff970faa2fb116..5d1d7893e2a13d85782fc26997a3df4da54f0047 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -9,5 +9,5 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-export {SSRProvider, useSSRSafeId, useIsSSR} from './SSRProvider';
+export {SSRProvider, useSSRSafeId, useIsSSR, AppInitialRenderContext} from './SSRProvider';
 export type {SSRProviderProps} from './SSRProvider';
