diff --git a/dist/import.mjs b/dist/import.mjs
index b7f59c1ff16472ec3db0487957c6f6aad2794ff1..6e4b3f64205282e7a8949fc623738045d1d3f678 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -2,7 +2,7 @@ import {useRef as $eCAIO$useRef, useEffect as $eCAIO$useEffect, useMemo as $eCAI
 import {getFocusableTreeWalker as $eCAIO$getFocusableTreeWalker, focusSafely as $eCAIO$focusSafely} from "@react-aria/focus";
 import {focusWithoutScrolling as $eCAIO$focusWithoutScrolling, useEvent as $eCAIO$useEvent, scrollIntoView as $eCAIO$scrollIntoView, scrollIntoViewport as $eCAIO$scrollIntoViewport, mergeProps as $eCAIO$mergeProps, isAppleDevice as $eCAIO$isAppleDevice, isMac as $eCAIO$isMac} from "@react-aria/utils";
 import {useLocale as $eCAIO$useLocale, useCollator as $eCAIO$useCollator} from "@react-aria/i18n";
-import {usePress as $eCAIO$usePress, useLongPress as $eCAIO$useLongPress} from "@react-aria/interactions";
+import {usePress as $eCAIO$usePress, useLongPress as $eCAIO$useLongPress, getInteractionModality} from "@react-aria/interactions";
 
 /*
  * Copyright 2020 Adobe. All rights reserved.
@@ -303,14 +303,25 @@ function $ae20dd8cbca75726$export$d6daf82dcd84e87c(options) {
     }, []);
     // If not virtualized, scroll the focused element into view when the focusedKey changes.
     // When virtualized, Virtualizer handles this internally.
+    //
+    // NOTE(calebmer): Use our preferred scroll behavior for selectable collections.
+    // We implement this behavior elsewhere in the product like
+    // `<ContentEditorMentionFloater>`. Consolidate?
+    //
+    // 1. Don't scroll on hover. Only scroll on mouse movement.
+    // 2. If scrolling to the first or last element then completely scroll the view.
     (0, $eCAIO$useEffect)(()=>{
+        let modality = getInteractionModality();
         if (!isVirtualized && manager.focusedKey && (scrollRef === null || scrollRef === void 0 ? void 0 : scrollRef.current)) {
             let element = scrollRef.current.querySelector(`[data-key="${manager.focusedKey}"]`);
-            if (element) {
-                (0, $eCAIO$scrollIntoView)(scrollRef.current, element);
-                (0, $eCAIO$scrollIntoViewport)(element, {
-                    containingElement: ref.current
-                });
+            if (element && modality === 'keyboard') {
+                if (!element.previousSibling) {
+                    scrollRef.current.scrollTop = 0;
+                } else if (!element.nextSibling) {
+                    scrollRef.current.scrollTop = scrollRef.current.scrollHeight - scrollRef.current.clientHeight;
+                } else {
+                    (0, $eCAIO$scrollIntoView)(scrollRef.current, element);
+                }
             }
         }
     }, [
diff --git a/dist/main.js b/dist/main.js
index f870c4da0d6464eaa909739ec1928533f0ad1f7a..9c7cb7865de0f9356c18f7566a0f265e8fa7f323 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -312,14 +312,24 @@ function $b6837c2f80a3c32f$export$d6daf82dcd84e87c(options) {
     }, []);
     // If not virtualized, scroll the focused element into view when the focusedKey changes.
     // When virtualized, Virtualizer handles this internally.
+    //
+    // NOTE(calebmer): Use our preferred scroll behavior for selectable collections.
+    // We implement this behavior elsewhere in the product like
+    // `<ContentEditorMentionFloater>`. Consolidate?
+    //
+    // 1. Don't scroll on hover. Only scroll on mouse movement.
+    // 2. If scrolling to the first or last element then completely scroll the view.
     (0, $glPPV$react.useEffect)(()=>{
         if (!isVirtualized && manager.focusedKey && (scrollRef === null || scrollRef === void 0 ? void 0 : scrollRef.current)) {
             let element = scrollRef.current.querySelector(`[data-key="${manager.focusedKey}"]`);
-            if (element) {
-                (0, $glPPV$reactariautils.scrollIntoView)(scrollRef.current, element);
-                (0, $glPPV$reactariautils.scrollIntoViewport)(element, {
-                    containingElement: ref.current
-                });
+            if (element && modality === 'keyboard') {
+                if (!element.previousSibling) {
+                    scrollRef.current.scrollTop = 0;
+                } else if (!element.nextSibling) {
+                    scrollRef.current.scrollTop = scrollRef.current.scrollHeight - scrollRef.current.clientHeight;
+                } else {
+                    $glPPV$reactariautils.scrollIntoView(scrollRef.current, element);
+                }
             }
         }
     }, [
diff --git a/dist/module.js b/dist/module.js
index b7f59c1ff16472ec3db0487957c6f6aad2794ff1..6e4b3f64205282e7a8949fc623738045d1d3f678 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -2,7 +2,7 @@ import {useRef as $eCAIO$useRef, useEffect as $eCAIO$useEffect, useMemo as $eCAI
 import {getFocusableTreeWalker as $eCAIO$getFocusableTreeWalker, focusSafely as $eCAIO$focusSafely} from "@react-aria/focus";
 import {focusWithoutScrolling as $eCAIO$focusWithoutScrolling, useEvent as $eCAIO$useEvent, scrollIntoView as $eCAIO$scrollIntoView, scrollIntoViewport as $eCAIO$scrollIntoViewport, mergeProps as $eCAIO$mergeProps, isAppleDevice as $eCAIO$isAppleDevice, isMac as $eCAIO$isMac} from "@react-aria/utils";
 import {useLocale as $eCAIO$useLocale, useCollator as $eCAIO$useCollator} from "@react-aria/i18n";
-import {usePress as $eCAIO$usePress, useLongPress as $eCAIO$useLongPress} from "@react-aria/interactions";
+import {usePress as $eCAIO$usePress, useLongPress as $eCAIO$useLongPress, getInteractionModality} from "@react-aria/interactions";
 
 /*
  * Copyright 2020 Adobe. All rights reserved.
@@ -303,14 +303,25 @@ function $ae20dd8cbca75726$export$d6daf82dcd84e87c(options) {
     }, []);
     // If not virtualized, scroll the focused element into view when the focusedKey changes.
     // When virtualized, Virtualizer handles this internally.
+    //
+    // NOTE(calebmer): Use our preferred scroll behavior for selectable collections.
+    // We implement this behavior elsewhere in the product like
+    // `<ContentEditorMentionFloater>`. Consolidate?
+    //
+    // 1. Don't scroll on hover. Only scroll on mouse movement.
+    // 2. If scrolling to the first or last element then completely scroll the view.
     (0, $eCAIO$useEffect)(()=>{
+        let modality = getInteractionModality();
         if (!isVirtualized && manager.focusedKey && (scrollRef === null || scrollRef === void 0 ? void 0 : scrollRef.current)) {
             let element = scrollRef.current.querySelector(`[data-key="${manager.focusedKey}"]`);
-            if (element) {
-                (0, $eCAIO$scrollIntoView)(scrollRef.current, element);
-                (0, $eCAIO$scrollIntoViewport)(element, {
-                    containingElement: ref.current
-                });
+            if (element && modality === 'keyboard') {
+                if (!element.previousSibling) {
+                    scrollRef.current.scrollTop = 0;
+                } else if (!element.nextSibling) {
+                    scrollRef.current.scrollTop = scrollRef.current.scrollHeight - scrollRef.current.clientHeight;
+                } else {
+                    (0, $eCAIO$scrollIntoView)(scrollRef.current, element);
+                }
             }
         }
     }, [
diff --git a/src/useSelectableCollection.ts b/src/useSelectableCollection.ts
index 515a7bb4267bbe92ad5dc2702bf6ea469bdc857c..7525fc4ff0d0aa292b3b9098763ea4b5d69c987d 100644
--- a/src/useSelectableCollection.ts
+++ b/src/useSelectableCollection.ts
@@ -18,6 +18,7 @@ import {isCtrlKeyPressed, isNonContiguousSelectionModifier} from './utils';
 import {MultipleSelectionManager} from '@react-stately/selection';
 import {useLocale} from '@react-aria/i18n';
 import {useTypeSelect} from './useTypeSelect';
+import {getInteractionModality} from '@react-aria/interactions';
 
 export interface AriaSelectableCollectionOptions {
   /**
@@ -355,12 +356,25 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
 
   // If not virtualized, scroll the focused element into view when the focusedKey changes.
   // When virtualized, Virtualizer handles this internally.
+  //
+  // NOTE(calebmer): Use our preferred scroll behavior for selectable collections.
+  // We implement this behavior elsewhere in the product like
+  // `<ContentEditorMentionFloater>`. Consolidate?
+  //
+  // 1. Don't scroll on hover. Only scroll on mouse movement.
+  // 2. If scrolling to the first or last element then completely scroll the view.
   useEffect(() => {
+    let modality = getInteractionModality();
     if (!isVirtualized && manager.focusedKey && scrollRef?.current) {
       let element = scrollRef.current.querySelector(`[data-key="${manager.focusedKey}"]`) as HTMLElement;
-      if (element) {
-        scrollIntoView(scrollRef.current, element);
-        scrollIntoViewport(element, {containingElement: ref.current});
+      if (element && modality === 'keyboard') {
+        if (!element.previousSibling) {
+          scrollRef.current.scrollTop = 0;
+        } else if (!element.nextSibling) {
+          scrollRef.current.scrollTop = scrollRef.current.scrollHeight - scrollRef.current.clientHeight;
+        } else {
+          scrollIntoView(scrollRef.current, element);
+        }
       }
     }
   }, [isVirtualized, scrollRef, manager.focusedKey, ref]);