diff --git a/dist/import.mjs b/dist/import.mjs
index 8ff0b5bd929ab46203f2e719793c4af61dc1711c..d4dac1c2e36d9b111d056044fdce0402bf370975 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -836,10 +836,13 @@ function $3dfb0f96be0d6a08$export$4a931266a3838b86(state, ref, disableArrowNavig
         }
     };
     // Focus the first placeholder segment from the end on mouse down/touch up in the field.
-    let focusLast = ()=>{
-        var _window_event;
+    //
+    // NOTE(calebmer): `window.event` is unreliable for getting the element to
+    // focus on. In development we've found a global `MessageEvent` here with a
+    // non-DOM element target that causes errors. Instead, explicitly pass in the
+    // target from the event we care about.
+    let focusLast = (target)=>{
         // Try to find the segment prior to the element that was clicked on.
-        let target = (_window_event = window.event) === null || _window_event === void 0 ? void 0 : _window_event.target;
         let walker = (0, $cIPI0$getFocusableTreeWalker)(ref.current, {
             tabbable: true
         });
@@ -866,15 +869,15 @@ function $3dfb0f96be0d6a08$export$4a931266a3838b86(state, ref, disableArrowNavig
     let { pressProps: pressProps  } = (0, $cIPI0$usePress)({
         preventFocusOnPress: true,
         allowTextSelectionOnPress: true,
-        onPressStart (e) {
-            if (e.pointerType === "mouse") focusLast();
-        },
         onPress (e) {
             if (e.pointerType !== "mouse") focusLast();
         }
     });
     return (0, $cIPI0$mergeProps)(pressProps, {
-        onKeyDown: onKeyDown
+        onKeyDown: onKeyDown,
+        onMouseDown (e) {
+            focusLast(e.target);
+        }
     });
 }
 
diff --git a/dist/main.js b/dist/main.js
index 6f6dc83a926718624afd8aea18800ad7f2293eba..2e9ad1df99f3f4d3075b8911f8bea6f1ae5887a0 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -846,10 +846,13 @@ function $715562ad3b4cced4$export$4a931266a3838b86(state, ref, disableArrowNavig
         }
     };
     // Focus the first placeholder segment from the end on mouse down/touch up in the field.
-    let focusLast = ()=>{
-        var _window_event;
+    //
+    // NOTE(calebmer): `window.event` is unreliable for getting the element to
+    // focus on. In development we've found a global `MessageEvent` here with a
+    // non-DOM element target that causes errors. Instead, explicitly pass in the
+    // target from the event we care about.
+    let focusLast = (target)=>{
         // Try to find the segment prior to the element that was clicked on.
-        let target = (_window_event = window.event) === null || _window_event === void 0 ? void 0 : _window_event.target;
         let walker = (0, $IwcIq$reactariafocus.getFocusableTreeWalker)(ref.current, {
             tabbable: true
         });
@@ -876,15 +879,15 @@ function $715562ad3b4cced4$export$4a931266a3838b86(state, ref, disableArrowNavig
     let { pressProps: pressProps  } = (0, $IwcIq$reactariainteractions.usePress)({
         preventFocusOnPress: true,
         allowTextSelectionOnPress: true,
-        onPressStart (e) {
-            if (e.pointerType === "mouse") focusLast();
-        },
         onPress (e) {
             if (e.pointerType !== "mouse") focusLast();
         }
     });
     return (0, $IwcIq$reactariautils.mergeProps)(pressProps, {
-        onKeyDown: onKeyDown
+        onKeyDown: onKeyDown,
+        onMouseDown (e) {
+            focusLast(e.target);
+        }
     });
 }
 
diff --git a/dist/module.js b/dist/module.js
index 8ff0b5bd929ab46203f2e719793c4af61dc1711c..d4dac1c2e36d9b111d056044fdce0402bf370975 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -836,10 +836,13 @@ function $3dfb0f96be0d6a08$export$4a931266a3838b86(state, ref, disableArrowNavig
         }
     };
     // Focus the first placeholder segment from the end on mouse down/touch up in the field.
-    let focusLast = ()=>{
-        var _window_event;
+    //
+    // NOTE(calebmer): `window.event` is unreliable for getting the element to
+    // focus on. In development we've found a global `MessageEvent` here with a
+    // non-DOM element target that causes errors. Instead, explicitly pass in the
+    // target from the event we care about.
+    let focusLast = (target)=>{
         // Try to find the segment prior to the element that was clicked on.
-        let target = (_window_event = window.event) === null || _window_event === void 0 ? void 0 : _window_event.target;
         let walker = (0, $cIPI0$getFocusableTreeWalker)(ref.current, {
             tabbable: true
         });
@@ -866,15 +869,15 @@ function $3dfb0f96be0d6a08$export$4a931266a3838b86(state, ref, disableArrowNavig
     let { pressProps: pressProps  } = (0, $cIPI0$usePress)({
         preventFocusOnPress: true,
         allowTextSelectionOnPress: true,
-        onPressStart (e) {
-            if (e.pointerType === "mouse") focusLast();
-        },
         onPress (e) {
             if (e.pointerType !== "mouse") focusLast();
         }
     });
     return (0, $cIPI0$mergeProps)(pressProps, {
-        onKeyDown: onKeyDown
+        onKeyDown: onKeyDown,
+        onMouseDown (e) {
+            focusLast(e.target);
+        }
     });
 }
 
diff --git a/src/useDatePickerGroup.ts b/src/useDatePickerGroup.ts
index d09cbc5cc433adc6016a100526be39c460933033..04c31d1ebea546235ae6979ff6117e06309bd108 100644
--- a/src/useDatePickerGroup.ts
+++ b/src/useDatePickerGroup.ts
@@ -45,9 +45,13 @@ export function useDatePickerGroup(state: DatePickerState | DateRangePickerState
   };
 
   // Focus the first placeholder segment from the end on mouse down/touch up in the field.
-  let focusLast = () => {
+  //
+  // NOTE(calebmer): `window.event` is unreliable for getting the element to
+  // focus on. In development we've found a global `MessageEvent` here with a
+  // non-DOM element target that causes errors. Instead, explicitly pass in the
+  // target from the event we care about.
+  let focusLast = (target?: FocusableElement) => {
     // Try to find the segment prior to the element that was clicked on.
-    let target = window.event?.target as FocusableElement;
     let walker = getFocusableTreeWalker(ref.current, {tabbable: true});
     if (target) {
       walker.currentNode = target;
@@ -83,11 +87,6 @@ export function useDatePickerGroup(state: DatePickerState | DateRangePickerState
   let {pressProps} = usePress({
     preventFocusOnPress: true,
     allowTextSelectionOnPress: true,
-    onPressStart(e) {
-      if (e.pointerType === 'mouse') {
-        focusLast();
-      }
-    },
     onPress(e) {
       if (e.pointerType !== 'mouse') {
         focusLast();
@@ -95,5 +94,10 @@ export function useDatePickerGroup(state: DatePickerState | DateRangePickerState
     }
   });
 
-  return mergeProps(pressProps, {onKeyDown});
+  return mergeProps(pressProps, {
+    onKeyDown,
+    onMouseDown(e) {
+      focusLast(e.target);
+    }
+  });
 }
