diff --git a/dist/import.mjs b/dist/import.mjs
index 7361913febdd7b6395867621a1d2c94203dcca77..59f996ab9804b2182a0bf585c0e470940712b819 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -327,7 +327,17 @@ function $9bf71ea28793e738$var$useFocusContainment(scopeRef, contain) {
             if (raf.current) cancelAnimationFrame(raf.current);
             raf.current = requestAnimationFrame(()=>{
                 // Use document.activeElement instead of e.relatedTarget so we can tell if user clicked into iframe
-                if ($9bf71ea28793e738$var$shouldContainFocus(scopeRef) && !$9bf71ea28793e738$var$isElementInChildScope(document.activeElement, scopeRef)) {
+                //
+                // NOTE(calebmer): If blur moved focus to the body (as calling `element.blur()`
+                // does) then instead of returning focus to the last focused element, instead
+                // move focus to the first element in our active scope. We have `tabIndex="-1"`
+                // on our `<Modal>` component so it can receive focus.
+                //
+                // This is important since we render peeks in `<SearchModal>` which may call
+                // `blur()`. We want `blur()` to have expected platform behavior. An example
+                // you can test is `<TaskCollectionInput>` in `<SearchModal>` which calls
+                // `blur()` after you pick a collection.
+                if ((document.activeElement !== document.body || !$9bf71ea28793e738$var$activeScope) && $9bf71ea28793e738$var$shouldContainFocus(scopeRef) && !$9bf71ea28793e738$var$isElementInChildScope(document.activeElement, scopeRef)) {
                     $9bf71ea28793e738$var$activeScope = scopeRef;
                     if (document.body.contains(e.target)) {
                         focusedNode.current = e.target;
@@ -363,8 +373,35 @@ function $9bf71ea28793e738$var$useFocusContainment(scopeRef, contain) {
 function $9bf71ea28793e738$var$isElementInAnyScope(element) {
     return $9bf71ea28793e738$var$isElementInChildScope(element);
 }
-function $9bf71ea28793e738$var$isElementInScope(element, scope) {
-    return scope.some((node)=>node.contains(element));
+function $9bf71ea28793e738$var$isElementInScope(childElement, scope) {
+    // NOTE(calebmer): Instead of a simple `element.contains()` implementation,
+    // follow `data-ownedby` attributes which we set on overlays so we can
+    // properly determine that portaled overlay elements are a child of the
+    // element which spawned them.
+    return scope.some((ownerElement) => {
+        const seenElements = new Set();
+
+        let currentElement = childElement;
+        while (currentElement !== null) {
+            if (currentElement === ownerElement) return true;
+
+            if (seenElements.has(currentElement)) throw new Error("Element ownership cycle detected");
+            seenElements.add(currentElement);
+
+            const ownerElementId = currentElement.getAttribute("data-ownedby");
+            const currentOwnerElement = ownerElementId
+                ? document.getElementById(ownerElementId)
+                : null;
+
+            let currentParentElement = currentElement.parentNode;
+            while (currentParentElement !== null && !(currentParentElement instanceof Element))
+                currentParentElement = currentParentElement.parentNode;
+
+            currentElement = currentOwnerElement ?? currentParentElement;
+        }
+
+        return false;
+    });
 }
 function $9bf71ea28793e738$var$isElementInChildScope(element, scope = null) {
     // If the element is within a top layer element (e.g. toasts), always allow moving focus there.
@@ -915,4 +952,5 @@ function $83013635b024ae3d$export$eac1895992b9f3d6(ref, options) {
 
 
 export {$9bf71ea28793e738$export$20e40289641fbbb6 as FocusScope, $9bf71ea28793e738$export$10c5169755ce7bd7 as useFocusManager, $9bf71ea28793e738$export$2d6ec8fc375ceafa as getFocusableTreeWalker, $9bf71ea28793e738$export$c5251b9e124bf29 as createFocusManager, $9bf71ea28793e738$export$1258395f99bf9cbf as isElementInChildOfActiveScope, $907718708eab68af$export$1a38b4ad7f578e1d as FocusRing, $e6afbd83fe6ebbd2$export$13f3202a3e5ddd5 as FocusableProvider, $e6afbd83fe6ebbd2$export$4c014de7c8940b4c as useFocusable, $f7dceffc5ad7768b$export$4e328f61c538687f as useFocusRing, $6a99195332edec8b$export$80f3e147d781571c as focusSafely, $83013635b024ae3d$export$eac1895992b9f3d6 as useHasTabbableChild};
-//# sourceMappingURL=module.js.map
+// NOTE(calebmer): Make it easier to debug the patched code by removing
+// source maps.
diff --git a/dist/main.js b/dist/main.js
index afa0b41d35e67a2751887c0586f637859606e6e6..99ceebdb6b1ae5d83f9a6aa42776aa6e4186f27b 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -345,7 +345,17 @@ function $a7a032acae3ddda9$var$useFocusContainment(scopeRef, contain) {
             if (raf.current) cancelAnimationFrame(raf.current);
             raf.current = requestAnimationFrame(()=>{
                 // Use document.activeElement instead of e.relatedTarget so we can tell if user clicked into iframe
-                if ($a7a032acae3ddda9$var$shouldContainFocus(scopeRef) && !$a7a032acae3ddda9$var$isElementInChildScope(document.activeElement, scopeRef)) {
+                //
+                // NOTE(calebmer): If blur moved focus to the body (as calling `element.blur()`
+                // does) then instead of returning focus to the last focused element, instead
+                // move focus to the first element in our active scope. We have `tabIndex="-1"`
+                // on our `<Modal>` component so it can receive focus.
+                //
+                // This is important since we render peeks in `<SearchModal>` which may call
+                // `blur()`. We want `blur()` to have expected platform behavior. An example
+                // you can test is `<TaskCollectionInput>` in `<SearchModal>` which calls
+                // `blur()` after you pick a collection.
+                if ((document.activeElement !== document.body || !$a7a032acae3ddda9$var$activeScope) && $a7a032acae3ddda9$var$shouldContainFocus(scopeRef) && !$a7a032acae3ddda9$var$isElementInChildScope(document.activeElement, scopeRef)) {
                     $a7a032acae3ddda9$var$activeScope = scopeRef;
                     if (document.body.contains(e.target)) {
                         focusedNode.current = e.target;
@@ -381,8 +391,35 @@ function $a7a032acae3ddda9$var$useFocusContainment(scopeRef, contain) {
 function $a7a032acae3ddda9$var$isElementInAnyScope(element) {
     return $a7a032acae3ddda9$var$isElementInChildScope(element);
 }
-function $a7a032acae3ddda9$var$isElementInScope(element, scope) {
-    return scope.some((node)=>node.contains(element));
+function $a7a032acae3ddda9$var$isElementInScope(childElement, scope) {
+    // NOTE(calebmer): Instead of a simple `element.contains()` implementation,
+    // follow `data-ownedby` attributes which we set on overlays so we can
+    // properly determine that portaled overlay elements are a child of the
+    // element which spawned them.
+    return scope.some((ownerElement) => {
+        const seenElements = new Set();
+
+        let currentElement = childElement;
+        while (currentElement !== null) {
+            if (currentElement === ownerElement) return true;
+
+            if (seenElements.has(currentElement)) throw new Error("Element ownership cycle detected");
+            seenElements.add(currentElement);
+
+            const ownerElementId = currentElement.getAttribute("data-ownedby");
+            const currentOwnerElement = ownerElementId
+                ? document.getElementById(ownerElementId)
+                : null;
+
+            let currentParentElement = currentElement.parentNode;
+            while (currentParentElement !== null && !(currentParentElement instanceof Element))
+                currentParentElement = currentParentElement.parentNode;
+
+            currentElement = currentOwnerElement ?? currentParentElement;
+        }
+
+        return false;
+    });
 }
 function $a7a032acae3ddda9$var$isElementInChildScope(element, scope = null) {
     // If the element is within a top layer element (e.g. toasts), always allow moving focus there.
@@ -932,4 +969,5 @@ function $259c6413a286f2e6$export$eac1895992b9f3d6(ref, options) {
 
 
 
-//# sourceMappingURL=main.js.map
+// NOTE(calebmer): Make it easier to debug the patched code by removing
+// source maps.
diff --git a/dist/module.js b/dist/module.js
index 7361913febdd7b6395867621a1d2c94203dcca77..59f996ab9804b2182a0bf585c0e470940712b819 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -327,7 +327,17 @@ function $9bf71ea28793e738$var$useFocusContainment(scopeRef, contain) {
             if (raf.current) cancelAnimationFrame(raf.current);
             raf.current = requestAnimationFrame(()=>{
                 // Use document.activeElement instead of e.relatedTarget so we can tell if user clicked into iframe
-                if ($9bf71ea28793e738$var$shouldContainFocus(scopeRef) && !$9bf71ea28793e738$var$isElementInChildScope(document.activeElement, scopeRef)) {
+                //
+                // NOTE(calebmer): If blur moved focus to the body (as calling `element.blur()`
+                // does) then instead of returning focus to the last focused element, instead
+                // move focus to the first element in our active scope. We have `tabIndex="-1"`
+                // on our `<Modal>` component so it can receive focus.
+                //
+                // This is important since we render peeks in `<SearchModal>` which may call
+                // `blur()`. We want `blur()` to have expected platform behavior. An example
+                // you can test is `<TaskCollectionInput>` in `<SearchModal>` which calls
+                // `blur()` after you pick a collection.
+                if ((document.activeElement !== document.body || !$9bf71ea28793e738$var$activeScope) && $9bf71ea28793e738$var$shouldContainFocus(scopeRef) && !$9bf71ea28793e738$var$isElementInChildScope(document.activeElement, scopeRef)) {
                     $9bf71ea28793e738$var$activeScope = scopeRef;
                     if (document.body.contains(e.target)) {
                         focusedNode.current = e.target;
@@ -363,8 +373,35 @@ function $9bf71ea28793e738$var$useFocusContainment(scopeRef, contain) {
 function $9bf71ea28793e738$var$isElementInAnyScope(element) {
     return $9bf71ea28793e738$var$isElementInChildScope(element);
 }
-function $9bf71ea28793e738$var$isElementInScope(element, scope) {
-    return scope.some((node)=>node.contains(element));
+function $9bf71ea28793e738$var$isElementInScope(childElement, scope) {
+    // NOTE(calebmer): Instead of a simple `element.contains()` implementation,
+    // follow `data-ownedby` attributes which we set on overlays so we can
+    // properly determine that portaled overlay elements are a child of the
+    // element which spawned them.
+    return scope.some((ownerElement) => {
+        const seenElements = new Set();
+
+        let currentElement = childElement;
+        while (currentElement !== null) {
+            if (currentElement === ownerElement) return true;
+
+            if (seenElements.has(currentElement)) throw new Error("Element ownership cycle detected");
+            seenElements.add(currentElement);
+
+            const ownerElementId = currentElement.getAttribute("data-ownedby");
+            const currentOwnerElement = ownerElementId
+                ? document.getElementById(ownerElementId)
+                : null;
+
+            let currentParentElement = currentElement.parentNode;
+            while (currentParentElement !== null && !(currentParentElement instanceof Element))
+                currentParentElement = currentParentElement.parentNode;
+
+            currentElement = currentOwnerElement ?? currentParentElement;
+        }
+
+        return false;
+    });
 }
 function $9bf71ea28793e738$var$isElementInChildScope(element, scope = null) {
     // If the element is within a top layer element (e.g. toasts), always allow moving focus there.
@@ -915,4 +952,5 @@ function $83013635b024ae3d$export$eac1895992b9f3d6(ref, options) {
 
 
 export {$9bf71ea28793e738$export$20e40289641fbbb6 as FocusScope, $9bf71ea28793e738$export$10c5169755ce7bd7 as useFocusManager, $9bf71ea28793e738$export$2d6ec8fc375ceafa as getFocusableTreeWalker, $9bf71ea28793e738$export$c5251b9e124bf29 as createFocusManager, $9bf71ea28793e738$export$1258395f99bf9cbf as isElementInChildOfActiveScope, $907718708eab68af$export$1a38b4ad7f578e1d as FocusRing, $e6afbd83fe6ebbd2$export$13f3202a3e5ddd5 as FocusableProvider, $e6afbd83fe6ebbd2$export$4c014de7c8940b4c as useFocusable, $f7dceffc5ad7768b$export$4e328f61c538687f as useFocusRing, $6a99195332edec8b$export$80f3e147d781571c as focusSafely, $83013635b024ae3d$export$eac1895992b9f3d6 as useHasTabbableChild};
-//# sourceMappingURL=module.js.map
+// NOTE(calebmer): Make it easier to debug the patched code by removing
+// source maps.
diff --git a/src/FocusScope.tsx b/src/FocusScope.tsx
index dada73642c21d5232e1f34635b6b4f8a5df65c2f..5b53e89a9a7e083aa1499513110124a5fb2f8031 100644
--- a/src/FocusScope.tsx
+++ b/src/FocusScope.tsx
@@ -358,7 +358,17 @@ function useFocusContainment(scopeRef: RefObject<Element[]>, contain: boolean) {
       }
       raf.current = requestAnimationFrame(() => {
         // Use document.activeElement instead of e.relatedTarget so we can tell if user clicked into iframe
-        if (shouldContainFocus(scopeRef) && !isElementInChildScope(document.activeElement, scopeRef)) {
+        //
+        // NOTE(calebmer): If blur moved focus to the body (as calling `element.blur()`
+        // does) then instead of returning focus to the last focused element, instead
+        // move focus to the first element in our active scope. We have `tabIndex="-1"`
+        // on our `<Modal>` component so it can receive focus.
+        //
+        // This is important since we render peeks in `<SearchModal>` which may call
+        // `blur()`. We want `blur()` to have expected platform behavior. An example
+        // you can test is `<TaskCollectionInput>` in `<SearchModal>` which calls
+        // `blur()` after you pick a collection.
+        if ((document.activeElement !== document.body || !activeScope) && shouldContainFocus(scopeRef) && !isElementInChildScope(document.activeElement, scopeRef)) {
           activeScope = scopeRef;
           if (document.body.contains(e.target)) {
             focusedNode.current = e.target;
@@ -397,8 +407,35 @@ function isElementInAnyScope(element: Element) {
   return isElementInChildScope(element);
 }
 
-function isElementInScope(element: Element, scope: Element[]) {
-  return scope.some(node => node.contains(element));
+function isElementInScope(childElement: Element, scope: Element[]) {
+  // NOTE(calebmer): Instead of a simple `element.contains()` implementation,
+  // follow `data-ownedby` attributes which we set on overlays so we can
+  // properly determine that portaled overlay elements are a child of the
+  // element which spawned them.
+  return scope.some((ownerElement) => {
+    const seenElements = new Set();
+
+    let currentElement = childElement;
+    while (currentElement !== null) {
+      if (currentElement === ownerElement) return true;
+
+      if (seenElements.has(currentElement)) throw new Error("Element ownership cycle detected");
+      seenElements.add(currentElement);
+
+      const ownerElementId = currentElement.getAttribute("data-ownedby");
+      const currentOwnerElement = ownerElementId
+        ? document.getElementById(ownerElementId)
+        : null;
+
+      let currentParentElement = currentElement.parentNode;
+      while (currentParentElement !== null && !(currentParentElement instanceof Element))
+        currentParentElement = currentParentElement.parentNode;
+
+      currentElement = currentOwnerElement ?? currentParentElement;
+    }
+
+    return false;
+  });
 }
 
 function isElementInChildScope(element: Element, scope: ScopeRef = null) {
