diff --git a/dist/import.mjs b/dist/import.mjs
index 8304c96402b2f2b285445c005b5078b34993a100..8ac10d2330fdebd9cd5320aee6e8b7bab50f17f9 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -885,7 +885,13 @@ function $a1ea59d68270f0dd$export$f8168d8dd8fd66e6(props) {
 
 
 
-let $507fabe10e71c6fb$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $507fabe10e71c6fb$var$currentModality = "pointer";
 let $507fabe10e71c6fb$var$changeHandlers = new Set();
 let $507fabe10e71c6fb$var$hasSetupGlobalListeners = false;
 let $507fabe10e71c6fb$var$hasEventBeforeFocus = false;
@@ -1348,25 +1354,41 @@ function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
  * governing permissions and limitations under the License.
  */ function $93925083ecbb358c$export$48d1ea6320830260(handler) {
     if (!handler) return;
-    let shouldStopPropagation = true;
     return (e)=>{
         let event = {
             ...e,
             preventDefault () {
                 e.preventDefault();
+
+                // NOTE(calebmer): We don't like the `react-spectrum` convention
+                // of `event.stopPropagation()` by default. With that convention
+                // `<GlobalKeyDownEvent>` doesn't work when focus is in a
+                // `react-spectrum` component! So we remove the default
+                // `event.stopPropagation()`.
+                //
+                // In order to avoid breaking `react-spectrum` components which,
+                // when `event.preventDefault()` is called we also call
+                // `event.stopPropagation()`.
+                //
+                // This is the same as putting
+                // `if (!event.isDefaultPrevented()) event.continuePropagation()`
+                // in every `react-spectrum` component which we'd do if we
+                // didn't have this patch.
+                e.stopPropagation();
             },
             isDefaultPrevented () {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for
+                // `keydown` events. Don't warn when we do this. Treat repeat
+                // calls a noop like regular DOM events.
+                e.stopPropagation();
             },
-            continuePropagation () {
-                shouldStopPropagation = false;
-            }
+            continuePropagation () {}
         };
         handler(event);
-        if (shouldStopPropagation) e.stopPropagation();
     };
 }
 
diff --git a/dist/main.js b/dist/main.js
index dca2b48e87b91891af244269a71d87c3162a7006..7e8fb16c9243d7a071755b965f5c06f201025c93 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -910,7 +910,13 @@ function $5cb73d0ce355b0dc$export$f8168d8dd8fd66e6(props) {
 
 
 
-let $e77252a287ef94ab$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $e77252a287ef94ab$var$currentModality = "pointer";
 let $e77252a287ef94ab$var$changeHandlers = new Set();
 let $e77252a287ef94ab$var$hasSetupGlobalListeners = false;
 let $e77252a287ef94ab$var$hasEventBeforeFocus = false;
@@ -1373,25 +1379,41 @@ function $edcfa848c42f94f4$var$isValidEvent(event, ref) {
  * governing permissions and limitations under the License.
  */ function $951fbcbbca8db6ce$export$48d1ea6320830260(handler) {
     if (!handler) return;
-    let shouldStopPropagation = true;
     return (e)=>{
         let event = {
             ...e,
             preventDefault () {
                 e.preventDefault();
+
+                // NOTE(calebmer): We don't like the `react-spectrum` convention
+                // of `event.stopPropagation()` by default. With that convention
+                // `<GlobalKeyDownEvent>` doesn't work when focus is in a
+                // `react-spectrum` component! So we remove the default
+                // `event.stopPropagation()`.
+                //
+                // In order to avoid breaking `react-spectrum` components which,
+                // when `event.preventDefault()` is called we also call
+                // `event.stopPropagation()`.
+                //
+                // This is the same as putting
+                // `if (!event.isDefaultPrevented()) event.continuePropagation()`
+                // in every `react-spectrum` component which we'd do if we
+                // didn't have this patch.
+                e.stopPropagation();
             },
             isDefaultPrevented () {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
+                e.stopPropagation();
             },
-            continuePropagation () {
-                shouldStopPropagation = false;
-            }
+            continuePropagation () {}
         };
         handler(event);
-        if (shouldStopPropagation) e.stopPropagation();
     };
 }
 
diff --git a/dist/module.js b/dist/module.js
index 8304c96402b2f2b285445c005b5078b34993a100..492a2341131b88ac838afe622ed535e22fd60ae4 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -885,7 +885,13 @@ function $a1ea59d68270f0dd$export$f8168d8dd8fd66e6(props) {
 
 
 
-let $507fabe10e71c6fb$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $507fabe10e71c6fb$var$currentModality = "pointer";
 let $507fabe10e71c6fb$var$changeHandlers = new Set();
 let $507fabe10e71c6fb$var$hasSetupGlobalListeners = false;
 let $507fabe10e71c6fb$var$hasEventBeforeFocus = false;
@@ -1348,25 +1354,41 @@ function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
  * governing permissions and limitations under the License.
  */ function $93925083ecbb358c$export$48d1ea6320830260(handler) {
     if (!handler) return;
-    let shouldStopPropagation = true;
     return (e)=>{
         let event = {
             ...e,
             preventDefault () {
                 e.preventDefault();
+
+                // NOTE(calebmer): We don't like the `react-spectrum` convention
+                // of `event.stopPropagation()` by default. With that convention
+                // `<GlobalKeyDownEvent>` doesn't work when focus is in a
+                // `react-spectrum` component! So we remove the default
+                // `event.stopPropagation()`.
+                //
+                // In order to avoid breaking `react-spectrum` components which,
+                // when `event.preventDefault()` is called we also call
+                // `event.stopPropagation()`.
+                //
+                // This is the same as putting
+                // `if (!event.isDefaultPrevented()) event.continuePropagation()`
+                // in every `react-spectrum` component which we'd do if we
+                // didn't have this patch.
+                e.stopPropagation();
             },
             isDefaultPrevented () {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
+                e.stopPropagation();
             },
-            continuePropagation () {
-                shouldStopPropagation = false;
-            }
+            continuePropagation () {}
         };
         handler(event);
-        if (shouldStopPropagation) e.stopPropagation();
     };
 }
 
diff --git a/src/createEventHandler.ts b/src/createEventHandler.ts
index 5733c7e4221092879a6409ca983e919edff0da87..cb03a60028df9172451768e856dd614baf3a0b05 100644
--- a/src/createEventHandler.ts
+++ b/src/createEventHandler.ts
@@ -21,28 +21,41 @@ export function createEventHandler<T extends SyntheticEvent>(handler: (e: BaseEv
     return;
   }
 
-  let shouldStopPropagation = true;
   return (e: T) => {
     let event: BaseEvent<T> = {
       ...e,
       preventDefault() {
         e.preventDefault();
+
+        // NOTE(calebmer): We don't like the `react-spectrum` convention
+        // of `event.stopPropagation()` by default. With that convention
+        // `<GlobalKeyDownEvent>` doesn't work when focus is in a
+        // `react-spectrum` component! So we remove the default
+        // `event.stopPropagation()`.
+        //
+        // In order to avoid breaking `react-spectrum` components which,
+        // when `event.preventDefault()` is called we also call
+        // `event.stopPropagation()`.
+        //
+        // This is the same as putting
+        // `if (!event.isDefaultPrevented()) event.continuePropagation()`
+        // in every `react-spectrum` component which we'd do if we
+        // didn't have this patch.
+        e.stopPropagation();
       },
       isDefaultPrevented() {
         return e.isDefaultPrevented();
       },
       stopPropagation() {
-        console.error('stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.');
+        // NOTE(calebmer): Our style guide recommends calling both
+        // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+        // Don't warn when we do this. Treat repeat calls a noop like regular
+        // DOM events.
+        e.stopPropagation();
       },
-      continuePropagation() {
-        shouldStopPropagation = false;
-      }
+      continuePropagation() {}
     };
 
     handler(event);
-
-    if (shouldStopPropagation) {
-      e.stopPropagation();
-    }
   };
 }
diff --git a/src/useFocusVisible.ts b/src/useFocusVisible.ts
index 60fd6b5a77f5d9f26ffdb6e750e11a0cab6923e4..db1066703ef1c941ab6ee4c422b98931bf93c1f9 100644
--- a/src/useFocusVisible.ts
+++ b/src/useFocusVisible.ts
@@ -35,7 +35,13 @@ export interface FocusVisibleResult {
   isFocusVisible: boolean
 }
 
-let currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let currentModality = 'pointer';
 let changeHandlers = new Set<Handler>();
 let hasSetupGlobalListeners = false;
 let hasEventBeforeFocus = false;