diff --git a/dist/import.mjs b/dist/import.mjs
index 4530866c54f7a116b7aa9c3d168a8e84d502fc6f..3ffc417e79b11296c2f9d6b3002b0efe9e268d7f 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -805,7 +805,13 @@ function $a1ea59d68270f0dd$export$f8168d8dd8fd66e6(props) {
 // See https://github.com/facebook/react/tree/cc7c1aece46a6b69b41958d731e0fd27c94bfc6c/packages/react-interactions
 
 
-let $507fabe10e71c6fb$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $507fabe10e71c6fb$var$currentModality = "pointer";
 let $507fabe10e71c6fb$var$changeHandlers = new Set();
 let $507fabe10e71c6fb$var$hasSetupGlobalListeners = false;
 let $507fabe10e71c6fb$var$hasEventBeforeFocus = false;
@@ -1276,7 +1282,10 @@ function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
             },
             continuePropagation () {
                 shouldStopPropagation = false;
diff --git a/dist/main.js b/dist/main.js
index d97dd73099f89cf3d1e4f2d20367cb0eaa5c1f08..6a795358bcf4e48e211d1e94c57d3fc3a7f694de 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -829,7 +829,13 @@ function $5cb73d0ce355b0dc$export$f8168d8dd8fd66e6(props) {
 // See https://github.com/facebook/react/tree/cc7c1aece46a6b69b41958d731e0fd27c94bfc6c/packages/react-interactions
 
 
-let $e77252a287ef94ab$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $e77252a287ef94ab$var$currentModality = "pointer";
 let $e77252a287ef94ab$var$changeHandlers = new Set();
 let $e77252a287ef94ab$var$hasSetupGlobalListeners = false;
 let $e77252a287ef94ab$var$hasEventBeforeFocus = false;
@@ -1300,7 +1306,10 @@ function $edcfa848c42f94f4$var$isValidEvent(event, ref) {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
             },
             continuePropagation () {
                 shouldStopPropagation = false;
diff --git a/dist/module.js b/dist/module.js
index 4530866c54f7a116b7aa9c3d168a8e84d502fc6f..3ffc417e79b11296c2f9d6b3002b0efe9e268d7f 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -805,7 +805,13 @@ function $a1ea59d68270f0dd$export$f8168d8dd8fd66e6(props) {
 // See https://github.com/facebook/react/tree/cc7c1aece46a6b69b41958d731e0fd27c94bfc6c/packages/react-interactions
 
 
-let $507fabe10e71c6fb$var$currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let $507fabe10e71c6fb$var$currentModality = "pointer";
 let $507fabe10e71c6fb$var$changeHandlers = new Set();
 let $507fabe10e71c6fb$var$hasSetupGlobalListeners = false;
 let $507fabe10e71c6fb$var$hasEventBeforeFocus = false;
@@ -1276,7 +1282,10 @@ function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
                 return e.isDefaultPrevented();
             },
             stopPropagation () {
-                console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.");
+                // NOTE(calebmer): Our style guide recommends calling both
+                // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+                // Don't warn when we do this. Treat repeat calls a noop like regular
+                // DOM events.
             },
             continuePropagation () {
                 shouldStopPropagation = false;
diff --git a/src/createEventHandler.ts b/src/createEventHandler.ts
index 5733c7e4221092879a6409ca983e919edff0da87..d748e49c5cda985cc5ae9f07edd7d2c7c7d6c874 100644
--- a/src/createEventHandler.ts
+++ b/src/createEventHandler.ts
@@ -32,7 +32,10 @@ export function createEventHandler<T extends SyntheticEvent>(handler: (e: BaseEv
         return e.isDefaultPrevented();
       },
       stopPropagation() {
-        console.error('stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.');
+        // NOTE(calebmer): Our style guide recommends calling both
+        // `event.preventDefault()` and `event.stopPropagation()` for `keydown` events.
+        // Don't warn when we do this. Treat repeat calls a noop like regular
+        // DOM events.
       },
       continuePropagation() {
         shouldStopPropagation = false;
diff --git a/src/useFocusVisible.ts b/src/useFocusVisible.ts
index 59ad8c87fd5e47f72aa33e634eaee29a9a2a6570..1767374069fbab3442cace1a4d7d9e62de8d7d59 100644
--- a/src/useFocusVisible.ts
+++ b/src/useFocusVisible.ts
@@ -34,7 +34,13 @@ export interface FocusVisibleResult {
   isFocusVisible: boolean
 }
 
-let currentModality = null;
+// NOTE(calebmer): Default modality to `pointer`. The types say modality is
+// non-null which isn't quite true. We expect pointer interaction to be the
+// dominant modality users use for our app. Also we found this fixes a bug with
+// our document editor when clicking in the sidebar jumps the cursor to the
+// start of the document because of a `useInteractionModality()` hook
+// scheduling a high priority update.
+let currentModality = 'pointer';
 let changeHandlers = new Set<Handler>();
 let hasSetupGlobalListeners = false;
 let hasEventBeforeFocus = false;