diff --git a/dist/components.js b/dist/components.js
index f94728730dba725146c1223659f4a7ba1af946b6..e7fd2e68d4631a28c729ab499fc06adc78a1a2b9 100644
--- a/dist/components.js
+++ b/dist/components.js
@@ -159,6 +159,13 @@ function RemixEntry({
 }
 
 function deserializeError(data) {
+  // NOTE(calebmer): We want to use our own error serialization/deserialization
+  // logic instead of the default logic Remix comes out of the box with. To do
+  // this we inject a global. Ideally there would be some first-class way other
+  // than injection to do this.
+  if (typeof __remixErrorSchema !== "undefined") {
+    return __remixErrorSchema.deserialize(data);
+  }
   let error = new Error(data.message);
   error.stack = data.stack;
   return error;
diff --git a/dist/data.js b/dist/data.js
index 7ee105347d2b523d74fe4a530b9287c859ad0504..03f2b144df33d94d624fabb4e47cecd23d461b69 100644
--- a/dist/data.js
+++ b/dist/data.js
@@ -33,6 +33,13 @@ async function fetchData(url, routeId, signal, submission) {
 
   if (isErrorResponse(response)) {
     let data = await response.json();
+    // NOTE(calebmer): We want to use our own error serialization/deserialization
+    // logic instead of the default logic Remix comes out of the box with. To do
+    // this we inject a global. Ideally there would be some first-class way other
+    // than injection to do this.
+    if (typeof __remixErrorSchema !== "undefined") {
+      return __remixErrorSchema.deserialize(data);
+    }
     let error = new Error(data.message);
     error.stack = data.stack;
     return error;
diff --git a/dist/esm/components.js b/dist/esm/components.js
index e884e7b984e4ef29fbcf2b2f5c9150513cc360f5..6a55f67235b9c1cdebd96dfd8cd8e8cc5f830c19 100644
--- a/dist/esm/components.js
+++ b/dist/esm/components.js
@@ -135,6 +135,13 @@ function RemixEntry({
 }
 
 function deserializeError(data) {
+  // NOTE(calebmer): We want to use our own error serialization/deserialization
+  // logic instead of the default logic Remix comes out of the box with. To do
+  // this we inject a global. Ideally there would be some first-class way other
+  // than injection to do this.
+  if (typeof __remixErrorSchema !== "undefined") {
+    return __remixErrorSchema.deserialize(data);
+  }
   let error = new Error(data.message);
   error.stack = data.stack;
   return error;
diff --git a/dist/esm/data.js b/dist/esm/data.js
index 69473f0d789453847a9880071387d8f96e0dabb1..da230284a1da41d1d939a7eeae79228c27ef7e66 100644
--- a/dist/esm/data.js
+++ b/dist/esm/data.js
@@ -29,6 +29,13 @@ async function fetchData(url, routeId, signal, submission) {
 
   if (isErrorResponse(response)) {
     let data = await response.json();
+    // NOTE(calebmer): We want to use our own error serialization/deserialization
+    // logic instead of the default logic Remix comes out of the box with. To do
+    // this we inject a global. Ideally there would be some first-class way other
+    // than injection to do this.
+    if (typeof __remixErrorSchema !== "undefined") {
+      return __remixErrorSchema.deserialize(data);
+    }
     let error = new Error(data.message);
     error.stack = data.stack;
     return error;
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 2e11fbc1fa3278bbdf1317f7c2a9533669b4de63..02cb4fd85475d56730a0eee36046af27e1c569bc 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -14,3 +14,10 @@ export { Form, Link, Links, LiveReload, Meta, NavLink, PrefetchPageLinks, Script
 export { useCatch } from './errorBoundaries.js';
 export { ScrollRestoration } from './scroll-restoration.js';
 export { RemixServer } from './server.js';
+
+// NOTE(calebmer): Export some extra Remix internals that we use. By exporting
+// these internals we make it harder to upgrade Remix and take responsibility
+// for correct usage of this dependency's internals.
+export { RemixEntryContext } from './components.js';
+export { matchClientRoutes } from './routeMatching.js';
+export { createTransitionManager } from './transition.js';
diff --git a/dist/index.d.ts b/dist/index.d.ts
index f4cf93355c80de06a2497d8589fc375f0defac63..32f9bf2d1145af91805e03cca26eff1ecb19c230 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -13,3 +13,12 @@ export { ScrollRestoration } from "./scroll-restoration";
 export type { RemixServerProps } from "./server";
 export { RemixServer } from "./server";
 export type { Fetcher } from "./transition";
+
+// NOTE(calebmer): Export some extra Remix internals that we use. By exporting
+// these internals we make it harder to upgrade Remix and take responsibility
+// for correct usage of this dependency's internals.
+export { RemixEntryContext } from "./components";
+export { matchClientRoutes } from "./routeMatching";
+export { createTransitionManager } from "./transition";
+export type { ClientRoute } from "./routes";
+export type { AppState } from "./errors";
diff --git a/dist/index.js b/dist/index.js
index f0f257179dc33aeb11dc060867afcfbaf3b69c62..822b10ca9de2aa35402b2d9d4c2532a2f1baa929 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -18,6 +18,9 @@ var components = require('./components.js');
 var errorBoundaries = require('./errorBoundaries.js');
 var scrollRestoration = require('./scroll-restoration.js');
 var server = require('./server.js');
+var routeMatching = require('./routeMatching.js');
+var routes = require('./routes.js');
+var transition = require('./transition.js');
 
 
 
@@ -82,3 +85,10 @@ exports.useTransition = components.useTransition;
 exports.useCatch = errorBoundaries.useCatch;
 exports.ScrollRestoration = scrollRestoration.ScrollRestoration;
 exports.RemixServer = server.RemixServer;
+
+// NOTE(calebmer): Export some extra Remix internals that we use. By exporting
+// these internals we make it harder to upgrade Remix and take responsibility
+// for correct usage of this dependency's internals.
+exports.RemixEntryContext = components.RemixEntryContext;
+exports.matchClientRoutes = routeMatching.matchClientRoutes;
+exports.createTransitionManager = transition.createTransitionManager;
