diff --git a/dist/compiler/js/compiler.js b/dist/compiler/js/compiler.js
index f76c5bab53a3d925d6b9b6450268508a6f78576c..6a8d785c98201964945ccd4ce38b9679cea5421e 100644
--- a/dist/compiler/js/compiler.js
+++ b/dist/compiler/js/compiler.js
@@ -31,6 +31,7 @@ var cssSideEffectImports = require('../plugins/cssSideEffectImports.js');
 var vanillaExtract = require('../plugins/vanillaExtract.js');
 var invariant = require('../../invariant.js');
 var hmr = require('./plugins/hmr.js');
+var fs = require('fs');
 
 function _interopNamespace(e) {
   if (e && e.__esModule) return e;
@@ -89,7 +90,10 @@ const createEsbuildConfig = (ctx, refs) => {
     hmr: ctx.options.mode === "development" && ctx.config.future.v2_dev !== false
   }), cssImports.cssFilePlugin(ctx), absoluteCssUrlsPlugin.absoluteCssUrlsPlugin(), external.externalPlugin(/^https?:\/\//, {
     sideEffects: false
-  }), mdx.mdxPlugin(ctx), emptyModules.emptyModulesPlugin(ctx, /\.server(\.[jt]sx?)?$/), emptyModules.emptyModulesPlugin(ctx, /^@remix-run\/(deno|cloudflare|node)(\/.*)?$/, {
+  // NOTE(calebmer): Don't include files in a `/server/` directory in the
+  // browser build in addition to not including files with a `.server.js`
+  // extension.
+  }), mdx.mdxPlugin(ctx), emptyModules.emptyModulesPlugin(ctx, /(\.server|\/server\/.*)(\.[mc]?[jt]sx?)?$/), emptyModules.emptyModulesPlugin(ctx, /^@remix-run\/(deno|cloudflare|node)(\/.*)?$/, {
     includeNodeModules: true
   }), esbuildPluginsNodeModulesPolyfill.nodeModulesPolyfillPlugin(), external.externalPlugin(/^node:.*/, {
     sideEffects: false
diff --git a/dist/compiler/manifest.js b/dist/compiler/manifest.js
index 1adfd7ee810bf08fd87cdcbc3d39e9e19526a9fa..4e3241908713bbe23ce850460cc64b2d273c5d83 100644
--- a/dist/compiler/manifest.js
+++ b/dist/compiler/manifest.js
@@ -60,7 +60,13 @@ async function create({
   for (let key of Object.keys(metafile.outputs).sort()) {
     let output = metafile.outputs[key];
     if (!output.entryPoint) continue;
-    if (path__namespace.resolve(output.entryPoint) === config.entryClientFilePath) {
+    // NOTE(calebmer): Bazel makes symlinks to files in runfiles directory
+    // so these two paths may be in different directories. Only look for whether
+    // the path after `/bazel-out/` is the same name as our entry client file.
+    // If the path is in a `.runfiles` directory we also want to ignore that.
+    const splitResolvedEntryPoint = path__namespace.resolve(output.entryPoint).split(/\/bazel-out\/[^/]+\/bin\/(?:.+\.runfiles\/[^/]+\/)?/);
+    const splitEntryClientFilePath = config.entryClientFilePath.split(/\/bazel-out\/[^/]+\/bin\/(?:.+\.runfiles\/[^/]+\/)?/);
+    if (splitResolvedEntryPoint.length === splitEntryClientFilePath.length && splitResolvedEntryPoint[splitResolvedEntryPoint.length - 1] === splitEntryClientFilePath[splitEntryClientFilePath.length - 1]) {
       entry = {
         module: resolveUrl(key),
         imports: resolveImports(output.imports)
diff --git a/dist/compiler/server/compiler.js b/dist/compiler/server/compiler.js
index 10850b2ec9650954dab23be2fd838ae41354e911..8fa4e8d05d7288f3dca57c7631d631b22bcdf468 100644
--- a/dist/compiler/server/compiler.js
+++ b/dist/compiler/server/compiler.js
@@ -29,6 +29,7 @@ var entry = require('./plugins/entry.js');
 var routes = require('./plugins/routes.js');
 var external = require('../plugins/external.js');
 var cssBundlePlugin = require('../plugins/cssBundlePlugin.js');
+var fs = require('fs');
 
 function _interopNamespace(e) {
   if (e && e.__esModule) return e;
@@ -117,7 +118,11 @@ const createEsbuildConfig = (ctx, refs) => {
 
     jsx: "automatic",
     jsxDev: ctx.options.mode !== "production",
-    plugins
+    plugins,
+    // NOTE(calebmer): Cloudflare uses this weird module import syntax to
+    // support worker site static assets.
+    // See: https://github.com/cloudflare/wrangler/pull/2126
+    external: ctx.config.serverConditions && ctx.config.serverConditions.includes("worker") ? ["__STATIC_CONTENT_MANIFEST"] : undefined,
   };
 };
 const create = async (ctx, refs) => {
diff --git a/dist/compiler/server/plugins/bareImports.js b/dist/compiler/server/plugins/bareImports.js
index 54f469ac40950ace84f0ccb9f8b40376b305b860..2fa03a1019217f49be1ce0d51774d5c2f33b6e41 100644
--- a/dist/compiler/server/plugins/bareImports.js
+++ b/dist/compiler/server/plugins/bareImports.js
@@ -109,7 +109,9 @@ function serverBareModulesPlugin(ctx) {
             return undefined;
           }
         }
-        if (!isNodeBuiltIn(packageName) && kind !== "dynamic-import" && ctx.config.serverPlatform === "node") {
+        // NOTE(calebmer): If our server module format is `esm` then it is ok to
+        // import an ESM-only package.
+        if (!isNodeBuiltIn(packageName) && kind !== "dynamic-import" && ctx.config.serverPlatform === "node" && ctx.config.serverModuleFormat !== "esm") {
           warnOnceIfEsmOnlyPackage(ctx, packageName, path$1, importer);
         }
 
diff --git a/dist/compiler/server/write.js b/dist/compiler/server/write.js
index aa35b4b724b363a955a1a6091c0ce4f523d5b0d6..aa783ead0ca1357e34adadec3ebc71b72b9e7d47 100644
--- a/dist/compiler/server/write.js
+++ b/dist/compiler/server/write.js
@@ -42,7 +42,10 @@ async function write(config, outputFiles) {
     if (file.path.endsWith(".js") || file.path.endsWith(".mjs")) {
       // fix sourceMappingURL to be relative to current path instead of /build
       let filename = file.path.substring(file.path.lastIndexOf(path__namespace.sep) + 1);
-      let escapedFilename = filename.replace(/\./g, "\\.");
+      // NOTE(calebmer): When we build for Cloudflare pages the server
+      // JavaScript file is named `[[path]].js`. The `[` and `]` characters must
+      // be escaped within a regex.
+      let escapedFilename = filename.replace(/\./g, "\\.").replace(/\[/g, "\\[").replace(/\]/g, "\\]");
       let pattern = `(//# sourceMappingURL=)(.*)${escapedFilename}`;
       let contents = Buffer.from(file.contents).toString("utf-8");
       contents = contents.replace(new RegExp(pattern), `$1${filename}`);
diff --git a/dist/config/flat-routes.js b/dist/config/flat-routes.js
index 3f394c82055229325aeafd04301664df1a3cb9c7..8d27f117cafd79c9410ad6cb4294b39908a46018 100644
--- a/dist/config/flat-routes.js
+++ b/dist/config/flat-routes.js
@@ -89,7 +89,12 @@ function flatRoutes(appDirectory, ignoredFilePatterns = [], prefix = "routes") {
     // If it's a directory, don't recurse into it, instead just look for a route module
     if (entry.isDirectory()) {
       route = findRouteModuleForFolder(appDirectory, filepath, ignoredFileRegex);
-    } else if (entry.isFile()) {
+    } else if (
+      entry.isFile() &&
+      // NOTE(calebmer): Ignore source maps which will be in the `routes` directory due to
+      // our Bazel build.
+      !entry.name.endsWith(".map")
+    ) {
       route = findRouteModuleForFile(appDirectory, filepath, ignoredFileRegex);
     }
     if (route) routes.push(route);
diff --git a/dist/config/routesConvention.js b/dist/config/routesConvention.js
index ca84eda743956c49910b1083fe6d02f390f20495..cdb48c77efba15b81ee349b2d1bcc80e68f1f07a 100644
--- a/dist/config/routesConvention.js
+++ b/dist/config/routesConvention.js
@@ -70,6 +70,15 @@ function defineConventionalRoutes(appDir, ignoredFilePatterns) {
       files[routeId] = path__namespace.join("routes", file);
       return;
     }
+
+    // NOTE(calebmer): When executing in a Bazel output directory we may have
+    // `.js` and `.js.map` since the `.js` file is transpiled from a `.ts` file.
+    // We want to ignore the `.js.map` file if the `.js` file is a valid
+    // route file.
+    if (file.endsWith(".map") && isRouteModuleFile(file.slice(0, -4))) {
+      return;
+    }
+
     throw new Error(`Invalid route module file: ${path__namespace.join(appDir, "routes", file)}`);
   });
   let routeIds = Object.keys(files).sort(byLongestFirst);