diff --git a/dist/config/flat-routes.js b/dist/config/flat-routes.js
index 70c183de6d9a19bd97e694a7c2e0f3ad67c0a0de..8dcbbc049418b85766b1865084cdfb5819781777 100644
--- a/dist/config/flat-routes.js
+++ b/dist/config/flat-routes.js
@@ -84,12 +84,23 @@ function flatRoutes(appDirectory, ignoredFilePatterns = [], prefix = "routes") {
   });
   let routes$1 = [];
   for (let entry of entries) {
+    let entryName = entry.name;
     let filepath = routes.normalizeSlashes(path__default["default"].join(routesDir, entry.name));
     let route = null;
+    // NOTE(calebmer): If the entry is a symbolic link then get the stats of the
+    // underlying link.
+    if (entry.isSymbolicLink()) {
+      entry = fs__default["default"].statSync(filepath);
+    }
     // If it's a directory, don't recurse into it, instead just look for a route module
     if (entry.isDirectory()) {
       route = findRouteModuleForFolder(appDirectory, filepath, ignoredFileRegex);
-    } else if (entry.isFile()) {
+    } else if (
+      entry.isFile() &&
+      // NOTE(calebmer): Ignore source maps which will be in the `routes` directory due to
+      // our Bazel build.
+      !entryName.endsWith(".map")
+    ) {
       route = findRouteModuleForFile(appDirectory, filepath, ignoredFileRegex);
     }
     if (route) routes$1.push(route);
diff --git a/dist/vite/plugin.js b/dist/vite/plugin.js
index bf608523a542d69398ed7afa51562c63a36ce8ed..7e02c8a8a86a15ddf0ea0e3c3e997a75ed2996a4 100644
--- a/dist/vite/plugin.js
+++ b/dist/vite/plugin.js
@@ -370,7 +370,8 @@ const remixVitePlugin = (remixUserConfig = {}) => {
     let buildDirectory = path__namespace.resolve(rootDirectory, resolvedRemixUserConfig.buildDirectory);
     let {
       serverBuildFile,
-      serverBundles
+      serverBundles,
+      additionalServerInputPath
     } = resolvedRemixUserConfig;
     let publicPath = viteUserConfig.base ?? "/";
     if (basename !== "/" && viteCommand === "serve" && !((_viteUserConfig$serve = viteUserConfig.server) !== null && _viteUserConfig$serve !== void 0 && _viteUserConfig$serve.middlewareMode) && !basename.startsWith(publicPath)) {
@@ -394,7 +395,8 @@ const remixVitePlugin = (remixUserConfig = {}) => {
       serverBuildFile,
       serverBundles,
       serverModuleFormat,
-      ssr
+      ssr,
+      additionalServerInputPath
     });
     for (let preset of remixUserConfig.presets ?? []) {
       var _preset$remixConfigRe;
@@ -691,9 +693,18 @@ const remixVitePlugin = (remixUserConfig = {}) => {
               rollupOptions: {
                 ...baseRollupOptions,
                 preserveEntrySignatures: "exports-only",
-                input: serverBuildId,
+                // NOTE(calebmer): We want to include an additional file in the Rollup server
+                // build. This additional file is our Node.js server. We want the server to
+                // share modules with the server code bundled by Rollup which is why it needs
+                // to be included in the Rollup build here.
+                input: ctx.remixConfig.additionalServerInputPath ? {
+                  [ctx.remixConfig.serverBuildFile]: serverBuildId,
+                  [path.basename(ctx.remixConfig.additionalServerInputPath)]: ctx.remixConfig.additionalServerInputPath,
+                } : {
+                  [ctx.remixConfig.serverBuildFile]: serverBuildId
+                },
                 output: {
-                  entryFileNames: ctx.remixConfig.serverBuildFile,
+                  entryFileNames: "[name]",
                   format: ctx.remixConfig.serverModuleFormat
                 }
               }
@@ -945,7 +956,11 @@ const remixVitePlugin = (remixUserConfig = {}) => {
       let resolved = await this.resolve(id, importer, options);
       if (!resolved) return;
       let serverFileRE = /\.server(\.[cm]?[jt]sx?)?$/;
-      let serverDirRE = /\/\.server\//;
+      // NOTE(calebmer): Don't include files in a `/server/` directory in the
+      // browser build in addition to not including files with a `.server.js`
+      // extension. Remix ignores files in a `/.server/` directory by default.
+      // We extend that to include `/server/` per our convention.
+      let serverDirRE = /\/\.?server\//;
       let isDotServer = serverFileRE.test(resolved.id) || serverDirRE.test(resolved.id);
       if (!isDotServer) return;
       if (!importer) return;
@@ -1043,29 +1058,14 @@ const remixVitePlugin = (remixUserConfig = {}) => {
       let isJSX = filepath.endsWith("x");
       let useFastRefresh = !ssr && (isJSX || code.includes(devRuntime));
       if (!useFastRefresh) return;
-      let result = await babel__default["default"].transformAsync(code, {
-        configFile: false,
-        babelrc: false,
-        filename: id,
-        sourceFileName: filepath,
-        parserOpts: {
-          sourceType: "module",
-          allowAwaitOutsideFunction: true
-        },
-        plugins: [[require("react-refresh/babel"), {
-          skipEnvCheck: true
-        }]],
-        sourceMaps: true
-      });
-      if (result === null) return;
-      code = result.code;
+      // NOTE(calebmer): Don't use Babel to perform `react-refresh` transform.
+      // The SWC transpiler run by Bazel performs the `react-refresh` transform.
       let refreshContentRE = /\$Refresh(?:Reg|Sig)\$\(/;
       if (refreshContentRE.test(code)) {
         code = addRefreshWrapper(ctx.remixConfig, code, id);
       }
       return {
         code,
-        map: result.map
       };
     }
   }, {
@@ -1149,6 +1149,20 @@ if (import.meta.hot && !inWebWorker) {
 }`;
 function getRoute(pluginConfig, file) {
   let vite = importViteEsmSync.importViteEsmSync();
+  // NOTE(calebmer): Vite may follow symlinks out of the runfiles directory we're
+  // building in. So `pluginConfig.appDirectory` will be in the runfiles
+  // directory but `file` won't be. Detect this case and update `file` to a path
+  // in the runfiles directory so we can properly check if `file` is a route
+  // or not.
+  if (!file.startsWith(`${pluginConfig.appDirectory}/`)) {
+    let fileParts = file.split(/\/bazel-out\/[^/]+\/bin\//, 2);
+    if (fileParts.length === 2) {
+      let match = pluginConfig.appDirectory.match(/^.*\/bazel-out\/[^/]+\/bin\/.+\.runfiles\/[^/]+\//, 2);
+      if (match) {
+        file = match[0] + fileParts[1];
+      }
+    }
+  }
   let routePath = vite.normalizePath(path__namespace.relative(pluginConfig.appDirectory, file));
   let route = Object.values(pluginConfig.routes).find(r => vite.normalizePath(r.file) === routePath);
   return route;
diff --git a/dist/vite/styles.js b/dist/vite/styles.js
index fae54295b133f2639ab676dbe675cecf1eefdb5e..1ed1b6c0d2f30d4d3b953b520f018515cd068b71 100644
--- a/dist/vite/styles.js
+++ b/dist/vite/styles.js
@@ -12,6 +12,7 @@
 
 Object.defineProperty(exports, '__esModule', { value: true });
 
+var fs = require('node:fs/promises');
 var path = require('node:path');
 var router = require('@remix-run/router');
 var resolveFileUrl = require('./resolve-file-url.js');
@@ -58,6 +59,23 @@ const getStylesForFiles = async ({
       let normalizedPath = path__namespace.resolve(rootDirectory, file).replace(/\\/g, "/");
       let node = await viteDevServer.moduleGraph.getModuleById(normalizedPath);
 
+      // NOTE(calebmer): If a module doesn't exist for `normalizedPath` that might be
+      // because `normalizedPath` is a symlink to another path. Resolve
+      // `normalizedPath` and try finding the module again.
+      if (!node) {
+        try {
+          let stats = await fs.lstat(normalizedPath);
+          if (stats.isSymbolicLink()) {
+            normalizedPath = await fs.readlink(normalizedPath);
+            node = await viteDevServer.moduleGraph.getModuleById(normalizedPath);
+          }
+        } catch (error) {
+          if (error.code !== "ENOENT") {
+            throw error;
+          }
+        }
+      }
+
       // If the module is only present in the client module graph, the module
       // won't have been found on the first request to the server. If so, we
       // request the module so it's in the module graph, then try again.
