diff --git a/dist/compiler.js b/dist/compiler.js
index c72e157fb3620e55187bce69afe8ab2a3ce8abc6..1339ab07b3ac023f8967bb87efe874847227eb38 100644
--- a/dist/compiler.js
+++ b/dist/compiler.js
@@ -310,7 +310,10 @@ async function createBrowserBuild(config, options) {
     entryPoints[id] = config.routes[id].file + "?browser";
   }
 
-  let plugins = [cssFilePlugin.cssFilePlugin(options), urlImportsPlugin.urlImportsPlugin(), mdx.mdxPlugin(config), browserRouteModulesPlugin.browserRouteModulesPlugin(config, /\?browser$/), emptyModulesPlugin.emptyModulesPlugin(config, /\.server(\.[jt]sx?)?$/), nodeModulesPolyfill.NodeModulesPolyfillPlugin(), esbuildPluginPnp.pnpPlugin()];
+  // NOTE(calebmer): Don't include files in a `/server/` directory in the
+  // browser build in addition to not including files with a `.server.js`
+  // extension.
+  let plugins = [cssFilePlugin.cssFilePlugin(options), urlImportsPlugin.urlImportsPlugin(), mdx.mdxPlugin(config), browserRouteModulesPlugin.browserRouteModulesPlugin(config, /\?browser$/), emptyModulesPlugin.emptyModulesPlugin(config, /(\.server|\/server\/.*)(\.[jt]sx?)?$/), nodeModulesPolyfill.NodeModulesPolyfillPlugin(), esbuildPluginPnp.pnpPlugin()];
   return esbuild__namespace.build({
     entryPoints,
     outdir: config.assetsBuildDirectory,
@@ -430,7 +433,10 @@ async function writeServerBuildResult(config, outputFiles) {
     if (file.path.endsWith(".js")) {
       // fix sourceMappingURL to be relative to current path instead of /build
       let filename = file.path.substring(file.path.lastIndexOf(path__namespace.sep) + 1);
-      let escapedFilename = filename.replace(/\./g, "\\.");
+      // NOTE(calebmer): When we build for Cloudflare pages the server
+      // JavaScript file is named `[[path]].js`. The `[` and `]` characters must
+      // be escaped within a regex.
+      let escapedFilename = filename.replace(/\./g, "\\.").replace(/\[/g, "\\[").replace(/\]/g, "\\]");
       let pattern = `(//# sourceMappingURL=)(.*)${escapedFilename}`;
       let contents = Buffer.from(file.contents).toString("utf-8");
       contents = contents.replace(new RegExp(pattern), `$1${filename}`);