diff --git a/dist/compiler/js/compiler.js b/dist/compiler/js/compiler.js
index 3520f44b09e799abb863df2fc3c3f0493d4ee9d9..dbe69ccb21ba013ef65f7515cc29f0929770fd92 100644
--- a/dist/compiler/js/compiler.js
+++ b/dist/compiler/js/compiler.js
@@ -31,6 +31,7 @@ var vanillaExtract = require('../plugins/vanillaExtract.js');
 var invariant = require('../../invariant.js');
 var hmr = require('./plugins/hmr.js');
 var analysis = require('../analysis.js');
+var fs = require('fs');
 
 function _interopNamespace(e) {
   if (e && e.__esModule) return e;
@@ -85,7 +86,10 @@ const createEsbuildConfig = (ctx, refs) => {
     hmr: ctx.options.mode === "development"
   }), cssImports.cssFilePlugin(ctx), absoluteCssUrlsPlugin.absoluteCssUrlsPlugin(), external.externalPlugin(/^https?:\/\//, {
     sideEffects: false
-  }), mdx.mdxPlugin(ctx), emptyModules.emptyModulesPlugin(ctx, /\.server(\.[jt]sx?)?$/), emptyModules.emptyModulesPlugin(ctx, /^@remix-run\/(deno|cloudflare|node)(\/.*)?$/, {
+  // NOTE(calebmer): Don't include files in a `/server/` directory in the
+  // browser build in addition to not including files with a `.server.js`
+  // extension.
+  }), mdx.mdxPlugin(ctx), emptyModules.emptyModulesPlugin(ctx, /(\.server|(^|\/)server\/.*)(\.[mc]?[jt]sx?)?$/), emptyModules.emptyModulesPlugin(ctx, /^@remix-run\/(deno|cloudflare|node)(\/.*)?$/, {
     includeNodeModules: true
   }), browserNodeBuiltinsPolyfill.browserNodeBuiltinsPolyfillPlugin(ctx)];
   if (ctx.options.mode === "development") {
diff --git a/dist/compiler/manifest.js b/dist/compiler/manifest.js
index f1d0c51ae5fcf1343b9e6d6b08fa3e6159fc7374..5cab38238fa1bde4e72e9025ff5703cdc697796d 100644
--- a/dist/compiler/manifest.js
+++ b/dist/compiler/manifest.js
@@ -60,7 +60,13 @@ async function create({
   for (let key of Object.keys(metafile.outputs).sort()) {
     let output = metafile.outputs[key];
     if (!output.entryPoint) continue;
-    if (path__namespace.resolve(output.entryPoint) === config.entryClientFilePath) {
+    // NOTE(calebmer): Bazel makes symlinks to files in runfiles directory
+    // so these two paths may be in different directories. Only look for whether
+    // the path after `/bazel-out/` is the same name as our entry client file.
+    // If the path is in a `.runfiles` directory we also want to ignore that.
+    const splitResolvedEntryPoint = path__namespace.resolve(output.entryPoint).split(/\/bazel-out\/[^/]+\/bin\/(?:.+\.runfiles\/[^/]+\/)?/);
+    const splitEntryClientFilePath = config.entryClientFilePath.split(/\/bazel-out\/[^/]+\/bin\/(?:.+\.runfiles\/[^/]+\/)?/);
+    if (splitResolvedEntryPoint.length === splitEntryClientFilePath.length && splitResolvedEntryPoint[splitResolvedEntryPoint.length - 1] === splitEntryClientFilePath[splitEntryClientFilePath.length - 1]) {
       entry = {
         module: resolveUrl(key),
         imports: resolveImports(output.imports)
diff --git a/dist/compiler/plugins/cssImports.js b/dist/compiler/plugins/cssImports.js
index 6e2815b13c8f1f9f9f5cf26742246e0acb778f18..6b033fde9c16893bab5a7f2ff6c576e7c996e477 100644
--- a/dist/compiler/plugins/cssImports.js
+++ b/dist/compiler/plugins/cssImports.js
@@ -52,7 +52,7 @@ function normalizePathSlashes(p) {
  * This plugin loads css files with the "css" loader (bundles and moves assets to assets directory)
  * and exports the url of the css file as its default export.
  */
-function cssFilePlugin(ctx) {
+function cssFilePlugin(ctx, publicPathOverride) {
   return {
     name: "css-file",
     async setup(build) {
@@ -107,7 +107,7 @@ function cssFilePlugin(ctx) {
             mainFields,
             nodePaths,
             platform,
-            publicPath,
+            publicPath: publicPathOverride || publicPath,
             sourceRoot,
             target,
             treeShaking,
diff --git a/dist/compiler/plugins/emptyModules.js b/dist/compiler/plugins/emptyModules.js
index bc282d8cd8b2330fdc3c8ba04529b7abd84a56ad..98b2434cc4014ce3cc6f35d5160b99de75327c3d 100644
--- a/dist/compiler/plugins/emptyModules.js
+++ b/dist/compiler/plugins/emptyModules.js
@@ -53,7 +53,11 @@ function emptyModulesPlugin({
         // Limit this behavior to modules found in only the `app` directory.
         // This allows node_modules to use the `.server.js` and `.client.js`
         // naming conventions with different semantics.
-        path__namespace.resolve(args.resolveDir, args.path).startsWith(config.appDirectory)) {
+        //
+        // NOTE(calebmer): Explicitly test for `node_modules` in the path. The
+        // bulk of our code lives outside of the `app` directory where it's
+        // important that empty modules transforms still apply.
+        !/(^|\/)node_modules(\/|$)/.test(path__namespace.resolve(args.resolveDir, args.path))) {
           return {
             path: args.path,
             namespace: "empty-module"
diff --git a/dist/compiler/server/compiler.js b/dist/compiler/server/compiler.js
index 8de93275fdc6b210a76bca1a59d98cacb15dd27d..e48e8be6636f4bfd2ebdf5755a6e650557339dbf 100644
--- a/dist/compiler/server/compiler.js
+++ b/dist/compiler/server/compiler.js
@@ -12,6 +12,7 @@
 
 Object.defineProperty(exports, '__esModule', { value: true });
 
+var path = require('node:path');
 var esbuild = require('esbuild');
 var loaders = require('../utils/loaders.js');
 var cssModuleImports = require('../plugins/cssModuleImports.js');
@@ -66,7 +67,7 @@ const createEsbuildConfig = (ctx, refs) => {
     outputCss: false
   }), vanillaExtract.vanillaExtractPlugin(ctx, {
     outputCss: false
-  }), cssSideEffectImports.cssSideEffectImportsPlugin(ctx), cssImports.cssFilePlugin(ctx), absoluteCssUrlsPlugin.absoluteCssUrlsPlugin(), external.externalPlugin(/^https?:\/\//, {
+  }), cssSideEffectImports.cssSideEffectImportsPlugin(ctx), cssImports.cssFilePlugin(ctx, ctx.config.publicPath), absoluteCssUrlsPlugin.absoluteCssUrlsPlugin(), external.externalPlugin(/^https?:\/\//, {
     sideEffects: false
   }), mdx.mdxPlugin(ctx), emptyModules.emptyModulesPlugin(ctx, /\.client(\.[jt]sx?)?$/), routes.serverRouteModulesPlugin(ctx), entry.serverEntryModulePlugin(ctx), manifest.serverAssetsManifestPlugin(refs), bareImports.serverBareModulesPlugin(ctx), external.externalPlugin(/^node:.*/, {
     sideEffects: false
@@ -78,7 +79,18 @@ const createEsbuildConfig = (ctx, refs) => {
     absWorkingDir: ctx.config.rootDirectory,
     stdin,
     entryPoints,
-    outfile: ctx.config.serverBuildPath,
+    // NOTE(calebmer): Enable code splitting for server builds. Otherwise we can't
+    // use `import()` to optimize server startup time to avoid importing expensive
+    // modules. esbuild unconditionally imports external modules (even if they're
+    // behind `import()`) when bundle splitting is off.
+    //
+    // When turning on splitting we also need to switch `outfile` for `outdir`. We
+    // also need to remove `publicPath` or else we'll get absolute import paths. We
+    // still need to pass `publicPath` to `cssImports.cssFilePlugin()` to get the
+    // right URL though.
+    splitting: true,
+    outdir: path.dirname(ctx.config.serverBuildPath),
+    chunkNames: "chunks/[name]-[hash]",
     conditions: ctx.config.serverConditions,
     platform: ctx.config.serverPlatform,
     format: ctx.config.serverModuleFormat,
@@ -93,7 +105,8 @@ const createEsbuildConfig = (ctx, refs) => {
     minifySyntax: true,
     minify: ctx.options.mode === "production" && ctx.config.serverMinify,
     mainFields: ctx.config.serverMainFields,
-    target: "node18",
+    // NOTE(calebmer): Target the version of Node.js we run in.
+    target: "node20.9.0",
     loader: loaders.loaders,
     bundle: true,
     logLevel: "silent",
@@ -106,7 +119,6 @@ const createEsbuildConfig = (ctx, refs) => {
     // The server build needs to know how to generate asset URLs for imports
     // of CSS and other files.
     assetNames: "_assets/[name]-[hash]",
-    publicPath: ctx.config.publicPath,
     define: {
       "process.env.NODE_ENV": JSON.stringify(ctx.options.mode),
       "process.env.REMIX_DEV_ORIGIN": JSON.stringify(ctx.options.REMIX_DEV_ORIGIN ?? "")
diff --git a/dist/compiler/server/write.js b/dist/compiler/server/write.js
index af5a2de82d11d76b6505de9f205e5f78eff7f9b3..c4633e7db48f70802a13fc8373ff57acecc67a56 100644
--- a/dist/compiler/server/write.js
+++ b/dist/compiler/server/write.js
@@ -44,7 +44,10 @@ async function write(config, outputFiles) {
     if ([".js", ".cjs", ".mjs"].some(ext => file.path.endsWith(ext))) {
       // fix sourceMappingURL to be relative to current path instead of /build
       let filename = file.path.substring(file.path.lastIndexOf(path__namespace.sep) + 1);
-      let escapedFilename = filename.replace(/([.[\]])/g, "\\$1");
+      // NOTE(calebmer): When we build for Cloudflare pages the server
+      // JavaScript file is named `[[path]].js`. The `[` and `]` characters must
+      // be escaped within a regex.
+      let escapedFilename = filename.replace(/\./g, "\\.").replace(/\[/g, "\\[").replace(/\]/g, "\\]");
       let pattern = `(//# sourceMappingURL=)(.*)${escapedFilename}`;
       let contents = Buffer.from(file.contents).toString("utf-8");
       contents = contents.replace(new RegExp(pattern), `$1${filename}`);
diff --git a/dist/config/flat-routes.js b/dist/config/flat-routes.js
index 70c183de6d9a19bd97e694a7c2e0f3ad67c0a0de..8bee451e7d26c013d2b4ae5854bd698c65fd1eb3 100644
--- a/dist/config/flat-routes.js
+++ b/dist/config/flat-routes.js
@@ -89,7 +89,12 @@ function flatRoutes(appDirectory, ignoredFilePatterns = [], prefix = "routes") {
     // If it's a directory, don't recurse into it, instead just look for a route module
     if (entry.isDirectory()) {
       route = findRouteModuleForFolder(appDirectory, filepath, ignoredFileRegex);
-    } else if (entry.isFile()) {
+    } else if (
+      entry.isFile() &&
+      // NOTE(calebmer): Ignore source maps which will be in the `routes` directory due to
+      // our Bazel build.
+      !entry.name.endsWith(".map")
+    ) {
       route = findRouteModuleForFile(appDirectory, filepath, ignoredFileRegex);
     }
     if (route) routes$1.push(route);
