diff --git a/dist/config/flat-routes.js b/dist/config/flat-routes.js
index 70c183de6d9a19bd97e694a7c2e0f3ad67c0a0de..8bee451e7d26c013d2b4ae5854bd698c65fd1eb3 100644
--- a/dist/config/flat-routes.js
+++ b/dist/config/flat-routes.js
@@ -89,7 +89,12 @@ function flatRoutes(appDirectory, ignoredFilePatterns = [], prefix = "routes") {
     // If it's a directory, don't recurse into it, instead just look for a route module
     if (entry.isDirectory()) {
       route = findRouteModuleForFolder(appDirectory, filepath, ignoredFileRegex);
-    } else if (entry.isFile()) {
+    } else if (
+      entry.isFile() &&
+      // NOTE(calebmer): Ignore source maps which will be in the `routes` directory due to
+      // our Bazel build.
+      !entry.name.endsWith(".map")
+    ) {
       route = findRouteModuleForFile(appDirectory, filepath, ignoredFileRegex);
     }
     if (route) routes$1.push(route);
diff --git a/dist/vite/plugin.js b/dist/vite/plugin.js
index bf608523a542d69398ed7afa51562c63a36ce8ed..dfeea3adaef336991412dbc17a5813f795aa2116 100644
--- a/dist/vite/plugin.js
+++ b/dist/vite/plugin.js
@@ -370,7 +370,8 @@ const remixVitePlugin = (remixUserConfig = {}) => {
     let buildDirectory = path__namespace.resolve(rootDirectory, resolvedRemixUserConfig.buildDirectory);
     let {
       serverBuildFile,
-      serverBundles
+      serverBundles,
+      additionalServerInputPath
     } = resolvedRemixUserConfig;
     let publicPath = viteUserConfig.base ?? "/";
     if (basename !== "/" && viteCommand === "serve" && !((_viteUserConfig$serve = viteUserConfig.server) !== null && _viteUserConfig$serve !== void 0 && _viteUserConfig$serve.middlewareMode) && !basename.startsWith(publicPath)) {
@@ -394,7 +395,8 @@ const remixVitePlugin = (remixUserConfig = {}) => {
       serverBuildFile,
       serverBundles,
       serverModuleFormat,
-      ssr
+      ssr,
+      additionalServerInputPath
     });
     for (let preset of remixUserConfig.presets ?? []) {
       var _preset$remixConfigRe;
@@ -691,9 +693,18 @@ const remixVitePlugin = (remixUserConfig = {}) => {
               rollupOptions: {
                 ...baseRollupOptions,
                 preserveEntrySignatures: "exports-only",
-                input: serverBuildId,
+                // NOTE(calebmer): We want to include an additional file in the Rollup server
+                // build. This additional file is our Node.js server. We want the server to
+                // share modules with the server code bundled by Rollup which is why it needs
+                // to be included in the Rollup build here.
+                input: ctx.remixConfig.additionalServerInputPath ? {
+                  [ctx.remixConfig.serverBuildFile]: serverBuildId,
+                  [path.basename(ctx.remixConfig.additionalServerInputPath)]: ctx.remixConfig.additionalServerInputPath,
+                } : {
+                  [ctx.remixConfig.serverBuildFile]: serverBuildId
+                },
                 output: {
-                  entryFileNames: ctx.remixConfig.serverBuildFile,
+                  entryFileNames: "[name]",
                   format: ctx.remixConfig.serverModuleFormat
                 }
               }
@@ -945,7 +956,11 @@ const remixVitePlugin = (remixUserConfig = {}) => {
       let resolved = await this.resolve(id, importer, options);
       if (!resolved) return;
       let serverFileRE = /\.server(\.[cm]?[jt]sx?)?$/;
-      let serverDirRE = /\/\.server\//;
+      // NOTE(calebmer): Don't include files in a `/server/` directory in the
+      // browser build in addition to not including files with a `.server.js`
+      // extension. Remix ignores files in a `/.server/` directory by default.
+      // We extend that to include `/server/` per our convention.
+      let serverDirRE = /\/\.?server\//;
       let isDotServer = serverFileRE.test(resolved.id) || serverDirRE.test(resolved.id);
       if (!isDotServer) return;
       if (!importer) return;
@@ -1043,29 +1058,14 @@ const remixVitePlugin = (remixUserConfig = {}) => {
       let isJSX = filepath.endsWith("x");
       let useFastRefresh = !ssr && (isJSX || code.includes(devRuntime));
       if (!useFastRefresh) return;
-      let result = await babel__default["default"].transformAsync(code, {
-        configFile: false,
-        babelrc: false,
-        filename: id,
-        sourceFileName: filepath,
-        parserOpts: {
-          sourceType: "module",
-          allowAwaitOutsideFunction: true
-        },
-        plugins: [[require("react-refresh/babel"), {
-          skipEnvCheck: true
-        }]],
-        sourceMaps: true
-      });
-      if (result === null) return;
-      code = result.code;
+      // NOTE(calebmer): Don't use Babel to perform `react-refresh` transform.
+      // The SWC transpiler run by Bazel performs the `react-refresh` transform.
       let refreshContentRE = /\$Refresh(?:Reg|Sig)\$\(/;
       if (refreshContentRE.test(code)) {
         code = addRefreshWrapper(ctx.remixConfig, code, id);
       }
       return {
         code,
-        map: result.map
       };
     }
   }, {
