diff --git a/index.d.ts b/index.d.ts
index 50f1817d786262b69d1ea72fba99c37ca8169ebb..84bffc3e3815e374fdb46ff04bfd4fe7eec30701 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -62,6 +62,8 @@ type VoidOrUndefinedOnly = void | { [UNDEFINED_VOID_ONLY]: never };
 export = React;
 export as namespace React;
 
+export declare const MEMO: unique symbol;
+
 declare namespace React {
     //
     // React Elements
@@ -879,6 +881,12 @@ declare namespace React {
         current: T;
     }
 
+    /**
+     * A value that has been memoized by `useMemo()` or `useCallback()` and so is safe to use as an
+     * effect dependency. `null & {}` is `never` so we leave null as-is.
+     */
+    type Memo<T> = T extends null | undefined ? T : T & {[MEMO]: never};
+
     // This will technically work if you give a Consumer<T> or Provider<T> but it's deprecated and warns
     /**
      * Accepts a context object (the value returned from `React.createContext`) and returns the current
@@ -894,7 +902,7 @@ declare namespace React {
      * @version 16.8.0
      * @see https://reactjs.org/docs/hooks-reference.html#usestate
      */
-    function useState<S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
+    function useState<S>(initialState: S | (() => S)): [S, Memo<Dispatch<SetStateAction<S>>>];
     // convenience overload when first argument is omitted
     /**
      * Returns a stateful value, and a function to update it.
@@ -902,7 +910,7 @@ declare namespace React {
      * @version 16.8.0
      * @see https://reactjs.org/docs/hooks-reference.html#usestate
      */
-    function useState<S = undefined>(): [S | undefined, Dispatch<SetStateAction<S | undefined>>];
+    function useState<S = undefined>(): [S | undefined, Memo<Dispatch<SetStateAction<S | undefined>>>];
     /**
      * An alternative to `useState`.
      *
@@ -918,7 +926,7 @@ declare namespace React {
         reducer: R,
         initializerArg: I,
         initializer: (arg: I) => ReducerStateWithoutAction<R>
-    ): [ReducerStateWithoutAction<R>, DispatchWithoutAction];
+    ): [ReducerStateWithoutAction<R>, Memo<DispatchWithoutAction>];
     /**
      * An alternative to `useState`.
      *
@@ -934,7 +942,7 @@ declare namespace React {
         reducer: R,
         initializerArg: ReducerStateWithoutAction<R>,
         initializer?: undefined
-    ): [ReducerStateWithoutAction<R>, DispatchWithoutAction];
+    ): [ReducerStateWithoutAction<R>, Memo<DispatchWithoutAction>];
     /**
      * An alternative to `useState`.
      *
@@ -952,7 +960,7 @@ declare namespace React {
         reducer: R,
         initializerArg: I & ReducerState<R>,
         initializer: (arg: I & ReducerState<R>) => ReducerState<R>
-    ): [ReducerState<R>, Dispatch<ReducerAction<R>>];
+    ): [ReducerState<R>, Memo<Dispatch<ReducerAction<R>>>];
     /**
      * An alternative to `useState`.
      *
@@ -968,7 +976,7 @@ declare namespace React {
         reducer: R,
         initializerArg: I,
         initializer: (arg: I) => ReducerState<R>
-    ): [ReducerState<R>, Dispatch<ReducerAction<R>>];
+    ): [ReducerState<R>, Memo<Dispatch<ReducerAction<R>>>];
     /**
      * An alternative to `useState`.
      *
@@ -993,7 +1001,7 @@ declare namespace React {
         reducer: R,
         initialState: ReducerState<R>,
         initializer?: undefined
-    ): [ReducerState<R>, Dispatch<ReducerAction<R>>];
+    ): [ReducerState<R>, Memo<Dispatch<ReducerAction<R>>>];
     /**
      * `useRef` returns a mutable ref object whose `.current` property is initialized to the passed argument
      * (`initialValue`). The returned object will persist for the full lifetime of the component.
@@ -1080,7 +1088,7 @@ declare namespace React {
     // A specific function type would not trigger implicit any.
     // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/52873#issuecomment-845806435 for a comparison between `Function` and more specific types.
     // tslint:disable-next-line ban-types
-    function useCallback<T extends Function>(callback: T, deps: DependencyList): T;
+    function useCallback<T extends Function>(callback: T, deps: DependencyList): Memo<T>;
     /**
      * `useMemo` will only recompute the memoized value when one of the `deps` has changed.
      *
@@ -1088,7 +1096,7 @@ declare namespace React {
      * @see https://reactjs.org/docs/hooks-reference.html#usememo
      */
     // allow undefined, but don't make it optional as that is very likely a mistake
-    function useMemo<T>(factory: () => T, deps: DependencyList | undefined): T;
+    function useMemo<T>(factory: () => T, deps: DependencyList | undefined): Memo<T>;
     /**
      * `useDebugValue` can be used to display a label for custom hooks in React DevTools.
      *
