diff --git a/index.d.ts b/index.d.ts
index 1fd8dbcc2d80518bb4f26d33d560fb73483e0def..d72cbbd23bcecd0b6e1c722e2b0c4e7c8acb13b4 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -7,7 +7,8 @@ declare namespace createRBTree {
     /** Represents a functional red-black tree. */
     interface Tree<K, V> {
         /** Returns the root node of the tree. */
-        root: Node<K, V>;
+        // NOTE(calebmer): Fix incorrect type in type definition.
+        root: Node<K, V> | null;
 
         /** A sorted array of all keys in the tree. */
         readonly keys: K[];
@@ -161,6 +162,10 @@ declare namespace createRBTree {
 
         /** If true, then the iterator is not at the beginning of the sequence. */
         readonly hasPrev: boolean;
+
+        // NOTE(calebmer): We have some advanced use for the internal
+        // iterator stack.
+        readonly _stack: ReadonlyArray<Node<K, V>>;
     }
 
     /** Represents a node in a red-black tree. */
@@ -172,10 +177,12 @@ declare namespace createRBTree {
         value: V;
 
         /** The left subtree of the node. */
-        left: Tree<K, V>;
+        // NOTE(calebmer): Fix incorrect type in type definition.
+        left: Node<K, V> | null;        
 
         /** The right subtree of the node. */
-        right: Tree<K, V>;
+        // NOTE(calebmer): Fix incorrect type in type definition.
+        right: Node<K, V> | null;
     }
 }
 