diff --git a/index.d.ts b/index.d.ts
index 1fd8dbcc2d80518bb4f26d33d560fb73483e0def..ed3192b3523a80a570563d87fe67306464d0e7c6 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -7,7 +7,11 @@ declare namespace createRBTree {
     /** Represents a functional red-black tree. */
     interface Tree<K, V> {
         /** Returns the root node of the tree. */
-        root: Node<K, V>;
+        // NOTE(calebmer): Fix incorrect type in type definition.
+        readonly root: Node<K, V> | null;
+
+        // NOTE(calebmer): Allow access to the comparison function.
+        readonly _compare: (key1: K, key2: K) => number;
 
         /** A sorted array of all keys in the tree. */
         readonly keys: K[];
@@ -107,7 +111,9 @@ declare namespace createRBTree {
          * @param key The key of the item to look up.
          * @returns The value of the first node associated with `key`.
          */
-        get: (key: K) => V | void;
+        // NOTE(calebmer): `void` has weird semantics. It's kinda like `unknown`.
+        // What's meant here is `undefined`.
+        get: (key: K) => V | undefined;
     }
 
     /** Iterates through the nodes in a red-black tree. */
@@ -161,6 +167,10 @@ declare namespace createRBTree {
 
         /** If true, then the iterator is not at the beginning of the sequence. */
         readonly hasPrev: boolean;
+
+        // NOTE(calebmer): We have some advanced use for the internal
+        // iterator stack.
+        readonly _stack: ReadonlyArray<Node<K, V>>;
     }
 
     /** Represents a node in a red-black tree. */
@@ -172,10 +182,16 @@ declare namespace createRBTree {
         value: V;
 
         /** The left subtree of the node. */
-        left: Tree<K, V>;
+        // NOTE(calebmer): Fix incorrect type in type definition.
+        left: Node<K, V> | null;        
 
         /** The right subtree of the node. */
-        right: Tree<K, V>;
+        // NOTE(calebmer): Fix incorrect type in type definition.
+        right: Node<K, V> | null;
+
+        /** The number of entries in this node and it's subtrees. */
+        // NOTE(calebmer): Expose private property for advanced use cases.
+        readonly _count: number;
     }
 }
 