diff --git a/dist/vanilla-extract-integration.cjs.dev.js b/dist/vanilla-extract-integration.cjs.dev.js
index fcedd61c17e6ea740b3f270a1bbae1c343b542e9..cdacf248a9eaf9aa4791c570ebc1985118cc9826 100644
--- a/dist/vanilla-extract-integration.cjs.dev.js
+++ b/dist/vanilla-extract-integration.cjs.dev.js
@@ -456,7 +456,6 @@ const vanillaExtractTransformPlugin = ({
 }) => ({
   name: 'vanilla-extract-filescope',
   setup(build) {
-    const packageInfo = getPackageInfo(build.initialOptions.absWorkingDir);
     build.onLoad({
       filter: cssFileFilter
     }, async ({
@@ -465,9 +464,15 @@ const vanillaExtractTransformPlugin = ({
       const originalSource = await fs.promises.readFile(path$1, 'utf-8');
       const source = await transform({
         source: originalSource,
-        filePath: path$1,
-        rootPath: build.initialOptions.absWorkingDir,
-        packageName: packageInfo.name,
+        // NOTE(calebmer): Replace platform specific
+        // `bazel-out/${platform}-${compilationMode}/bin` path with platform
+        // neutral `bazel-bin`. That way hashes are consistent across platforms
+        // and compilation modes.
+        filePath: path$1.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        rootPath: build.initialOptions.absWorkingDir.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        // NOTE(calebmer): Hardcode package name since it is annoying to make
+        // `package.json` available to Bazel hermetic builds.
+        packageName: "cyberworlds",
         identOption: identOption ?? (build.initialOptions.minify ? 'short' : 'debug')
       });
       return {
diff --git a/dist/vanilla-extract-integration.cjs.prod.js b/dist/vanilla-extract-integration.cjs.prod.js
index 61d28b2293909013fe9a8d3e0b89ddc180aced98..b4f43bad41e17ba55099eb3ab933ae854684d057 100644
--- a/dist/vanilla-extract-integration.cjs.prod.js
+++ b/dist/vanilla-extract-integration.cjs.prod.js
@@ -456,7 +456,6 @@ const vanillaExtractTransformPlugin = ({
 }) => ({
   name: 'vanilla-extract-filescope',
   setup(build) {
-    const packageInfo = getPackageInfo(build.initialOptions.absWorkingDir);
     build.onLoad({
       filter: cssFileFilter
     }, async ({
@@ -465,9 +464,15 @@ const vanillaExtractTransformPlugin = ({
       const originalSource = await fs.promises.readFile(path$1, 'utf-8');
       const source = await transform({
         source: originalSource,
-        filePath: path$1,
-        rootPath: build.initialOptions.absWorkingDir,
-        packageName: packageInfo.name,
+        // NOTE(calebmer): Replace platform specific
+        // `bazel-out/${platform}-${compilationMode}/bin` path with platform
+        // neutral `bazel-bin`. That way hashes are consistent across platforms
+        // and compilation modes.
+        filePath: path$1.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        rootPath: build.initialOptions.absWorkingDir.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        // NOTE(calebmer): Hardcode package name since it is annoying to make
+        // `package.json` available to Bazel hermetic builds.
+        packageName: "cyberworlds",
         identOption: identOption ?? (build.initialOptions.minify ? 'short' : 'debug')
       });
       return {
