diff --git a/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.dev.js b/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.dev.js
index a4217a87334b6110462b939bc15d02e855411af4..4a2b482bb55fe71302eb877ea3f4c8fccfa17310 100644
--- a/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.dev.js
+++ b/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.dev.js
@@ -81,7 +81,7 @@ function pitch() {
           fileName,
           source: serializedCss
         })}`;
-        const request = loaderUtils__default["default"].stringifyRequest(this, `${fileName}!=!${virtualResourceLoader}!${emptyCssExtractionFile}`);
+        const request = loaderUtils__default["default"].stringifyRequest(this, `${fileName}!=!${virtualResourceLoader}!${emptyCssExtractionFile}?as=${fileName}`);
         return `import ${request}`;
       }
     });
diff --git a/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.prod.js b/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.prod.js
index a4217a87334b6110462b939bc15d02e855411af4..4a2b482bb55fe71302eb877ea3f4c8fccfa17310 100644
--- a/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.prod.js
+++ b/loader/dist/vanilla-extract-webpack-plugin-loader.cjs.prod.js
@@ -81,7 +81,7 @@ function pitch() {
           fileName,
           source: serializedCss
         })}`;
-        const request = loaderUtils__default["default"].stringifyRequest(this, `${fileName}!=!${virtualResourceLoader}!${emptyCssExtractionFile}`);
+        const request = loaderUtils__default["default"].stringifyRequest(this, `${fileName}!=!${virtualResourceLoader}!${emptyCssExtractionFile}?as=${fileName}`);
         return `import ${request}`;
       }
     });
diff --git a/loader/dist/vanilla-extract-webpack-plugin-loader.esm.js b/loader/dist/vanilla-extract-webpack-plugin-loader.esm.js
index e34b484b8e2caaea14c3967cda7a1f18c06286bc..b26201d9509e71c910ad4594c06bb7bb87ee2f34 100644
--- a/loader/dist/vanilla-extract-webpack-plugin-loader.esm.js
+++ b/loader/dist/vanilla-extract-webpack-plugin-loader.esm.js
@@ -70,7 +70,7 @@ function pitch() {
           fileName,
           source: serializedCss
         })}`;
-        const request = loaderUtils.stringifyRequest(this, `${fileName}!=!${virtualResourceLoader}!${emptyCssExtractionFile}`);
+        const request = loaderUtils.stringifyRequest(this, `${fileName}!=!${virtualResourceLoader}!${emptyCssExtractionFile}?as=${fileName}`);
         return `import ${request}`;
       }
     });