diff --git a/dist/vanilla-extract-integration.cjs.dev.js b/dist/vanilla-extract-integration.cjs.dev.js
index 02fde7da688663a9a88dfe73bdd9f891c434e160..15e886ec7f735594d16fe82e1ea377bf17d6afca 100644
--- a/dist/vanilla-extract-integration.cjs.dev.js
+++ b/dist/vanilla-extract-integration.cjs.dev.js
@@ -310,7 +310,6 @@ const vanillaExtractFilescopePlugin = () => ({
   name: 'vanilla-extract-filescope',
 
   setup(build) {
-    const packageInfo = getPackageInfo(build.initialOptions.absWorkingDir);
     build.onLoad({
       filter: cssFileFilter
     }, async ({
@@ -319,9 +318,15 @@ const vanillaExtractFilescopePlugin = () => ({
       const originalSource = await fs.promises.readFile(path$1, 'utf-8');
       const source = addFileScope({
         source: originalSource,
-        filePath: path$1,
-        rootPath: build.initialOptions.absWorkingDir,
-        packageName: packageInfo.name
+        // NOTE(calebmer): Replace platform specific
+        // `bazel-out/${platform}-${compilationMode}/bin` path with platform
+        // neutral `bazel-bin`. That way hashes are consistent across platforms
+        // and compilation modes.
+        filePath: path$1.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        rootPath: build.initialOptions.absWorkingDir.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        // NOTE(calebmer): Hardcode package name since it is annoying to make
+        // `package.json` available to Bazel hermetic builds.
+        packageName: "cyberworlds",
       });
       return {
         contents: source,
diff --git a/dist/vanilla-extract-integration.cjs.prod.js b/dist/vanilla-extract-integration.cjs.prod.js
index 34629118701b35046d2d0ddad450a315061daef4..7508474a5604fa03c70ced3186bd58cfb30652eb 100644
--- a/dist/vanilla-extract-integration.cjs.prod.js
+++ b/dist/vanilla-extract-integration.cjs.prod.js
@@ -310,7 +310,6 @@ const vanillaExtractFilescopePlugin = () => ({
   name: 'vanilla-extract-filescope',
 
   setup(build) {
-    const packageInfo = getPackageInfo(build.initialOptions.absWorkingDir);
     build.onLoad({
       filter: cssFileFilter
     }, async ({
@@ -319,9 +318,15 @@ const vanillaExtractFilescopePlugin = () => ({
       const originalSource = await fs.promises.readFile(path$1, 'utf-8');
       const source = addFileScope({
         source: originalSource,
-        filePath: path$1,
-        rootPath: build.initialOptions.absWorkingDir,
-        packageName: packageInfo.name
+        // NOTE(calebmer): Replace platform specific
+        // `bazel-out/${platform}-${compilationMode}/bin` path with platform
+        // neutral `bazel-bin`. That way hashes are consistent across platforms
+        // and compilation modes.
+        filePath: path$1.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        rootPath: build.initialOptions.absWorkingDir.replace(/(^|\/)bazel-out\/[^/]+\/bin(\/|$)/, "$1bazel-bin$2"),
+        // NOTE(calebmer): Hardcode package name since it is annoying to make
+        // `package.json` available to Bazel hermetic builds.
+        packageName: "cyberworlds",
       });
       return {
         contents: source,