diff -ur a/swc/private/swc.bzl b/swc/private/swc.bzl
--- a/swc/private/swc.bzl	2022-11-03 19:19:53.000000000 -0400
+++ b/swc/private/swc.bzl	2022-11-03 19:17:45.000000000 -0400
@@ -122,6 +122,18 @@
             env = env,
             executable = ctx.executable.swc_cli,
             progress_message = "Transpiling with swc %{label}",
+            # NOTE(calebmer): We want SWC to correctly transform our imports using the `baseUrl`
+            # and `paths` configuration options. This requires the input file and output file to
+            # be in the same directory. `copy_file_to_bin_action()` is executed with
+            # `execution_requirements` including `"no-sandbox": "1"`. This means our input file
+            # is linked from `execroot` when we create a sandbox however we want to output in
+            # the sandbox. So execute SWC outside of a sandbox too so our input and output have
+            # the same base path.
+            #
+            # Not sandboxing means we have to trust SWC to be hermetic. This is a dangerous
+            # assumption. SWC could look at `package.json` or `.swcrc` files it is not supposed
+            # to. Generally SWC is a small, focused, tool so we trust it to behave hermetically.
+            execution_requirements = {"no-sandbox": "1"},
         )
     else:
         srcs = [_relative_to_package(src.path, ctx) for src in ctx.files.srcs]
@@ -170,6 +182,18 @@
                 executable = ctx.executable.swc_cli,
                 mnemonic = "SWCTranspile",
                 progress_message = "Transpiling with swc %{label} [swc %{input}]",
+                # NOTE(calebmer): We want SWC to correctly transform our imports using the `baseUrl`
+                # and `paths` configuration options. This requires the input file and output file to
+                # be in the same directory. `copy_file_to_bin_action()` is executed with
+                # `execution_requirements` including `"no-sandbox": "1"`. This means our input file
+                # is linked from `execroot` when we create a sandbox however we want to output in
+                # the sandbox. So execute SWC outside of a sandbox too so our input and output have
+                # the same base path.
+                #
+                # Not sandboxing means we have to trust SWC to be hermetic. This is a dangerous
+                # assumption. SWC could look at `package.json` or `.swcrc` files it is not supposed
+                # to. Generally SWC is a small, focused, tool so we trust it to behave hermetically.
+                execution_requirements = {"no-sandbox": "1"},
             )

     output_sources_depset = depset(output_sources)