diff --git a/esbuild-jest.d.ts b/esbuild-jest.d.ts
index 633c44def658bb3249888d7a78d4601227912ca4..4ec691b813a5268b5300ec5664fb1ef125332466 100644
--- a/esbuild-jest.d.ts
+++ b/esbuild-jest.d.ts
@@ -2,6 +2,8 @@ import { Transformer } from '@jest/transform';
 import { Loader } from 'esbuild';
 
 interface Options {
+    // NOTE(calebmer): Allow setting the `jsx` option.
+    jsx?: string;
     jsxFactory?: string;
     jsxFragment?: string;
     sourcemap?: boolean | 'inline' | 'external';
diff --git a/esbuild-jest.es.js b/esbuild-jest.es.js
index 140e050cb5aa1bbc1589cf1d5781ae373486fc8a..77412d0238c7167c5f3167b778670995dc131379 100644
--- a/esbuild-jest.es.js
+++ b/esbuild-jest.es.js
@@ -49,6 +49,11 @@ const createTransformer = (options)=>({
                 loader,
                 format: (options === null || options === void 0 ? void 0 : options.format) || 'cjs',
                 target: (options === null || options === void 0 ? void 0 : options.target) || 'es2018',
+                // NOTE(calebmer): Allow setting the `jsx` option.
+                ...(options === null || options === void 0 ? void 0 : options.jsx) ? {
+                    jsx: options.jsx
+                } : {
+                },
                 ...(options === null || options === void 0 ? void 0 : options.jsxFactory) ? {
                     jsxFactory: options.jsxFactory
                 } : {
diff --git a/esbuild-jest.js b/esbuild-jest.js
index 82ddd865ebb2b889e523b88172183ebf940669f1..ca490d966800fef7b76b4c451236e84dbd9900ad 100644
--- a/esbuild-jest.js
+++ b/esbuild-jest.js
@@ -55,6 +55,11 @@ const createTransformer = (options)=>({
                 loader,
                 format: (options === null || options === void 0 ? void 0 : options.format) || 'cjs',
                 target: (options === null || options === void 0 ? void 0 : options.target) || 'es2018',
+                // NOTE(calebmer): Allow setting the `jsx` option.
+                ...(options === null || options === void 0 ? void 0 : options.jsx) ? {
+                    jsx: options.jsx
+                } : {
+                },
                 ...(options === null || options === void 0 ? void 0 : options.jsxFactory) ? {
                     jsxFactory: options.jsxFactory
                 } : {