diff --git a/cjs/models.d.ts b/cjs/models.d.ts
index cd0ad6147e8b1d9d0b2a09244e597309360e0e26..4ba6f9d4342d9d0581a3573338a6ad35f1ef1447 100644
--- a/cjs/models.d.ts
+++ b/cjs/models.d.ts
@@ -25,8 +25,12 @@ declare const gpt_3_5_0301_spec: {
     readonly supported_features: ["fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_3_5_0301_spec as 'gpt-3.5-0301' };
-export { gpt_3_5_0301_spec as 'gpt-3.5' };
+// NOTE(calebmer): Our version of TypeScript doesn't support `export` with
+// string syntax. Instead of upgrading TypeScript, we're patching the module
+// which uses this feature for now.
+//
+// export { gpt_3_5_0301_spec as 'gpt-3.5-0301' };
+// export { gpt_3_5_0301_spec as 'gpt-3.5' };
 declare const gpt_3_5_turbo_0613_spec: {
     readonly name: "gpt-3.5-turbo-0613";
     readonly slug: "gpt-3-5-turbo-0613";
@@ -53,7 +57,7 @@ declare const gpt_3_5_turbo_0613_spec: {
     readonly supported_features: ["fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_3_5_turbo_0613_spec as 'gpt-3.5-turbo-0613' };
+// export { gpt_3_5_turbo_0613_spec as 'gpt-3.5-turbo-0613' };
 declare const gpt_4_1106_preview_spec: {
     readonly name: "gpt-4-1106-preview";
     readonly slug: "gpt-4-1106-preview";
@@ -80,7 +84,7 @@ declare const gpt_4_1106_preview_spec: {
     readonly knowledge_cutoff: Date;
     readonly reasoning_tokens: false;
 };
-export { gpt_4_1106_preview_spec as 'gpt-4-1106-preview' };
+// export { gpt_4_1106_preview_spec as 'gpt-4-1106-preview' };
 declare const gpt_4_32k_spec: {
     readonly name: "gpt-4-32k";
     readonly slug: "gpt-4-32k";
@@ -107,7 +111,7 @@ declare const gpt_4_32k_spec: {
     readonly supported_features: ["fine_tuning", "streaming"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_32k_spec as 'gpt-4-32k' };
+// export { gpt_4_32k_spec as 'gpt-4-32k' };
 declare const text_ada_001_spec: {
     readonly name: "text-ada-001";
     readonly slug: "text-ada-001";
@@ -128,7 +132,7 @@ declare const text_ada_001_spec: {
         };
     };
 };
-export { text_ada_001_spec as 'text-ada-001' };
+// export { text_ada_001_spec as 'text-ada-001' };
 declare const text_babbage_001_spec: {
     readonly name: "text-babbage-001";
     readonly slug: "text-babbage-001";
@@ -149,7 +153,7 @@ declare const text_babbage_001_spec: {
         };
     };
 };
-export { text_babbage_001_spec as 'text-babbage-001' };
+// export { text_babbage_001_spec as 'text-babbage-001' };
 declare const text_curie_001_spec: {
     readonly name: "text-curie-001";
     readonly slug: "text-curie-001";
@@ -170,7 +174,7 @@ declare const text_curie_001_spec: {
         };
     };
 };
-export { text_curie_001_spec as 'text-curie-001' };
+// export { text_curie_001_spec as 'text-curie-001' };
 declare const text_davinci_001_spec: {
     readonly name: "text-davinci-001";
     readonly slug: "text-davinci-001";
@@ -191,7 +195,7 @@ declare const text_davinci_001_spec: {
         };
     };
 };
-export { text_davinci_001_spec as 'text-davinci-001' };
+// export { text_davinci_001_spec as 'text-davinci-001' };
 declare const text_davinci_002_spec: {
     readonly name: "text-davinci-002";
     readonly slug: "text-davinci-002";
@@ -212,7 +216,7 @@ declare const text_davinci_002_spec: {
         };
     };
 };
-export { text_davinci_002_spec as 'text-davinci-002' };
+// export { text_davinci_002_spec as 'text-davinci-002' };
 declare const text_davinci_003_spec: {
     readonly name: "text-davinci-003";
     readonly slug: "text-davinci-003";
@@ -233,7 +237,7 @@ declare const text_davinci_003_spec: {
         };
     };
 };
-export { text_davinci_003_spec as 'text-davinci-003' };
+// export { text_davinci_003_spec as 'text-davinci-003' };
 declare const ada_spec: {
     readonly name: "ada";
     readonly slug: "ada";
@@ -254,7 +258,7 @@ declare const ada_spec: {
         };
     };
 };
-export { ada_spec as 'ada' };
+// export { ada_spec as 'ada' };
 declare const babbage_spec: {
     readonly name: "babbage";
     readonly slug: "babbage";
@@ -275,7 +279,7 @@ declare const babbage_spec: {
         };
     };
 };
-export { babbage_spec as 'babbage' };
+// export { babbage_spec as 'babbage' };
 declare const curie_spec: {
     readonly name: "curie";
     readonly slug: "curie";
@@ -296,7 +300,7 @@ declare const curie_spec: {
         };
     };
 };
-export { curie_spec as 'curie' };
+// export { curie_spec as 'curie' };
 declare const davinci_spec: {
     readonly name: "davinci";
     readonly slug: "davinci";
@@ -317,7 +321,7 @@ declare const davinci_spec: {
         };
     };
 };
-export { davinci_spec as 'davinci' };
+// export { davinci_spec as 'davinci' };
 declare const code_davinci_001_spec: {
     readonly name: "code-davinci-001";
     readonly slug: "code-davinci-001";
@@ -338,7 +342,7 @@ declare const code_davinci_001_spec: {
         };
     };
 };
-export { code_davinci_001_spec as 'code-davinci-001' };
+// export { code_davinci_001_spec as 'code-davinci-001' };
 declare const code_davinci_002_spec: {
     readonly name: "code-davinci-002";
     readonly slug: "code-davinci-002";
@@ -359,7 +363,7 @@ declare const code_davinci_002_spec: {
         };
     };
 };
-export { code_davinci_002_spec as 'code-davinci-002' };
+// export { code_davinci_002_spec as 'code-davinci-002' };
 declare const davinci_codex_spec: {
     readonly name: "davinci-codex";
     readonly slug: "davinci-codex";
@@ -380,7 +384,7 @@ declare const davinci_codex_spec: {
         };
     };
 };
-export { davinci_codex_spec as 'davinci-codex' };
+// export { davinci_codex_spec as 'davinci-codex' };
 declare const code_davinci_edit_001_spec: {
     readonly name: "code-davinci-edit-001";
     readonly slug: "code-davinci-edit-001";
@@ -401,7 +405,7 @@ declare const code_davinci_edit_001_spec: {
         };
     };
 };
-export { code_davinci_edit_001_spec as 'code-davinci-edit-001' };
+// export { code_davinci_edit_001_spec as 'code-davinci-edit-001' };
 declare const code_cushman_001_spec: {
     readonly name: "code-cushman-001";
     readonly slug: "code-cushman-001";
@@ -422,7 +426,7 @@ declare const code_cushman_001_spec: {
         };
     };
 };
-export { code_cushman_001_spec as 'code-cushman-001' };
+// export { code_cushman_001_spec as 'code-cushman-001' };
 declare const code_cushman_002_spec: {
     readonly name: "code-cushman-002";
     readonly slug: "code-cushman-002";
@@ -443,7 +447,7 @@ declare const code_cushman_002_spec: {
         };
     };
 };
-export { code_cushman_002_spec as 'code-cushman-002' };
+// export { code_cushman_002_spec as 'code-cushman-002' };
 declare const cushman_codex_spec: {
     readonly name: "cushman-codex";
     readonly slug: "cushman-codex";
@@ -464,7 +468,7 @@ declare const cushman_codex_spec: {
         };
     };
 };
-export { cushman_codex_spec as 'cushman-codex' };
+// export { cushman_codex_spec as 'cushman-codex' };
 declare const code_search_ada_code_001_spec: {
     readonly name: "code-search-ada-code-001";
     readonly slug: "code-search-ada-code-001";
@@ -484,7 +488,7 @@ declare const code_search_ada_code_001_spec: {
         };
     };
 };
-export { code_search_ada_code_001_spec as 'code-search-ada-code-001' };
+// export { code_search_ada_code_001_spec as 'code-search-ada-code-001' };
 declare const code_search_ada_text_001_spec: {
     readonly name: "code-search-ada-text-001";
     readonly slug: "code-search-ada-text-001";
@@ -504,7 +508,7 @@ declare const code_search_ada_text_001_spec: {
         };
     };
 };
-export { code_search_ada_text_001_spec as 'code-search-ada-text-001' };
+// export { code_search_ada_text_001_spec as 'code-search-ada-text-001' };
 declare const text_davinci_edit_001_spec: {
     readonly name: "text-davinci-edit-001";
     readonly slug: "text-davinci-edit-001";
@@ -525,7 +529,7 @@ declare const text_davinci_edit_001_spec: {
         };
     };
 };
-export { text_davinci_edit_001_spec as 'text-davinci-edit-001' };
+// export { text_davinci_edit_001_spec as 'text-davinci-edit-001' };
 declare const text_similarity_ada_001_spec: {
     readonly name: "text-similarity-ada-001";
     readonly slug: "text-similarity-ada-001";
@@ -545,7 +549,7 @@ declare const text_similarity_ada_001_spec: {
         };
     };
 };
-export { text_similarity_ada_001_spec as 'text-similarity-ada-001' };
+// export { text_similarity_ada_001_spec as 'text-similarity-ada-001' };
 declare const text_search_ada_doc_001_spec: {
     readonly name: "text-search-ada-doc-001";
     readonly slug: "text-search-ada-doc-001";
@@ -565,7 +569,7 @@ declare const text_search_ada_doc_001_spec: {
         };
     };
 };
-export { text_search_ada_doc_001_spec as 'text-search-ada-doc-001' };
+// export { text_search_ada_doc_001_spec as 'text-search-ada-doc-001' };
 declare const text_search_ada_query_001_spec: {
     readonly name: "text-search-ada-query-001";
     readonly slug: "text-search-ada-query-001";
@@ -585,7 +589,7 @@ declare const text_search_ada_query_001_spec: {
         };
     };
 };
-export { text_search_ada_query_001_spec as 'text-search-ada-query-001' };
+// export { text_search_ada_query_001_spec as 'text-search-ada-query-001' };
 declare const text_similarity_babbage_001_spec: {
     readonly name: "text-similarity-babbage-001";
     readonly slug: "text-similarity-babbage-001";
@@ -605,7 +609,7 @@ declare const text_similarity_babbage_001_spec: {
         };
     };
 };
-export { text_similarity_babbage_001_spec as 'text-similarity-babbage-001' };
+// export { text_similarity_babbage_001_spec as 'text-similarity-babbage-001' };
 declare const text_search_babbage_doc_001_spec: {
     readonly name: "text-search-babbage-doc-001";
     readonly slug: "text-search-babbage-doc-001";
@@ -625,7 +629,7 @@ declare const text_search_babbage_doc_001_spec: {
         };
     };
 };
-export { text_search_babbage_doc_001_spec as 'text-search-babbage-doc-001' };
+// export { text_search_babbage_doc_001_spec as 'text-search-babbage-doc-001' };
 declare const text_search_babbage_query_001_spec: {
     readonly name: "text-search-babbage-query-001";
     readonly slug: "text-search-babbage-query-001";
@@ -645,7 +649,7 @@ declare const text_search_babbage_query_001_spec: {
         };
     };
 };
-export { text_search_babbage_query_001_spec as 'text-search-babbage-query-001' };
+// export { text_search_babbage_query_001_spec as 'text-search-babbage-query-001' };
 declare const code_search_babbage_code_001_spec: {
     readonly name: "code-search-babbage-code-001";
     readonly slug: "code-search-babbage-code-001";
@@ -665,7 +669,7 @@ declare const code_search_babbage_code_001_spec: {
         };
     };
 };
-export { code_search_babbage_code_001_spec as 'code-search-babbage-code-001' };
+// export { code_search_babbage_code_001_spec as 'code-search-babbage-code-001' };
 declare const code_search_babbage_text_001_spec: {
     readonly name: "code-search-babbage-text-001";
     readonly slug: "code-search-babbage-text-001";
@@ -685,7 +689,7 @@ declare const code_search_babbage_text_001_spec: {
         };
     };
 };
-export { code_search_babbage_text_001_spec as 'code-search-babbage-text-001' };
+// export { code_search_babbage_text_001_spec as 'code-search-babbage-text-001' };
 declare const text_similarity_curie_001_spec: {
     readonly name: "text-similarity-curie-001";
     readonly slug: "text-similarity-curie-001";
@@ -705,7 +709,7 @@ declare const text_similarity_curie_001_spec: {
         };
     };
 };
-export { text_similarity_curie_001_spec as 'text-similarity-curie-001' };
+// export { text_similarity_curie_001_spec as 'text-similarity-curie-001' };
 declare const text_search_curie_doc_001_spec: {
     readonly name: "text-search-curie-doc-001";
     readonly slug: "text-search-curie-doc-001";
@@ -725,7 +729,7 @@ declare const text_search_curie_doc_001_spec: {
         };
     };
 };
-export { text_search_curie_doc_001_spec as 'text-search-curie-doc-001' };
+// export { text_search_curie_doc_001_spec as 'text-search-curie-doc-001' };
 declare const text_search_curie_query_001_spec: {
     readonly name: "text-search-curie-query-001";
     readonly slug: "text-search-curie-query-001";
@@ -745,7 +749,7 @@ declare const text_search_curie_query_001_spec: {
         };
     };
 };
-export { text_search_curie_query_001_spec as 'text-search-curie-query-001' };
+// export { text_search_curie_query_001_spec as 'text-search-curie-query-001' };
 declare const text_similarity_davinci_001_spec: {
     readonly name: "text-similarity-davinci-001";
     readonly slug: "text-similarity-davinci-001";
@@ -765,7 +769,7 @@ declare const text_similarity_davinci_001_spec: {
         };
     };
 };
-export { text_similarity_davinci_001_spec as 'text-similarity-davinci-001' };
+// export { text_similarity_davinci_001_spec as 'text-similarity-davinci-001' };
 declare const text_search_davinci_doc_001_spec: {
     readonly name: "text-search-davinci-doc-001";
     readonly slug: "text-search-davinci-doc-001";
@@ -785,7 +789,7 @@ declare const text_search_davinci_doc_001_spec: {
         };
     };
 };
-export { text_search_davinci_doc_001_spec as 'text-search-davinci-doc-001' };
+// export { text_search_davinci_doc_001_spec as 'text-search-davinci-doc-001' };
 declare const text_search_davinci_query_001_spec: {
     readonly name: "text-search-davinci-query-001";
     readonly slug: "text-search-davinci-query-001";
@@ -805,4 +809,4 @@ declare const text_search_davinci_query_001_spec: {
         };
     };
 };
-export { text_search_davinci_query_001_spec as 'text-search-davinci-query-001' };
+// export { text_search_davinci_query_001_spec as 'text-search-davinci-query-001' };
diff --git a/cjs/models.gen.d.ts b/cjs/models.gen.d.ts
index 310ef99e41b742b935b1493bc8de434ba958fe78..63475fafd5770fc2c3642439a818105c79adc008 100644
--- a/cjs/models.gen.d.ts
+++ b/cjs/models.gen.d.ts
@@ -23,7 +23,11 @@ declare const babbage_002_spec: {
         };
     };
 };
-export { babbage_002_spec as 'babbage-002' };
+// NOTE(calebmer): Our version of TypeScript doesn't support `export` with
+// string syntax. Instead of upgrading TypeScript, we're patching the module
+// which uses this feature for now.
+//
+// export { babbage_002_spec as 'babbage-002' };
 declare const chatgpt_4o_latest_spec: {
     readonly name: "chatgpt-4o-latest";
     readonly slug: "chatgpt-4o-latest";
@@ -46,7 +50,7 @@ declare const chatgpt_4o_latest_spec: {
         };
     };
 };
-export { chatgpt_4o_latest_spec as 'chatgpt-4o-latest' };
+// export { chatgpt_4o_latest_spec as 'chatgpt-4o-latest' };
 declare const codex_mini_latest_spec: {
     readonly name: "codex-mini-latest";
     readonly slug: "codex-mini-latest";
@@ -63,7 +67,7 @@ declare const codex_mini_latest_spec: {
     readonly supported_endpoints: ["responses"];
     readonly reasoning_tokens: true;
 };
-export { codex_mini_latest_spec as 'codex-mini-latest' };
+// export { codex_mini_latest_spec as 'codex-mini-latest' };
 declare const computer_use_preview_2025_03_11_spec: {
     readonly name: "computer-use-preview-2025-03-11";
     readonly slug: "computer-use-preview-2025-03-11";
@@ -90,8 +94,8 @@ declare const computer_use_preview_2025_03_11_spec: {
         };
     };
 };
-export { computer_use_preview_2025_03_11_spec as 'computer-use-preview-2025-03-11' };
-export { computer_use_preview_2025_03_11_spec as 'computer-use-preview' };
+// export { computer_use_preview_2025_03_11_spec as 'computer-use-preview-2025-03-11' };
+// export { computer_use_preview_2025_03_11_spec as 'computer-use-preview' };
 declare const dall_e_2_spec: {
     readonly name: "dall-e-2";
     readonly slug: "dall-e-2";
@@ -105,7 +109,7 @@ declare const dall_e_2_spec: {
     readonly supported_features: ["inpainting"];
     readonly reasoning_tokens: false;
 };
-export { dall_e_2_spec as 'dall-e-2' };
+// export { dall_e_2_spec as 'dall-e-2' };
 declare const dall_e_3_spec: {
     readonly name: "dall-e-3";
     readonly slug: "dall-e-3";
@@ -128,7 +132,7 @@ declare const dall_e_3_spec: {
         };
     };
 };
-export { dall_e_3_spec as 'dall-e-3' };
+// export { dall_e_3_spec as 'dall-e-3' };
 declare const davinci_002_spec: {
     readonly name: "davinci-002";
     readonly slug: "davinci-002";
@@ -154,7 +158,7 @@ declare const davinci_002_spec: {
         };
     };
 };
-export { davinci_002_spec as 'davinci-002' };
+// export { davinci_002_spec as 'davinci-002' };
 declare const gpt_3_5_turbo_16k_0613_spec: {
     readonly name: "gpt-3.5-turbo-16k-0613";
     readonly slug: "gpt-3-5-turbo-16k-0613";
@@ -181,7 +185,7 @@ declare const gpt_3_5_turbo_16k_0613_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_16k_0613_spec as 'gpt-3.5-turbo-16k-0613' };
+// export { gpt_3_5_turbo_16k_0613_spec as 'gpt-3.5-turbo-16k-0613' };
 declare const gpt_3_5_turbo_instruct_spec: {
     readonly name: "gpt-3.5-turbo-instruct";
     readonly slug: "gpt-3-5-turbo-instruct";
@@ -204,7 +208,7 @@ declare const gpt_3_5_turbo_instruct_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_instruct_spec as 'gpt-3.5-turbo-instruct' };
+// export { gpt_3_5_turbo_instruct_spec as 'gpt-3.5-turbo-instruct' };
 declare const gpt_3_5_turbo_0125_spec: {
     readonly name: "gpt-3.5-turbo-0125";
     readonly slug: "gpt-3-5-turbo-0125";
@@ -231,7 +235,7 @@ declare const gpt_3_5_turbo_0125_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo-0125' };
+// export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo-0125' };
 declare const gpt_3_5_turbo_1106_spec: {
     readonly name: "gpt-3.5-turbo-1106";
     readonly slug: "gpt-3-5-turbo-1106";
@@ -258,8 +262,8 @@ declare const gpt_3_5_turbo_1106_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_1106_spec as 'gpt-3.5-turbo-1106' };
-export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo' };
+// export { gpt_3_5_turbo_1106_spec as 'gpt-3.5-turbo-1106' };
+// export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo' };
 declare const gpt_4_0125_preview_spec: {
     readonly name: "gpt-4-0125-preview";
     readonly slug: "gpt-4-0125-preview";
@@ -286,7 +290,7 @@ declare const gpt_4_0125_preview_spec: {
         };
     };
 };
-export { gpt_4_0125_preview_spec as 'gpt-4-0125-preview' };
+// export { gpt_4_0125_preview_spec as 'gpt-4-0125-preview' };
 declare const gpt_4_1106_vision_preview_spec: {
     readonly name: "gpt-4-1106-vision-preview";
     readonly slug: "gpt-4-1106-vision-preview";
@@ -314,8 +318,8 @@ declare const gpt_4_1106_vision_preview_spec: {
         };
     };
 };
-export { gpt_4_1106_vision_preview_spec as 'gpt-4-1106-vision-preview' };
-export { gpt_4_0125_preview_spec as 'gpt-4-turbo-preview' };
+// export { gpt_4_1106_vision_preview_spec as 'gpt-4-1106-vision-preview' };
+// export { gpt_4_0125_preview_spec as 'gpt-4-turbo-preview' };
 declare const gpt_4_turbo_2024_04_09_spec: {
     readonly name: "gpt-4-turbo-2024-04-09";
     readonly slug: "gpt-4-turbo-2024-04-09";
@@ -342,8 +346,8 @@ declare const gpt_4_turbo_2024_04_09_spec: {
         };
     };
 };
-export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo-2024-04-09' };
-export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo' };
+// export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo-2024-04-09' };
+// export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo' };
 declare const gpt_4_1_mini_2025_04_14_spec: {
     readonly name: "gpt-4.1-mini-2025-04-14";
     readonly slug: "gpt-4.1-mini-2025-04-14";
@@ -360,8 +364,8 @@ declare const gpt_4_1_mini_2025_04_14_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch", "fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini-2025-04-14' };
-export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini' };
+// export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini-2025-04-14' };
+// export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini' };
 declare const gpt_4_1_nano_2025_04_14_spec: {
     readonly name: "gpt-4.1-nano-2025-04-14";
     readonly slug: "gpt-4.1-nano-2025-04-14";
@@ -378,8 +382,8 @@ declare const gpt_4_1_nano_2025_04_14_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch", "fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano-2025-04-14' };
-export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano' };
+// export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano-2025-04-14' };
+// export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano' };
 declare const gpt_4_1_2025_04_14_spec: {
     readonly name: "gpt-4.1-2025-04-14";
     readonly slug: "gpt-4.1-2025-04-14";
@@ -406,8 +410,8 @@ declare const gpt_4_1_2025_04_14_spec: {
         };
     };
 };
-export { gpt_4_1_2025_04_14_spec as 'gpt-4.1-2025-04-14' };
-export { gpt_4_1_2025_04_14_spec as 'gpt-4.1' };
+// export { gpt_4_1_2025_04_14_spec as 'gpt-4.1-2025-04-14' };
+// export { gpt_4_1_2025_04_14_spec as 'gpt-4.1' };
 declare const gpt_4_5_preview_2025_02_27_spec: {
     readonly name: "gpt-4.5-preview-2025-02-27";
     readonly slug: "gpt-4.5-preview-2025-02-27";
@@ -424,8 +428,8 @@ declare const gpt_4_5_preview_2025_02_27_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview-2025-02-27' };
-export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview' };
+// export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview-2025-02-27' };
+// export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview' };
 declare const gpt_4_0613_spec: {
     readonly name: "gpt-4-0613";
     readonly slug: "gpt-4-0613";
@@ -452,7 +456,7 @@ declare const gpt_4_0613_spec: {
         };
     };
 };
-export { gpt_4_0613_spec as 'gpt-4-0613' };
+// export { gpt_4_0613_spec as 'gpt-4-0613' };
 declare const gpt_4_0314_spec: {
     readonly name: "gpt-4-0314";
     readonly slug: "gpt-4-0314";
@@ -479,8 +483,8 @@ declare const gpt_4_0314_spec: {
         };
     };
 };
-export { gpt_4_0314_spec as 'gpt-4-0314' };
-export { gpt_4_0613_spec as 'gpt-4' };
+// export { gpt_4_0314_spec as 'gpt-4-0314' };
+// export { gpt_4_0613_spec as 'gpt-4' };
 declare const gpt_4o_audio_preview_2025_06_03_spec: {
     readonly name: "gpt-4o-audio-preview-2025-06-03";
     readonly slug: "gpt-4o-audio-preview-2025-06-03";
@@ -507,7 +511,7 @@ declare const gpt_4o_audio_preview_2025_06_03_spec: {
         };
     };
 };
-export { gpt_4o_audio_preview_2025_06_03_spec as 'gpt-4o-audio-preview-2025-06-03' };
+// export { gpt_4o_audio_preview_2025_06_03_spec as 'gpt-4o-audio-preview-2025-06-03' };
 declare const gpt_4o_audio_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-audio-preview-2024-12-17";
     readonly slug: "gpt-4o-audio-preview-2024-12-17";
@@ -534,7 +538,7 @@ declare const gpt_4o_audio_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview-2024-12-17' };
+// export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview-2024-12-17' };
 declare const gpt_4o_audio_preview_2024_10_01_spec: {
     readonly name: "gpt-4o-audio-preview-2024-10-01";
     readonly slug: "gpt-4o-audio-preview-2024-10-01";
@@ -561,8 +565,8 @@ declare const gpt_4o_audio_preview_2024_10_01_spec: {
         };
     };
 };
-export { gpt_4o_audio_preview_2024_10_01_spec as 'gpt-4o-audio-preview-2024-10-01' };
-export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview' };
+// export { gpt_4o_audio_preview_2024_10_01_spec as 'gpt-4o-audio-preview-2024-10-01' };
+// export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview' };
 declare const gpt_4o_mini_audio_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-mini-audio-preview-2024-12-17";
     readonly slug: "gpt-4o-mini-audio-preview-2024-12-17";
@@ -589,8 +593,8 @@ declare const gpt_4o_mini_audio_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview-2024-12-17' };
-export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview' };
+// export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview-2024-12-17' };
+// export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview' };
 declare const gpt_4o_mini_realtime_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-mini-realtime-preview-2024-12-17";
     readonly slug: "gpt-4o-mini-realtime-preview-2024-12-17";
@@ -617,8 +621,8 @@ declare const gpt_4o_mini_realtime_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview-2024-12-17' };
-export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview' };
+// export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview-2024-12-17' };
+// export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview' };
 declare const gpt_4o_mini_search_preview_2025_03_11_spec: {
     readonly name: "gpt-4o-mini-search-preview-2025-03-11";
     readonly slug: "gpt-4o-mini-search-preview-2025-03-11";
@@ -635,8 +639,8 @@ declare const gpt_4o_mini_search_preview_2025_03_11_spec: {
     readonly supported_endpoints: ["chat_completions"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview-2025-03-11' };
-export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview' };
+// export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview-2025-03-11' };
+// export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview' };
 declare const gpt_4o_mini_transcribe_spec: {
     readonly name: "gpt-4o-mini-transcribe";
     readonly slug: "gpt-4o-mini-transcribe";
@@ -652,7 +656,7 @@ declare const gpt_4o_mini_transcribe_spec: {
     readonly supported_endpoints: ["transcription", "realtime"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_transcribe_spec as 'gpt-4o-mini-transcribe' };
+// export { gpt_4o_mini_transcribe_spec as 'gpt-4o-mini-transcribe' };
 declare const gpt_4o_mini_tts_spec: {
     readonly name: "gpt-4o-mini-tts";
     readonly slug: "gpt-4o-mini-tts";
@@ -666,7 +670,7 @@ declare const gpt_4o_mini_tts_spec: {
     readonly supported_endpoints: ["speech_generation"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_tts_spec as 'gpt-4o-mini-tts' };
+// export { gpt_4o_mini_tts_spec as 'gpt-4o-mini-tts' };
 declare const gpt_4o_mini_2024_07_18_spec: {
     readonly name: "gpt-4o-mini-2024-07-18";
     readonly slug: "gpt-4o-mini-2024-07-18";
@@ -683,8 +687,8 @@ declare const gpt_4o_mini_2024_07_18_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch", "fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini-2024-07-18' };
-export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini' };
+// export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini-2024-07-18' };
+// export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini' };
 declare const gpt_4o_realtime_preview_2025_06_03_spec: {
     readonly name: "gpt-4o-realtime-preview-2025-06-03";
     readonly slug: "gpt-4o-realtime-preview-2025-06-03";
@@ -711,7 +715,7 @@ declare const gpt_4o_realtime_preview_2025_06_03_spec: {
         };
     };
 };
-export { gpt_4o_realtime_preview_2025_06_03_spec as 'gpt-4o-realtime-preview-2025-06-03' };
+// export { gpt_4o_realtime_preview_2025_06_03_spec as 'gpt-4o-realtime-preview-2025-06-03' };
 declare const gpt_4o_realtime_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-realtime-preview-2024-12-17";
     readonly slug: "gpt-4o-realtime-preview-2024-12-17";
@@ -738,7 +742,7 @@ declare const gpt_4o_realtime_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview-2024-12-17' };
+// export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview-2024-12-17' };
 declare const gpt_4o_realtime_preview_2024_10_01_spec: {
     readonly name: "gpt-4o-realtime-preview-2024-10-01";
     readonly slug: "gpt-4o-realtime-preview-2024-10-01";
@@ -765,8 +769,8 @@ declare const gpt_4o_realtime_preview_2024_10_01_spec: {
         };
     };
 };
-export { gpt_4o_realtime_preview_2024_10_01_spec as 'gpt-4o-realtime-preview-2024-10-01' };
-export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview' };
+// export { gpt_4o_realtime_preview_2024_10_01_spec as 'gpt-4o-realtime-preview-2024-10-01' };
+// export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview' };
 declare const gpt_4o_search_preview_2025_03_11_spec: {
     readonly name: "gpt-4o-search-preview-2025-03-11";
     readonly slug: "gpt-4o-search-preview-2025-03-11";
@@ -793,8 +797,8 @@ declare const gpt_4o_search_preview_2025_03_11_spec: {
         };
     };
 };
-export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview-2025-03-11' };
-export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview' };
+// export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview-2025-03-11' };
+// export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview' };
 declare const gpt_4o_transcribe_spec: {
     readonly name: "gpt-4o-transcribe";
     readonly slug: "gpt-4o-transcribe";
@@ -810,7 +814,7 @@ declare const gpt_4o_transcribe_spec: {
     readonly supported_endpoints: ["transcription", "realtime"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_transcribe_spec as 'gpt-4o-transcribe' };
+// export { gpt_4o_transcribe_spec as 'gpt-4o-transcribe' };
 declare const gpt_4o_2024_11_20_spec: {
     readonly name: "gpt-4o-2024-11-20";
     readonly slug: "gpt-4o-2024-11-20";
@@ -837,7 +841,7 @@ declare const gpt_4o_2024_11_20_spec: {
         };
     };
 };
-export { gpt_4o_2024_11_20_spec as 'gpt-4o-2024-11-20' };
+// export { gpt_4o_2024_11_20_spec as 'gpt-4o-2024-11-20' };
 declare const gpt_4o_2024_08_06_spec: {
     readonly name: "gpt-4o-2024-08-06";
     readonly slug: "gpt-4o-2024-08-06";
@@ -864,7 +868,7 @@ declare const gpt_4o_2024_08_06_spec: {
         };
     };
 };
-export { gpt_4o_2024_08_06_spec as 'gpt-4o-2024-08-06' };
+// export { gpt_4o_2024_08_06_spec as 'gpt-4o-2024-08-06' };
 declare const gpt_4o_2024_05_13_spec: {
     readonly name: "gpt-4o-2024-05-13";
     readonly slug: "gpt-4o-2024-05-13";
@@ -881,8 +885,8 @@ declare const gpt_4o_2024_05_13_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_2024_05_13_spec as 'gpt-4o-2024-05-13' };
-export { gpt_4o_2024_08_06_spec as 'gpt-4o' };
+// export { gpt_4o_2024_05_13_spec as 'gpt-4o-2024-05-13' };
+// export { gpt_4o_2024_08_06_spec as 'gpt-4o' };
 declare const gpt_image_1_spec: {
     readonly name: "gpt-image-1";
     readonly slug: "gpt-image-1";
@@ -896,7 +900,7 @@ declare const gpt_image_1_spec: {
     readonly supported_features: ["inpainting"];
     readonly reasoning_tokens: false;
 };
-export { gpt_image_1_spec as 'gpt-image-1' };
+// export { gpt_image_1_spec as 'gpt-image-1' };
 declare const o1_mini_2024_09_12_spec: {
     readonly name: "o1-mini-2024-09-12";
     readonly slug: "o1-mini-2024-09-12";
@@ -923,8 +927,8 @@ declare const o1_mini_2024_09_12_spec: {
         };
     };
 };
-export { o1_mini_2024_09_12_spec as 'o1-mini-2024-09-12' };
-export { o1_mini_2024_09_12_spec as 'o1-mini' };
+// export { o1_mini_2024_09_12_spec as 'o1-mini-2024-09-12' };
+// export { o1_mini_2024_09_12_spec as 'o1-mini' };
 declare const o1_preview_2024_09_12_spec: {
     readonly name: "o1-preview-2024-09-12";
     readonly slug: "o1-preview-2024-09-12";
@@ -951,8 +955,8 @@ declare const o1_preview_2024_09_12_spec: {
         };
     };
 };
-export { o1_preview_2024_09_12_spec as 'o1-preview-2024-09-12' };
-export { o1_preview_2024_09_12_spec as 'o1-preview' };
+// export { o1_preview_2024_09_12_spec as 'o1-preview-2024-09-12' };
+// export { o1_preview_2024_09_12_spec as 'o1-preview' };
 declare const o1_pro_2025_03_19_spec: {
     readonly name: "o1-pro-2025-03-19";
     readonly slug: "o1-pro-2025-03-19";
@@ -979,8 +983,8 @@ declare const o1_pro_2025_03_19_spec: {
         };
     };
 };
-export { o1_pro_2025_03_19_spec as 'o1-pro-2025-03-19' };
-export { o1_pro_2025_03_19_spec as 'o1-pro' };
+// export { o1_pro_2025_03_19_spec as 'o1-pro-2025-03-19' };
+// export { o1_pro_2025_03_19_spec as 'o1-pro' };
 declare const o1_2024_12_17_spec: {
     readonly name: "o1-2024-12-17";
     readonly slug: "o1-2024-12-17";
@@ -1008,8 +1012,8 @@ declare const o1_2024_12_17_spec: {
         };
     };
 };
-export { o1_2024_12_17_spec as 'o1-2024-12-17' };
-export { o1_2024_12_17_spec as 'o1' };
+// export { o1_2024_12_17_spec as 'o1-2024-12-17' };
+// export { o1_2024_12_17_spec as 'o1' };
 declare const o3_mini_2025_01_31_spec: {
     readonly name: "o3-mini-2025-01-31";
     readonly slug: "o3-mini-2025-01-31";
@@ -1026,8 +1030,8 @@ declare const o3_mini_2025_01_31_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch"];
     readonly reasoning_tokens: true;
 };
-export { o3_mini_2025_01_31_spec as 'o3-mini-2025-01-31' };
-export { o3_mini_2025_01_31_spec as 'o3-mini' };
+// export { o3_mini_2025_01_31_spec as 'o3-mini-2025-01-31' };
+// export { o3_mini_2025_01_31_spec as 'o3-mini' };
 declare const o3_pro_2025_06_10_spec: {
     readonly name: "o3-pro-2025-06-10";
     readonly slug: "o3-pro-2025-06-10";
@@ -1054,8 +1058,8 @@ declare const o3_pro_2025_06_10_spec: {
         };
     };
 };
-export { o3_pro_2025_06_10_spec as 'o3-pro-2025-06-10' };
-export { o3_pro_2025_06_10_spec as 'o3-pro' };
+// export { o3_pro_2025_06_10_spec as 'o3-pro-2025-06-10' };
+// export { o3_pro_2025_06_10_spec as 'o3-pro' };
 declare const o3_2025_04_16_spec: {
     readonly name: "o3-2025-04-16";
     readonly slug: "o3-2025-04-16";
@@ -1083,8 +1087,8 @@ declare const o3_2025_04_16_spec: {
         };
     };
 };
-export { o3_2025_04_16_spec as 'o3-2025-04-16' };
-export { o3_2025_04_16_spec as 'o3' };
+// export { o3_2025_04_16_spec as 'o3-2025-04-16' };
+// export { o3_2025_04_16_spec as 'o3' };
 declare const o4_mini_2025_04_16_spec: {
     readonly name: "o4-mini-2025-04-16";
     readonly slug: "o4-mini-2025-04-16";
@@ -1101,8 +1105,8 @@ declare const o4_mini_2025_04_16_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "batch", "fine_tuning"];
     readonly reasoning_tokens: true;
 };
-export { o4_mini_2025_04_16_spec as 'o4-mini-2025-04-16' };
-export { o4_mini_2025_04_16_spec as 'o4-mini' };
+// export { o4_mini_2025_04_16_spec as 'o4-mini-2025-04-16' };
+// export { o4_mini_2025_04_16_spec as 'o4-mini' };
 declare const omni_moderation_2024_09_26_spec: {
     readonly name: "omni-moderation-2024-09-26";
     readonly slug: "omni-moderation-2024-09-26";
@@ -1116,8 +1120,8 @@ declare const omni_moderation_2024_09_26_spec: {
     readonly reasoning_tokens: false;
     readonly supported_features: ["image_input"];
 };
-export { omni_moderation_2024_09_26_spec as 'omni-moderation-2024-09-26' };
-export { omni_moderation_2024_09_26_spec as 'omni-moderation-latest' };
+// export { omni_moderation_2024_09_26_spec as 'omni-moderation-2024-09-26' };
+// export { omni_moderation_2024_09_26_spec as 'omni-moderation-latest' };
 declare const text_embedding_3_large_spec: {
     readonly name: "text-embedding-3-large";
     readonly slug: "text-embedding-3-large";
@@ -1129,7 +1133,7 @@ declare const text_embedding_3_large_spec: {
     };
     readonly supported_endpoints: ["embeddings", "batch"];
 };
-export { text_embedding_3_large_spec as 'text-embedding-3-large' };
+// export { text_embedding_3_large_spec as 'text-embedding-3-large' };
 declare const text_embedding_3_small_spec: {
     readonly name: "text-embedding-3-small";
     readonly slug: "text-embedding-3-small";
@@ -1141,7 +1145,7 @@ declare const text_embedding_3_small_spec: {
     };
     readonly supported_endpoints: ["embeddings", "batch"];
 };
-export { text_embedding_3_small_spec as 'text-embedding-3-small' };
+// export { text_embedding_3_small_spec as 'text-embedding-3-small' };
 declare const text_embedding_ada_002_spec: {
     readonly name: "text-embedding-ada-002";
     readonly slug: "text-embedding-ada-002";
@@ -1153,7 +1157,7 @@ declare const text_embedding_ada_002_spec: {
     };
     readonly supported_endpoints: ["embeddings", "batch"];
 };
-export { text_embedding_ada_002_spec as 'text-embedding-ada-002' };
+// export { text_embedding_ada_002_spec as 'text-embedding-ada-002' };
 declare const text_moderation_007_spec: {
     readonly name: "text-moderation-007";
     readonly slug: "text-moderation-007";
@@ -1168,9 +1172,9 @@ declare const text_moderation_007_spec: {
     readonly supported_endpoints: ["moderation"];
     readonly reasoning_tokens: false;
 };
-export { text_moderation_007_spec as 'text-moderation-007' };
-export { text_moderation_007_spec as 'text-moderation-latest' };
-export { text_moderation_007_spec as 'text-moderation-stable' };
+// export { text_moderation_007_spec as 'text-moderation-007' };
+// export { text_moderation_007_spec as 'text-moderation-latest' };
+// export { text_moderation_007_spec as 'text-moderation-stable' };
 declare const tts_1_hd_spec: {
     readonly name: "tts-1-hd";
     readonly slug: "tts-1-hd";
@@ -1184,7 +1188,7 @@ declare const tts_1_hd_spec: {
     readonly supported_endpoints: ["speech_generation"];
     readonly reasoning_tokens: false;
 };
-export { tts_1_hd_spec as 'tts-1-hd' };
+// export { tts_1_hd_spec as 'tts-1-hd' };
 declare const tts_1_spec: {
     readonly name: "tts-1";
     readonly slug: "tts-1";
@@ -1198,7 +1202,7 @@ declare const tts_1_spec: {
     readonly supported_endpoints: ["speech_generation"];
     readonly reasoning_tokens: false;
 };
-export { tts_1_spec as 'tts-1' };
+// export { tts_1_spec as 'tts-1' };
 declare const whisper_1_spec: {
     readonly name: "whisper-1";
     readonly slug: "whisper-1";
@@ -1211,4 +1215,4 @@ declare const whisper_1_spec: {
     readonly supported_endpoints: ["transcription", "translation"];
     readonly reasoning_tokens: false;
 };
-export { whisper_1_spec as 'whisper-1' };
+// export { whisper_1_spec as 'whisper-1' };
diff --git a/esm/models.d.ts b/esm/models.d.ts
index cd0ad6147e8b1d9d0b2a09244e597309360e0e26..4ba6f9d4342d9d0581a3573338a6ad35f1ef1447 100644
--- a/esm/models.d.ts
+++ b/esm/models.d.ts
@@ -25,8 +25,12 @@ declare const gpt_3_5_0301_spec: {
     readonly supported_features: ["fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_3_5_0301_spec as 'gpt-3.5-0301' };
-export { gpt_3_5_0301_spec as 'gpt-3.5' };
+// NOTE(calebmer): Our version of TypeScript doesn't support `export` with
+// string syntax. Instead of upgrading TypeScript, we're patching the module
+// which uses this feature for now.
+//
+// export { gpt_3_5_0301_spec as 'gpt-3.5-0301' };
+// export { gpt_3_5_0301_spec as 'gpt-3.5' };
 declare const gpt_3_5_turbo_0613_spec: {
     readonly name: "gpt-3.5-turbo-0613";
     readonly slug: "gpt-3-5-turbo-0613";
@@ -53,7 +57,7 @@ declare const gpt_3_5_turbo_0613_spec: {
     readonly supported_features: ["fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_3_5_turbo_0613_spec as 'gpt-3.5-turbo-0613' };
+// export { gpt_3_5_turbo_0613_spec as 'gpt-3.5-turbo-0613' };
 declare const gpt_4_1106_preview_spec: {
     readonly name: "gpt-4-1106-preview";
     readonly slug: "gpt-4-1106-preview";
@@ -80,7 +84,7 @@ declare const gpt_4_1106_preview_spec: {
     readonly knowledge_cutoff: Date;
     readonly reasoning_tokens: false;
 };
-export { gpt_4_1106_preview_spec as 'gpt-4-1106-preview' };
+// export { gpt_4_1106_preview_spec as 'gpt-4-1106-preview' };
 declare const gpt_4_32k_spec: {
     readonly name: "gpt-4-32k";
     readonly slug: "gpt-4-32k";
@@ -107,7 +111,7 @@ declare const gpt_4_32k_spec: {
     readonly supported_features: ["fine_tuning", "streaming"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_32k_spec as 'gpt-4-32k' };
+// export { gpt_4_32k_spec as 'gpt-4-32k' };
 declare const text_ada_001_spec: {
     readonly name: "text-ada-001";
     readonly slug: "text-ada-001";
@@ -128,7 +132,7 @@ declare const text_ada_001_spec: {
         };
     };
 };
-export { text_ada_001_spec as 'text-ada-001' };
+// export { text_ada_001_spec as 'text-ada-001' };
 declare const text_babbage_001_spec: {
     readonly name: "text-babbage-001";
     readonly slug: "text-babbage-001";
@@ -149,7 +153,7 @@ declare const text_babbage_001_spec: {
         };
     };
 };
-export { text_babbage_001_spec as 'text-babbage-001' };
+// export { text_babbage_001_spec as 'text-babbage-001' };
 declare const text_curie_001_spec: {
     readonly name: "text-curie-001";
     readonly slug: "text-curie-001";
@@ -170,7 +174,7 @@ declare const text_curie_001_spec: {
         };
     };
 };
-export { text_curie_001_spec as 'text-curie-001' };
+// export { text_curie_001_spec as 'text-curie-001' };
 declare const text_davinci_001_spec: {
     readonly name: "text-davinci-001";
     readonly slug: "text-davinci-001";
@@ -191,7 +195,7 @@ declare const text_davinci_001_spec: {
         };
     };
 };
-export { text_davinci_001_spec as 'text-davinci-001' };
+// export { text_davinci_001_spec as 'text-davinci-001' };
 declare const text_davinci_002_spec: {
     readonly name: "text-davinci-002";
     readonly slug: "text-davinci-002";
@@ -212,7 +216,7 @@ declare const text_davinci_002_spec: {
         };
     };
 };
-export { text_davinci_002_spec as 'text-davinci-002' };
+// export { text_davinci_002_spec as 'text-davinci-002' };
 declare const text_davinci_003_spec: {
     readonly name: "text-davinci-003";
     readonly slug: "text-davinci-003";
@@ -233,7 +237,7 @@ declare const text_davinci_003_spec: {
         };
     };
 };
-export { text_davinci_003_spec as 'text-davinci-003' };
+// export { text_davinci_003_spec as 'text-davinci-003' };
 declare const ada_spec: {
     readonly name: "ada";
     readonly slug: "ada";
@@ -254,7 +258,7 @@ declare const ada_spec: {
         };
     };
 };
-export { ada_spec as 'ada' };
+// export { ada_spec as 'ada' };
 declare const babbage_spec: {
     readonly name: "babbage";
     readonly slug: "babbage";
@@ -275,7 +279,7 @@ declare const babbage_spec: {
         };
     };
 };
-export { babbage_spec as 'babbage' };
+// export { babbage_spec as 'babbage' };
 declare const curie_spec: {
     readonly name: "curie";
     readonly slug: "curie";
@@ -296,7 +300,7 @@ declare const curie_spec: {
         };
     };
 };
-export { curie_spec as 'curie' };
+// export { curie_spec as 'curie' };
 declare const davinci_spec: {
     readonly name: "davinci";
     readonly slug: "davinci";
@@ -317,7 +321,7 @@ declare const davinci_spec: {
         };
     };
 };
-export { davinci_spec as 'davinci' };
+// export { davinci_spec as 'davinci' };
 declare const code_davinci_001_spec: {
     readonly name: "code-davinci-001";
     readonly slug: "code-davinci-001";
@@ -338,7 +342,7 @@ declare const code_davinci_001_spec: {
         };
     };
 };
-export { code_davinci_001_spec as 'code-davinci-001' };
+// export { code_davinci_001_spec as 'code-davinci-001' };
 declare const code_davinci_002_spec: {
     readonly name: "code-davinci-002";
     readonly slug: "code-davinci-002";
@@ -359,7 +363,7 @@ declare const code_davinci_002_spec: {
         };
     };
 };
-export { code_davinci_002_spec as 'code-davinci-002' };
+// export { code_davinci_002_spec as 'code-davinci-002' };
 declare const davinci_codex_spec: {
     readonly name: "davinci-codex";
     readonly slug: "davinci-codex";
@@ -380,7 +384,7 @@ declare const davinci_codex_spec: {
         };
     };
 };
-export { davinci_codex_spec as 'davinci-codex' };
+// export { davinci_codex_spec as 'davinci-codex' };
 declare const code_davinci_edit_001_spec: {
     readonly name: "code-davinci-edit-001";
     readonly slug: "code-davinci-edit-001";
@@ -401,7 +405,7 @@ declare const code_davinci_edit_001_spec: {
         };
     };
 };
-export { code_davinci_edit_001_spec as 'code-davinci-edit-001' };
+// export { code_davinci_edit_001_spec as 'code-davinci-edit-001' };
 declare const code_cushman_001_spec: {
     readonly name: "code-cushman-001";
     readonly slug: "code-cushman-001";
@@ -422,7 +426,7 @@ declare const code_cushman_001_spec: {
         };
     };
 };
-export { code_cushman_001_spec as 'code-cushman-001' };
+// export { code_cushman_001_spec as 'code-cushman-001' };
 declare const code_cushman_002_spec: {
     readonly name: "code-cushman-002";
     readonly slug: "code-cushman-002";
@@ -443,7 +447,7 @@ declare const code_cushman_002_spec: {
         };
     };
 };
-export { code_cushman_002_spec as 'code-cushman-002' };
+// export { code_cushman_002_spec as 'code-cushman-002' };
 declare const cushman_codex_spec: {
     readonly name: "cushman-codex";
     readonly slug: "cushman-codex";
@@ -464,7 +468,7 @@ declare const cushman_codex_spec: {
         };
     };
 };
-export { cushman_codex_spec as 'cushman-codex' };
+// export { cushman_codex_spec as 'cushman-codex' };
 declare const code_search_ada_code_001_spec: {
     readonly name: "code-search-ada-code-001";
     readonly slug: "code-search-ada-code-001";
@@ -484,7 +488,7 @@ declare const code_search_ada_code_001_spec: {
         };
     };
 };
-export { code_search_ada_code_001_spec as 'code-search-ada-code-001' };
+// export { code_search_ada_code_001_spec as 'code-search-ada-code-001' };
 declare const code_search_ada_text_001_spec: {
     readonly name: "code-search-ada-text-001";
     readonly slug: "code-search-ada-text-001";
@@ -504,7 +508,7 @@ declare const code_search_ada_text_001_spec: {
         };
     };
 };
-export { code_search_ada_text_001_spec as 'code-search-ada-text-001' };
+// export { code_search_ada_text_001_spec as 'code-search-ada-text-001' };
 declare const text_davinci_edit_001_spec: {
     readonly name: "text-davinci-edit-001";
     readonly slug: "text-davinci-edit-001";
@@ -525,7 +529,7 @@ declare const text_davinci_edit_001_spec: {
         };
     };
 };
-export { text_davinci_edit_001_spec as 'text-davinci-edit-001' };
+// export { text_davinci_edit_001_spec as 'text-davinci-edit-001' };
 declare const text_similarity_ada_001_spec: {
     readonly name: "text-similarity-ada-001";
     readonly slug: "text-similarity-ada-001";
@@ -545,7 +549,7 @@ declare const text_similarity_ada_001_spec: {
         };
     };
 };
-export { text_similarity_ada_001_spec as 'text-similarity-ada-001' };
+// export { text_similarity_ada_001_spec as 'text-similarity-ada-001' };
 declare const text_search_ada_doc_001_spec: {
     readonly name: "text-search-ada-doc-001";
     readonly slug: "text-search-ada-doc-001";
@@ -565,7 +569,7 @@ declare const text_search_ada_doc_001_spec: {
         };
     };
 };
-export { text_search_ada_doc_001_spec as 'text-search-ada-doc-001' };
+// export { text_search_ada_doc_001_spec as 'text-search-ada-doc-001' };
 declare const text_search_ada_query_001_spec: {
     readonly name: "text-search-ada-query-001";
     readonly slug: "text-search-ada-query-001";
@@ -585,7 +589,7 @@ declare const text_search_ada_query_001_spec: {
         };
     };
 };
-export { text_search_ada_query_001_spec as 'text-search-ada-query-001' };
+// export { text_search_ada_query_001_spec as 'text-search-ada-query-001' };
 declare const text_similarity_babbage_001_spec: {
     readonly name: "text-similarity-babbage-001";
     readonly slug: "text-similarity-babbage-001";
@@ -605,7 +609,7 @@ declare const text_similarity_babbage_001_spec: {
         };
     };
 };
-export { text_similarity_babbage_001_spec as 'text-similarity-babbage-001' };
+// export { text_similarity_babbage_001_spec as 'text-similarity-babbage-001' };
 declare const text_search_babbage_doc_001_spec: {
     readonly name: "text-search-babbage-doc-001";
     readonly slug: "text-search-babbage-doc-001";
@@ -625,7 +629,7 @@ declare const text_search_babbage_doc_001_spec: {
         };
     };
 };
-export { text_search_babbage_doc_001_spec as 'text-search-babbage-doc-001' };
+// export { text_search_babbage_doc_001_spec as 'text-search-babbage-doc-001' };
 declare const text_search_babbage_query_001_spec: {
     readonly name: "text-search-babbage-query-001";
     readonly slug: "text-search-babbage-query-001";
@@ -645,7 +649,7 @@ declare const text_search_babbage_query_001_spec: {
         };
     };
 };
-export { text_search_babbage_query_001_spec as 'text-search-babbage-query-001' };
+// export { text_search_babbage_query_001_spec as 'text-search-babbage-query-001' };
 declare const code_search_babbage_code_001_spec: {
     readonly name: "code-search-babbage-code-001";
     readonly slug: "code-search-babbage-code-001";
@@ -665,7 +669,7 @@ declare const code_search_babbage_code_001_spec: {
         };
     };
 };
-export { code_search_babbage_code_001_spec as 'code-search-babbage-code-001' };
+// export { code_search_babbage_code_001_spec as 'code-search-babbage-code-001' };
 declare const code_search_babbage_text_001_spec: {
     readonly name: "code-search-babbage-text-001";
     readonly slug: "code-search-babbage-text-001";
@@ -685,7 +689,7 @@ declare const code_search_babbage_text_001_spec: {
         };
     };
 };
-export { code_search_babbage_text_001_spec as 'code-search-babbage-text-001' };
+// export { code_search_babbage_text_001_spec as 'code-search-babbage-text-001' };
 declare const text_similarity_curie_001_spec: {
     readonly name: "text-similarity-curie-001";
     readonly slug: "text-similarity-curie-001";
@@ -705,7 +709,7 @@ declare const text_similarity_curie_001_spec: {
         };
     };
 };
-export { text_similarity_curie_001_spec as 'text-similarity-curie-001' };
+// export { text_similarity_curie_001_spec as 'text-similarity-curie-001' };
 declare const text_search_curie_doc_001_spec: {
     readonly name: "text-search-curie-doc-001";
     readonly slug: "text-search-curie-doc-001";
@@ -725,7 +729,7 @@ declare const text_search_curie_doc_001_spec: {
         };
     };
 };
-export { text_search_curie_doc_001_spec as 'text-search-curie-doc-001' };
+// export { text_search_curie_doc_001_spec as 'text-search-curie-doc-001' };
 declare const text_search_curie_query_001_spec: {
     readonly name: "text-search-curie-query-001";
     readonly slug: "text-search-curie-query-001";
@@ -745,7 +749,7 @@ declare const text_search_curie_query_001_spec: {
         };
     };
 };
-export { text_search_curie_query_001_spec as 'text-search-curie-query-001' };
+// export { text_search_curie_query_001_spec as 'text-search-curie-query-001' };
 declare const text_similarity_davinci_001_spec: {
     readonly name: "text-similarity-davinci-001";
     readonly slug: "text-similarity-davinci-001";
@@ -765,7 +769,7 @@ declare const text_similarity_davinci_001_spec: {
         };
     };
 };
-export { text_similarity_davinci_001_spec as 'text-similarity-davinci-001' };
+// export { text_similarity_davinci_001_spec as 'text-similarity-davinci-001' };
 declare const text_search_davinci_doc_001_spec: {
     readonly name: "text-search-davinci-doc-001";
     readonly slug: "text-search-davinci-doc-001";
@@ -785,7 +789,7 @@ declare const text_search_davinci_doc_001_spec: {
         };
     };
 };
-export { text_search_davinci_doc_001_spec as 'text-search-davinci-doc-001' };
+// export { text_search_davinci_doc_001_spec as 'text-search-davinci-doc-001' };
 declare const text_search_davinci_query_001_spec: {
     readonly name: "text-search-davinci-query-001";
     readonly slug: "text-search-davinci-query-001";
@@ -805,4 +809,4 @@ declare const text_search_davinci_query_001_spec: {
         };
     };
 };
-export { text_search_davinci_query_001_spec as 'text-search-davinci-query-001' };
+// export { text_search_davinci_query_001_spec as 'text-search-davinci-query-001' };
diff --git a/esm/models.gen.d.ts b/esm/models.gen.d.ts
index 310ef99e41b742b935b1493bc8de434ba958fe78..63475fafd5770fc2c3642439a818105c79adc008 100644
--- a/esm/models.gen.d.ts
+++ b/esm/models.gen.d.ts
@@ -23,7 +23,11 @@ declare const babbage_002_spec: {
         };
     };
 };
-export { babbage_002_spec as 'babbage-002' };
+// NOTE(calebmer): Our version of TypeScript doesn't support `export` with
+// string syntax. Instead of upgrading TypeScript, we're patching the module
+// which uses this feature for now.
+//
+// export { babbage_002_spec as 'babbage-002' };
 declare const chatgpt_4o_latest_spec: {
     readonly name: "chatgpt-4o-latest";
     readonly slug: "chatgpt-4o-latest";
@@ -46,7 +50,7 @@ declare const chatgpt_4o_latest_spec: {
         };
     };
 };
-export { chatgpt_4o_latest_spec as 'chatgpt-4o-latest' };
+// export { chatgpt_4o_latest_spec as 'chatgpt-4o-latest' };
 declare const codex_mini_latest_spec: {
     readonly name: "codex-mini-latest";
     readonly slug: "codex-mini-latest";
@@ -63,7 +67,7 @@ declare const codex_mini_latest_spec: {
     readonly supported_endpoints: ["responses"];
     readonly reasoning_tokens: true;
 };
-export { codex_mini_latest_spec as 'codex-mini-latest' };
+// export { codex_mini_latest_spec as 'codex-mini-latest' };
 declare const computer_use_preview_2025_03_11_spec: {
     readonly name: "computer-use-preview-2025-03-11";
     readonly slug: "computer-use-preview-2025-03-11";
@@ -90,8 +94,8 @@ declare const computer_use_preview_2025_03_11_spec: {
         };
     };
 };
-export { computer_use_preview_2025_03_11_spec as 'computer-use-preview-2025-03-11' };
-export { computer_use_preview_2025_03_11_spec as 'computer-use-preview' };
+// export { computer_use_preview_2025_03_11_spec as 'computer-use-preview-2025-03-11' };
+// export { computer_use_preview_2025_03_11_spec as 'computer-use-preview' };
 declare const dall_e_2_spec: {
     readonly name: "dall-e-2";
     readonly slug: "dall-e-2";
@@ -105,7 +109,7 @@ declare const dall_e_2_spec: {
     readonly supported_features: ["inpainting"];
     readonly reasoning_tokens: false;
 };
-export { dall_e_2_spec as 'dall-e-2' };
+// export { dall_e_2_spec as 'dall-e-2' };
 declare const dall_e_3_spec: {
     readonly name: "dall-e-3";
     readonly slug: "dall-e-3";
@@ -128,7 +132,7 @@ declare const dall_e_3_spec: {
         };
     };
 };
-export { dall_e_3_spec as 'dall-e-3' };
+// export { dall_e_3_spec as 'dall-e-3' };
 declare const davinci_002_spec: {
     readonly name: "davinci-002";
     readonly slug: "davinci-002";
@@ -154,7 +158,7 @@ declare const davinci_002_spec: {
         };
     };
 };
-export { davinci_002_spec as 'davinci-002' };
+// export { davinci_002_spec as 'davinci-002' };
 declare const gpt_3_5_turbo_16k_0613_spec: {
     readonly name: "gpt-3.5-turbo-16k-0613";
     readonly slug: "gpt-3-5-turbo-16k-0613";
@@ -181,7 +185,7 @@ declare const gpt_3_5_turbo_16k_0613_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_16k_0613_spec as 'gpt-3.5-turbo-16k-0613' };
+// export { gpt_3_5_turbo_16k_0613_spec as 'gpt-3.5-turbo-16k-0613' };
 declare const gpt_3_5_turbo_instruct_spec: {
     readonly name: "gpt-3.5-turbo-instruct";
     readonly slug: "gpt-3-5-turbo-instruct";
@@ -204,7 +208,7 @@ declare const gpt_3_5_turbo_instruct_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_instruct_spec as 'gpt-3.5-turbo-instruct' };
+// export { gpt_3_5_turbo_instruct_spec as 'gpt-3.5-turbo-instruct' };
 declare const gpt_3_5_turbo_0125_spec: {
     readonly name: "gpt-3.5-turbo-0125";
     readonly slug: "gpt-3-5-turbo-0125";
@@ -231,7 +235,7 @@ declare const gpt_3_5_turbo_0125_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo-0125' };
+// export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo-0125' };
 declare const gpt_3_5_turbo_1106_spec: {
     readonly name: "gpt-3.5-turbo-1106";
     readonly slug: "gpt-3-5-turbo-1106";
@@ -258,8 +262,8 @@ declare const gpt_3_5_turbo_1106_spec: {
         };
     };
 };
-export { gpt_3_5_turbo_1106_spec as 'gpt-3.5-turbo-1106' };
-export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo' };
+// export { gpt_3_5_turbo_1106_spec as 'gpt-3.5-turbo-1106' };
+// export { gpt_3_5_turbo_0125_spec as 'gpt-3.5-turbo' };
 declare const gpt_4_0125_preview_spec: {
     readonly name: "gpt-4-0125-preview";
     readonly slug: "gpt-4-0125-preview";
@@ -286,7 +290,7 @@ declare const gpt_4_0125_preview_spec: {
         };
     };
 };
-export { gpt_4_0125_preview_spec as 'gpt-4-0125-preview' };
+// export { gpt_4_0125_preview_spec as 'gpt-4-0125-preview' };
 declare const gpt_4_1106_vision_preview_spec: {
     readonly name: "gpt-4-1106-vision-preview";
     readonly slug: "gpt-4-1106-vision-preview";
@@ -314,8 +318,8 @@ declare const gpt_4_1106_vision_preview_spec: {
         };
     };
 };
-export { gpt_4_1106_vision_preview_spec as 'gpt-4-1106-vision-preview' };
-export { gpt_4_0125_preview_spec as 'gpt-4-turbo-preview' };
+// export { gpt_4_1106_vision_preview_spec as 'gpt-4-1106-vision-preview' };
+// export { gpt_4_0125_preview_spec as 'gpt-4-turbo-preview' };
 declare const gpt_4_turbo_2024_04_09_spec: {
     readonly name: "gpt-4-turbo-2024-04-09";
     readonly slug: "gpt-4-turbo-2024-04-09";
@@ -342,8 +346,8 @@ declare const gpt_4_turbo_2024_04_09_spec: {
         };
     };
 };
-export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo-2024-04-09' };
-export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo' };
+// export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo-2024-04-09' };
+// export { gpt_4_turbo_2024_04_09_spec as 'gpt-4-turbo' };
 declare const gpt_4_1_mini_2025_04_14_spec: {
     readonly name: "gpt-4.1-mini-2025-04-14";
     readonly slug: "gpt-4.1-mini-2025-04-14";
@@ -360,8 +364,8 @@ declare const gpt_4_1_mini_2025_04_14_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch", "fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini-2025-04-14' };
-export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini' };
+// export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini-2025-04-14' };
+// export { gpt_4_1_mini_2025_04_14_spec as 'gpt-4.1-mini' };
 declare const gpt_4_1_nano_2025_04_14_spec: {
     readonly name: "gpt-4.1-nano-2025-04-14";
     readonly slug: "gpt-4.1-nano-2025-04-14";
@@ -378,8 +382,8 @@ declare const gpt_4_1_nano_2025_04_14_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch", "fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano-2025-04-14' };
-export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano' };
+// export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano-2025-04-14' };
+// export { gpt_4_1_nano_2025_04_14_spec as 'gpt-4.1-nano' };
 declare const gpt_4_1_2025_04_14_spec: {
     readonly name: "gpt-4.1-2025-04-14";
     readonly slug: "gpt-4.1-2025-04-14";
@@ -406,8 +410,8 @@ declare const gpt_4_1_2025_04_14_spec: {
         };
     };
 };
-export { gpt_4_1_2025_04_14_spec as 'gpt-4.1-2025-04-14' };
-export { gpt_4_1_2025_04_14_spec as 'gpt-4.1' };
+// export { gpt_4_1_2025_04_14_spec as 'gpt-4.1-2025-04-14' };
+// export { gpt_4_1_2025_04_14_spec as 'gpt-4.1' };
 declare const gpt_4_5_preview_2025_02_27_spec: {
     readonly name: "gpt-4.5-preview-2025-02-27";
     readonly slug: "gpt-4.5-preview-2025-02-27";
@@ -424,8 +428,8 @@ declare const gpt_4_5_preview_2025_02_27_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview-2025-02-27' };
-export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview' };
+// export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview-2025-02-27' };
+// export { gpt_4_5_preview_2025_02_27_spec as 'gpt-4.5-preview' };
 declare const gpt_4_0613_spec: {
     readonly name: "gpt-4-0613";
     readonly slug: "gpt-4-0613";
@@ -452,7 +456,7 @@ declare const gpt_4_0613_spec: {
         };
     };
 };
-export { gpt_4_0613_spec as 'gpt-4-0613' };
+// export { gpt_4_0613_spec as 'gpt-4-0613' };
 declare const gpt_4_0314_spec: {
     readonly name: "gpt-4-0314";
     readonly slug: "gpt-4-0314";
@@ -479,8 +483,8 @@ declare const gpt_4_0314_spec: {
         };
     };
 };
-export { gpt_4_0314_spec as 'gpt-4-0314' };
-export { gpt_4_0613_spec as 'gpt-4' };
+// export { gpt_4_0314_spec as 'gpt-4-0314' };
+// export { gpt_4_0613_spec as 'gpt-4' };
 declare const gpt_4o_audio_preview_2025_06_03_spec: {
     readonly name: "gpt-4o-audio-preview-2025-06-03";
     readonly slug: "gpt-4o-audio-preview-2025-06-03";
@@ -507,7 +511,7 @@ declare const gpt_4o_audio_preview_2025_06_03_spec: {
         };
     };
 };
-export { gpt_4o_audio_preview_2025_06_03_spec as 'gpt-4o-audio-preview-2025-06-03' };
+// export { gpt_4o_audio_preview_2025_06_03_spec as 'gpt-4o-audio-preview-2025-06-03' };
 declare const gpt_4o_audio_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-audio-preview-2024-12-17";
     readonly slug: "gpt-4o-audio-preview-2024-12-17";
@@ -534,7 +538,7 @@ declare const gpt_4o_audio_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview-2024-12-17' };
+// export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview-2024-12-17' };
 declare const gpt_4o_audio_preview_2024_10_01_spec: {
     readonly name: "gpt-4o-audio-preview-2024-10-01";
     readonly slug: "gpt-4o-audio-preview-2024-10-01";
@@ -561,8 +565,8 @@ declare const gpt_4o_audio_preview_2024_10_01_spec: {
         };
     };
 };
-export { gpt_4o_audio_preview_2024_10_01_spec as 'gpt-4o-audio-preview-2024-10-01' };
-export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview' };
+// export { gpt_4o_audio_preview_2024_10_01_spec as 'gpt-4o-audio-preview-2024-10-01' };
+// export { gpt_4o_audio_preview_2024_12_17_spec as 'gpt-4o-audio-preview' };
 declare const gpt_4o_mini_audio_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-mini-audio-preview-2024-12-17";
     readonly slug: "gpt-4o-mini-audio-preview-2024-12-17";
@@ -589,8 +593,8 @@ declare const gpt_4o_mini_audio_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview-2024-12-17' };
-export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview' };
+// export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview-2024-12-17' };
+// export { gpt_4o_mini_audio_preview_2024_12_17_spec as 'gpt-4o-mini-audio-preview' };
 declare const gpt_4o_mini_realtime_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-mini-realtime-preview-2024-12-17";
     readonly slug: "gpt-4o-mini-realtime-preview-2024-12-17";
@@ -617,8 +621,8 @@ declare const gpt_4o_mini_realtime_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview-2024-12-17' };
-export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview' };
+// export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview-2024-12-17' };
+// export { gpt_4o_mini_realtime_preview_2024_12_17_spec as 'gpt-4o-mini-realtime-preview' };
 declare const gpt_4o_mini_search_preview_2025_03_11_spec: {
     readonly name: "gpt-4o-mini-search-preview-2025-03-11";
     readonly slug: "gpt-4o-mini-search-preview-2025-03-11";
@@ -635,8 +639,8 @@ declare const gpt_4o_mini_search_preview_2025_03_11_spec: {
     readonly supported_endpoints: ["chat_completions"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview-2025-03-11' };
-export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview' };
+// export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview-2025-03-11' };
+// export { gpt_4o_mini_search_preview_2025_03_11_spec as 'gpt-4o-mini-search-preview' };
 declare const gpt_4o_mini_transcribe_spec: {
     readonly name: "gpt-4o-mini-transcribe";
     readonly slug: "gpt-4o-mini-transcribe";
@@ -652,7 +656,7 @@ declare const gpt_4o_mini_transcribe_spec: {
     readonly supported_endpoints: ["transcription", "realtime"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_transcribe_spec as 'gpt-4o-mini-transcribe' };
+// export { gpt_4o_mini_transcribe_spec as 'gpt-4o-mini-transcribe' };
 declare const gpt_4o_mini_tts_spec: {
     readonly name: "gpt-4o-mini-tts";
     readonly slug: "gpt-4o-mini-tts";
@@ -666,7 +670,7 @@ declare const gpt_4o_mini_tts_spec: {
     readonly supported_endpoints: ["speech_generation"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_tts_spec as 'gpt-4o-mini-tts' };
+// export { gpt_4o_mini_tts_spec as 'gpt-4o-mini-tts' };
 declare const gpt_4o_mini_2024_07_18_spec: {
     readonly name: "gpt-4o-mini-2024-07-18";
     readonly slug: "gpt-4o-mini-2024-07-18";
@@ -683,8 +687,8 @@ declare const gpt_4o_mini_2024_07_18_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch", "fine_tuning"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini-2024-07-18' };
-export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini' };
+// export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini-2024-07-18' };
+// export { gpt_4o_mini_2024_07_18_spec as 'gpt-4o-mini' };
 declare const gpt_4o_realtime_preview_2025_06_03_spec: {
     readonly name: "gpt-4o-realtime-preview-2025-06-03";
     readonly slug: "gpt-4o-realtime-preview-2025-06-03";
@@ -711,7 +715,7 @@ declare const gpt_4o_realtime_preview_2025_06_03_spec: {
         };
     };
 };
-export { gpt_4o_realtime_preview_2025_06_03_spec as 'gpt-4o-realtime-preview-2025-06-03' };
+// export { gpt_4o_realtime_preview_2025_06_03_spec as 'gpt-4o-realtime-preview-2025-06-03' };
 declare const gpt_4o_realtime_preview_2024_12_17_spec: {
     readonly name: "gpt-4o-realtime-preview-2024-12-17";
     readonly slug: "gpt-4o-realtime-preview-2024-12-17";
@@ -738,7 +742,7 @@ declare const gpt_4o_realtime_preview_2024_12_17_spec: {
         };
     };
 };
-export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview-2024-12-17' };
+// export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview-2024-12-17' };
 declare const gpt_4o_realtime_preview_2024_10_01_spec: {
     readonly name: "gpt-4o-realtime-preview-2024-10-01";
     readonly slug: "gpt-4o-realtime-preview-2024-10-01";
@@ -765,8 +769,8 @@ declare const gpt_4o_realtime_preview_2024_10_01_spec: {
         };
     };
 };
-export { gpt_4o_realtime_preview_2024_10_01_spec as 'gpt-4o-realtime-preview-2024-10-01' };
-export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview' };
+// export { gpt_4o_realtime_preview_2024_10_01_spec as 'gpt-4o-realtime-preview-2024-10-01' };
+// export { gpt_4o_realtime_preview_2024_12_17_spec as 'gpt-4o-realtime-preview' };
 declare const gpt_4o_search_preview_2025_03_11_spec: {
     readonly name: "gpt-4o-search-preview-2025-03-11";
     readonly slug: "gpt-4o-search-preview-2025-03-11";
@@ -793,8 +797,8 @@ declare const gpt_4o_search_preview_2025_03_11_spec: {
         };
     };
 };
-export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview-2025-03-11' };
-export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview' };
+// export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview-2025-03-11' };
+// export { gpt_4o_search_preview_2025_03_11_spec as 'gpt-4o-search-preview' };
 declare const gpt_4o_transcribe_spec: {
     readonly name: "gpt-4o-transcribe";
     readonly slug: "gpt-4o-transcribe";
@@ -810,7 +814,7 @@ declare const gpt_4o_transcribe_spec: {
     readonly supported_endpoints: ["transcription", "realtime"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_transcribe_spec as 'gpt-4o-transcribe' };
+// export { gpt_4o_transcribe_spec as 'gpt-4o-transcribe' };
 declare const gpt_4o_2024_11_20_spec: {
     readonly name: "gpt-4o-2024-11-20";
     readonly slug: "gpt-4o-2024-11-20";
@@ -837,7 +841,7 @@ declare const gpt_4o_2024_11_20_spec: {
         };
     };
 };
-export { gpt_4o_2024_11_20_spec as 'gpt-4o-2024-11-20' };
+// export { gpt_4o_2024_11_20_spec as 'gpt-4o-2024-11-20' };
 declare const gpt_4o_2024_08_06_spec: {
     readonly name: "gpt-4o-2024-08-06";
     readonly slug: "gpt-4o-2024-08-06";
@@ -864,7 +868,7 @@ declare const gpt_4o_2024_08_06_spec: {
         };
     };
 };
-export { gpt_4o_2024_08_06_spec as 'gpt-4o-2024-08-06' };
+// export { gpt_4o_2024_08_06_spec as 'gpt-4o-2024-08-06' };
 declare const gpt_4o_2024_05_13_spec: {
     readonly name: "gpt-4o-2024-05-13";
     readonly slug: "gpt-4o-2024-05-13";
@@ -881,8 +885,8 @@ declare const gpt_4o_2024_05_13_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch"];
     readonly reasoning_tokens: false;
 };
-export { gpt_4o_2024_05_13_spec as 'gpt-4o-2024-05-13' };
-export { gpt_4o_2024_08_06_spec as 'gpt-4o' };
+// export { gpt_4o_2024_05_13_spec as 'gpt-4o-2024-05-13' };
+// export { gpt_4o_2024_08_06_spec as 'gpt-4o' };
 declare const gpt_image_1_spec: {
     readonly name: "gpt-image-1";
     readonly slug: "gpt-image-1";
@@ -896,7 +900,7 @@ declare const gpt_image_1_spec: {
     readonly supported_features: ["inpainting"];
     readonly reasoning_tokens: false;
 };
-export { gpt_image_1_spec as 'gpt-image-1' };
+// export { gpt_image_1_spec as 'gpt-image-1' };
 declare const o1_mini_2024_09_12_spec: {
     readonly name: "o1-mini-2024-09-12";
     readonly slug: "o1-mini-2024-09-12";
@@ -923,8 +927,8 @@ declare const o1_mini_2024_09_12_spec: {
         };
     };
 };
-export { o1_mini_2024_09_12_spec as 'o1-mini-2024-09-12' };
-export { o1_mini_2024_09_12_spec as 'o1-mini' };
+// export { o1_mini_2024_09_12_spec as 'o1-mini-2024-09-12' };
+// export { o1_mini_2024_09_12_spec as 'o1-mini' };
 declare const o1_preview_2024_09_12_spec: {
     readonly name: "o1-preview-2024-09-12";
     readonly slug: "o1-preview-2024-09-12";
@@ -951,8 +955,8 @@ declare const o1_preview_2024_09_12_spec: {
         };
     };
 };
-export { o1_preview_2024_09_12_spec as 'o1-preview-2024-09-12' };
-export { o1_preview_2024_09_12_spec as 'o1-preview' };
+// export { o1_preview_2024_09_12_spec as 'o1-preview-2024-09-12' };
+// export { o1_preview_2024_09_12_spec as 'o1-preview' };
 declare const o1_pro_2025_03_19_spec: {
     readonly name: "o1-pro-2025-03-19";
     readonly slug: "o1-pro-2025-03-19";
@@ -979,8 +983,8 @@ declare const o1_pro_2025_03_19_spec: {
         };
     };
 };
-export { o1_pro_2025_03_19_spec as 'o1-pro-2025-03-19' };
-export { o1_pro_2025_03_19_spec as 'o1-pro' };
+// export { o1_pro_2025_03_19_spec as 'o1-pro-2025-03-19' };
+// export { o1_pro_2025_03_19_spec as 'o1-pro' };
 declare const o1_2024_12_17_spec: {
     readonly name: "o1-2024-12-17";
     readonly slug: "o1-2024-12-17";
@@ -1008,8 +1012,8 @@ declare const o1_2024_12_17_spec: {
         };
     };
 };
-export { o1_2024_12_17_spec as 'o1-2024-12-17' };
-export { o1_2024_12_17_spec as 'o1' };
+// export { o1_2024_12_17_spec as 'o1-2024-12-17' };
+// export { o1_2024_12_17_spec as 'o1' };
 declare const o3_mini_2025_01_31_spec: {
     readonly name: "o3-mini-2025-01-31";
     readonly slug: "o3-mini-2025-01-31";
@@ -1026,8 +1030,8 @@ declare const o3_mini_2025_01_31_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "assistants", "batch"];
     readonly reasoning_tokens: true;
 };
-export { o3_mini_2025_01_31_spec as 'o3-mini-2025-01-31' };
-export { o3_mini_2025_01_31_spec as 'o3-mini' };
+// export { o3_mini_2025_01_31_spec as 'o3-mini-2025-01-31' };
+// export { o3_mini_2025_01_31_spec as 'o3-mini' };
 declare const o3_pro_2025_06_10_spec: {
     readonly name: "o3-pro-2025-06-10";
     readonly slug: "o3-pro-2025-06-10";
@@ -1054,8 +1058,8 @@ declare const o3_pro_2025_06_10_spec: {
         };
     };
 };
-export { o3_pro_2025_06_10_spec as 'o3-pro-2025-06-10' };
-export { o3_pro_2025_06_10_spec as 'o3-pro' };
+// export { o3_pro_2025_06_10_spec as 'o3-pro-2025-06-10' };
+// export { o3_pro_2025_06_10_spec as 'o3-pro' };
 declare const o3_2025_04_16_spec: {
     readonly name: "o3-2025-04-16";
     readonly slug: "o3-2025-04-16";
@@ -1083,8 +1087,8 @@ declare const o3_2025_04_16_spec: {
         };
     };
 };
-export { o3_2025_04_16_spec as 'o3-2025-04-16' };
-export { o3_2025_04_16_spec as 'o3' };
+// export { o3_2025_04_16_spec as 'o3-2025-04-16' };
+// export { o3_2025_04_16_spec as 'o3' };
 declare const o4_mini_2025_04_16_spec: {
     readonly name: "o4-mini-2025-04-16";
     readonly slug: "o4-mini-2025-04-16";
@@ -1101,8 +1105,8 @@ declare const o4_mini_2025_04_16_spec: {
     readonly supported_endpoints: ["chat_completions", "responses", "batch", "fine_tuning"];
     readonly reasoning_tokens: true;
 };
-export { o4_mini_2025_04_16_spec as 'o4-mini-2025-04-16' };
-export { o4_mini_2025_04_16_spec as 'o4-mini' };
+// export { o4_mini_2025_04_16_spec as 'o4-mini-2025-04-16' };
+// export { o4_mini_2025_04_16_spec as 'o4-mini' };
 declare const omni_moderation_2024_09_26_spec: {
     readonly name: "omni-moderation-2024-09-26";
     readonly slug: "omni-moderation-2024-09-26";
@@ -1116,8 +1120,8 @@ declare const omni_moderation_2024_09_26_spec: {
     readonly reasoning_tokens: false;
     readonly supported_features: ["image_input"];
 };
-export { omni_moderation_2024_09_26_spec as 'omni-moderation-2024-09-26' };
-export { omni_moderation_2024_09_26_spec as 'omni-moderation-latest' };
+// export { omni_moderation_2024_09_26_spec as 'omni-moderation-2024-09-26' };
+// export { omni_moderation_2024_09_26_spec as 'omni-moderation-latest' };
 declare const text_embedding_3_large_spec: {
     readonly name: "text-embedding-3-large";
     readonly slug: "text-embedding-3-large";
@@ -1129,7 +1133,7 @@ declare const text_embedding_3_large_spec: {
     };
     readonly supported_endpoints: ["embeddings", "batch"];
 };
-export { text_embedding_3_large_spec as 'text-embedding-3-large' };
+// export { text_embedding_3_large_spec as 'text-embedding-3-large' };
 declare const text_embedding_3_small_spec: {
     readonly name: "text-embedding-3-small";
     readonly slug: "text-embedding-3-small";
@@ -1141,7 +1145,7 @@ declare const text_embedding_3_small_spec: {
     };
     readonly supported_endpoints: ["embeddings", "batch"];
 };
-export { text_embedding_3_small_spec as 'text-embedding-3-small' };
+// export { text_embedding_3_small_spec as 'text-embedding-3-small' };
 declare const text_embedding_ada_002_spec: {
     readonly name: "text-embedding-ada-002";
     readonly slug: "text-embedding-ada-002";
@@ -1153,7 +1157,7 @@ declare const text_embedding_ada_002_spec: {
     };
     readonly supported_endpoints: ["embeddings", "batch"];
 };
-export { text_embedding_ada_002_spec as 'text-embedding-ada-002' };
+// export { text_embedding_ada_002_spec as 'text-embedding-ada-002' };
 declare const text_moderation_007_spec: {
     readonly name: "text-moderation-007";
     readonly slug: "text-moderation-007";
@@ -1168,9 +1172,9 @@ declare const text_moderation_007_spec: {
     readonly supported_endpoints: ["moderation"];
     readonly reasoning_tokens: false;
 };
-export { text_moderation_007_spec as 'text-moderation-007' };
-export { text_moderation_007_spec as 'text-moderation-latest' };
-export { text_moderation_007_spec as 'text-moderation-stable' };
+// export { text_moderation_007_spec as 'text-moderation-007' };
+// export { text_moderation_007_spec as 'text-moderation-latest' };
+// export { text_moderation_007_spec as 'text-moderation-stable' };
 declare const tts_1_hd_spec: {
     readonly name: "tts-1-hd";
     readonly slug: "tts-1-hd";
@@ -1184,7 +1188,7 @@ declare const tts_1_hd_spec: {
     readonly supported_endpoints: ["speech_generation"];
     readonly reasoning_tokens: false;
 };
-export { tts_1_hd_spec as 'tts-1-hd' };
+// export { tts_1_hd_spec as 'tts-1-hd' };
 declare const tts_1_spec: {
     readonly name: "tts-1";
     readonly slug: "tts-1";
@@ -1198,7 +1202,7 @@ declare const tts_1_spec: {
     readonly supported_endpoints: ["speech_generation"];
     readonly reasoning_tokens: false;
 };
-export { tts_1_spec as 'tts-1' };
+// export { tts_1_spec as 'tts-1' };
 declare const whisper_1_spec: {
     readonly name: "whisper-1";
     readonly slug: "whisper-1";
@@ -1211,4 +1215,4 @@ declare const whisper_1_spec: {
     readonly supported_endpoints: ["transcription", "translation"];
     readonly reasoning_tokens: false;
 };
-export { whisper_1_spec as 'whisper-1' };
+// export { whisper_1_spec as 'whisper-1' };
