diff --git a/src/global-setup.js b/src/global-setup.js
index 9324b6d016551d20faafc4bc23aaf186b1dcc0ab..6e44c40dd083f73423577b742043a73ae4a64da8 100644
--- a/src/global-setup.js
+++ b/src/global-setup.js
@@ -37,6 +37,10 @@ const jest = {
     jestTimer.useRealTimers();
     return jest;
   },
+  runOnlyPendingTimers() {
+    jestTimer.runOnlyPendingTimers();
+    return jest;
+  },
 };
 
 globalThis.expect = expect;