diff --git a/src/index.js b/src/index.js
index 00c73166a63f76a95237dadf59ae0583199df8f5..8d7e4982cff41c8e141fe4db5f6fb1df61ddc352 100644
--- a/src/index.js
+++ b/src/index.js
@@ -4,22 +4,10 @@ const EventEmitter = require('events');
 const JSONB = require('json-buffer');
 
 const loadStore = options => {
-	const adapters = {
-		redis: '@keyv/redis',
-		rediss: '@keyv/redis',
-		mongodb: '@keyv/mongo',
-		mongo: '@keyv/mongo',
-		sqlite: '@keyv/sqlite',
-		postgresql: '@keyv/postgres',
-		postgres: '@keyv/postgres',
-		mysql: '@keyv/mysql',
-		etcd: '@keyv/etcd',
-		offline: '@keyv/offline',
-		tiered: '@keyv/tiered',
-	};
+	// NOTE(calebmer): Removed support for `keyv` store adapters since we don't want
+	// Webpack bundling all these adapter modules in the dynamic require.
 	if (options.adapter || options.uri) {
-		const adapter = options.adapter || /^[^:+]*/.exec(options.uri)[0];
-		return new (require(adapters[adapter]))(options);
+		throw new Error('keyv adapter support disabled');
 	}
 
 	return new Map();