diff --git a/lib/index.js b/lib/index.js
index 11e51158c5cece45ab736893d257fa98aef6cede..4208ff686bd724215cbf0dcc7e1f49748a4790a0 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -213,13 +213,23 @@ export function gfmTableToMarkdown(options) {
   function handleTableCell(node, _, state, info) {
     const exit = state.enter('tableCell')
     const subexit = state.enter('phrasing')
-    const value = state.containerPhrasing(node, {
+    let value = state.containerPhrasing(node, {
       ...info,
       before: around,
       after: around
     })
     subexit()
     exit()
+
+    // NOTE(calebmer): If a table cell ends in a trailing space, encode the last
+    // space as an HTML entity. Since GFM tables use space to align columns,
+    // trailing spaces will otherwise be lost.
+    if (value.length > 0 && value[value.length - 1] === ' ') {
+      value = value.slice(0, -1) + '&#x20;'
+    }
+    if (value.length > 0 && value[0] === ' ') {
+      value = '&#x20;' + value.slice(1)
+    }
     return value
   }
 
