diff --git a/dev/lib/index.d.ts b/dev/lib/index.d.ts
--- a/dev/lib/index.d.ts
+++ b/dev/lib/index.d.ts
@@ -156,6 +156,11 @@ export type FromMarkdownOptions = {
    * Extensions for this utility to change how tokens are turned into a tree.
    */
   mdastExtensions?: Array<Extension | Array<Extension>> | null | undefined
+  // NOTE(calebmer): We've patched `micromark-core-common` with these options.
+  allowUndefinedLinkReferenceIdentifiers?: boolean
+  allowAttentionWithoutClose?: boolean
+  allowCodeTextWithoutClose?: boolean
+  allowLabelWithoutClose?: boolean
 }
 /**
  * Configuration.
diff --git a/lib/index.d.ts b/lib/index.d.ts
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -156,6 +156,11 @@ export type FromMarkdownOptions = {
    * Extensions for this utility to change how tokens are turned into a tree.
    */
   mdastExtensions?: Array<Extension | Array<Extension>> | null | undefined
+  // NOTE(calebmer): We've patched `micromark-core-common` with these options.
+  allowUndefinedLinkReferenceIdentifiers?: boolean
+  allowAttentionWithoutClose?: boolean
+  allowCodeTextWithoutClose?: boolean
+  allowLabelWithoutClose?: boolean
 }
 /**
  * Configuration.
