diff --git a/dev/lib/parse.js b/dev/lib/parse.js
--- a/dev/lib/parse.js
+++ b/dev/lib/parse.js
@@ -27,6 +27,11 @@ export function parse(options) {
   /** @type {ParseContext} */
   const parser = {
     defined: [],
+    // NOTE(calebmer): We've patched `micromark-core-common` with these options.
+    allowUndefinedLinkReferenceIdentifiers: options.allowUndefinedLinkReferenceIdentifiers || false,
+    allowAttentionWithoutClose: options.allowAttentionWithoutClose || false,
+    allowCodeTextWithoutClose: options.allowCodeTextWithoutClose || false,
+    allowLabelWithoutClose: options.allowLabelWithoutClose || false,
     lazy: {},
     constructs,
     content: create(content),
diff --git a/lib/parse.js b/lib/parse.js
--- a/lib/parse.js
+++ b/lib/parse.js
@@ -27,6 +27,11 @@ export function parse(options) {
   /** @type {ParseContext} */
   const parser = {
     defined: [],
+    // NOTE(calebmer): We've patched `micromark-core-common` with these options.
+    allowUndefinedLinkReferenceIdentifiers: options.allowUndefinedLinkReferenceIdentifiers || false,
+    allowAttentionWithoutClose: options.allowAttentionWithoutClose || false,
+    allowCodeTextWithoutClose: options.allowCodeTextWithoutClose || false,
+    allowLabelWithoutClose: options.allowLabelWithoutClose || false,
     lazy: {},
     constructs,
     content: create(content),
