diff --git a/dev/lib/parse.js b/dev/lib/parse.js
index aa2c637696e38b398efdc22ddd856493d740c62c..36c8b6ed33e5a62c9332a7f4628e6df97dd7dcb3 100644
--- a/dev/lib/parse.js
+++ b/dev/lib/parse.js
@@ -27,6 +27,12 @@ export function parse(options) {
   /** @type {ParseContext} */
   const parser = {
     defined: [],
+    // NOTE(calebmer): We've patched `micromark-core-common` with these options.
+    allowUndefinedLinkReferenceIdentifiers: options.allowUndefinedLinkReferenceIdentifiers || false,
+    allowAttentionWithoutClose: options.allowAttentionWithoutClose || false,
+    allowCodeTextWithoutClose: options.allowCodeTextWithoutClose || false,
+    allowLabelWithoutClose: options.allowLabelWithoutClose || false,
+    allowResourceWithoutClose: options.allowResourceWithoutClose || false,
     lazy: {},
     constructs,
     content: create(content),
diff --git a/lib/parse.js b/lib/parse.js
index 4af5729cc24bb0ca1eb77a5c1cce51a23f3c43a0..7f7b7bc6331df1f314fe8541e754bb626730e07d 100644
--- a/lib/parse.js
+++ b/lib/parse.js
@@ -27,6 +27,12 @@ export function parse(options) {
   /** @type {ParseContext} */
   const parser = {
     defined: [],
+    // NOTE(calebmer): We've patched `micromark-core-common` with these options.
+    allowUndefinedLinkReferenceIdentifiers: options.allowUndefinedLinkReferenceIdentifiers || false,
+    allowAttentionWithoutClose: options.allowAttentionWithoutClose || false,
+    allowCodeTextWithoutClose: options.allowCodeTextWithoutClose || false,
+    allowLabelWithoutClose: options.allowLabelWithoutClose || false,
+    allowResourceWithoutClose: options.allowResourceWithoutClose || false,
     lazy: {},
     constructs,
     content: create(content),
