diff --git a/lib/index.js b/lib/index.js
index f8738b2f6c54941f5865a6be7232fb153a08268b..28707a7ac9bf62308b93a9f96813a7445033f6e0 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -59,7 +59,10 @@ function MJMLParser(xml, options = {}, includedIn = []) {
   } = options;
   const endingTags = (0, _flow2.default)((0, _filter2.default)(component => component.endingTag), (0, _map2.default)(component => component.getTagName()))({ ...components
   });
-  let cwd = process.cwd();
+  // NOTE(calebmer): Process global is not available in Cloudflare Workers.
+  // Use the same mock value as Webpack:
+  // https://github.com/webpack/node-libs-browser/blob/214057f118bad5da8b99db33fd16a5b6ceb42d9b/mock/process.js#L21
+  let cwd = isNode ? process.cwd() : '/';
 
   if (isNode && filePath) {
     try {