diff --git a/dist/build/webpack/loaders/utils.js b/dist/build/webpack/loaders/utils.js
index 5c3625d5038fe2d5e7225cc6364fe42bf5942729..52f40894ecbcb2fc84a5a0da0ccc841eae7bad73 100644
--- a/dist/build/webpack/loaders/utils.js
+++ b/dist/build/webpack/loaders/utils.js
@@ -18,7 +18,8 @@ function isClientComponentModule(mod) {
     const hasClientDirective = ((ref = mod.buildInfo.rsc) == null ? void 0 : ref.type) === _constants.RSC_MODULE_TYPES.client;
     return hasClientDirective || imageRegex.test(mod.resource);
 }
-const regexCSS = /\.(css|scss|sass)(\?.*)?$/;
+// NOTE(calebmer): Treat `vanilla-extract` files as CSS too.
+const regexCSS = /\.(css|scss|sass)(\.(js|jsx|ts|tsx|mjs))?(\?.*)?$/;
 exports.regexCSS = regexCSS;
 
 //# sourceMappingURL=utils.js.map
diff --git a/dist/lib/typescript/writeVscodeConfigurations.js b/dist/lib/typescript/writeVscodeConfigurations.js
index 067a032ec4d2b1ab86249640e9bf5e67b394f82f..cbd1547e6e6f884c11da25b8806a360df6489d0a 100644
--- a/dist/lib/typescript/writeVscodeConfigurations.js
+++ b/dist/lib/typescript/writeVscodeConfigurations.js
@@ -11,39 +11,10 @@ function _interopRequireDefault(obj) {
     };
 }
 async function writeVscodeConfigurations(baseDir) {
-    const vscodeSettings = _path.default.join(baseDir, ".vscode", "settings.json");
-    let settings = {};
-    let currentContent = "";
-    try {
-        currentContent = await _fs.promises.readFile(vscodeSettings, "utf8");
-        settings = JSON.parse(currentContent);
-    } catch (err) {}
-    const libPath = "." + _path.default.sep + _path.default.join("node_modules", "typescript", "lib");
-    if (settings["typescript.tsdk"] === libPath && settings["typescript.enablePromptUseWorkspaceTsdk"]) {
-        return;
-    }
-    settings["typescript.tsdk"] = libPath;
-    settings["typescript.enablePromptUseWorkspaceTsdk"] = true;
-    const content = JSON.stringify(settings, null, 2);
-    const vscodeFolder = _path.default.join(baseDir, ".vscode");
-    try {
-        await _fs.promises.lstat(vscodeFolder);
-    } catch (e) {
-        await _fs.promises.mkdir(vscodeFolder, {
-            recursive: true
-        });
-    }
-    await _fs.promises.writeFile(vscodeSettings, content);
-    // Write to .gitignore if it exists
-    const gitIgnore = _path.default.join(baseDir, ".gitignore");
-    try {
-        const gitIgnoreContent = await _fs.promises.readFile(gitIgnore, "utf8");
-        if (!gitIgnoreContent.includes(".vscode")) {
-            await _fs.promises.writeFile(gitIgnore, `${gitIgnoreContent}\n.vscode\n`);
-        }
-    } catch (e1) {
-        await _fs.promises.writeFile(gitIgnore, `.vscode\n`);
-    }
+    // NOTE(calebmer): This function doesn't currently support VSCode settings
+    // files with comments. There is a change in the Next.js GitHub to support
+    // this so we can remove our patch once that lands.
+    // https://github.com/vercel/next.js/blob/9ac42fd6f3fe143f32662487a6586e49acab3194/packages/next/lib/typescript/writeVscodeConfigurations.ts#L19
 }
 
 //# sourceMappingURL=writeVscodeConfigurations.js.map