diff --git a/lib/common/configLoader.js b/lib/common/configLoader.js
index dbced6dedc19b21613701aca015a2940352642fc..b076eb63c55123a2c0112c5eebf31f62c569ca97 100644
--- a/lib/common/configLoader.js
+++ b/lib/common/configLoader.js
@@ -264,6 +264,11 @@ async function loadEmptyConfigForMergeReports() {
   });
 }
 function restartWithExperimentalTsEsm(configFile, force = false) {
+  // NOTE(calebmer): We transform our own code through Bazel. Add an option to
+  // disable all Playwright code transformation. Since Playwright code
+  // transformation slows things down and messes with source maps.
+  if (process.env.PLAYWRIGHT_DISABLE_TRANSFORMS) return false;  
+
   // Opt-out switch.
   if (process.env.PW_DISABLE_TS_ESM) return false;
 
diff --git a/lib/reporters/base.js b/lib/reporters/base.js
index d46d5cbd2e347d837718d745a07a91bcee6fbc83..d4241ef04e7587105b9146e2c77559b96b8d40d7 100644
--- a/lib/reporters/base.js
+++ b/lib/reporters/base.js
@@ -309,13 +309,24 @@ function formatFailure(config, test, options = {}) {
         resultLines.push(colors.cyan(separator(`    attachment #${i + 1}: ${attachment.name} (${attachment.contentType})`)));
         if (attachment.path) {
           const relativePath = _path.default.relative(process.cwd(), attachment.path);
-          resultLines.push(colors.cyan(`    ${relativePath}`));
+          // NOTE(calebmer): The paths reported by Playwright for our test logs
+          // are very confusing and inaccessible since they're in a sandbox.
+          // Modify the reporter to print a friendlier path to our developers.
+          const bazelTestlogsMatch = relativePath.match(/(?:^|\/)testlogs\/(.+\/integration_tests)\/(.+)\/test\.outputs\/(.+)/);
+          if (!bazelTestlogsMatch) {
+            resultLines.push(colors.cyan(`    ${relativePath}`));
+          } else {
+            resultLines.push(colors.cyan(`    bazel-testlogs/${bazelTestlogsMatch[1]}/${bazelTestlogsMatch[2]}/test.outputs/outputs.zip contains ${bazelTestlogsMatch[3]}`));
+          }
           // Make this extensible
           if (attachment.name === 'trace') {
-            const packageManagerCommand = (0, _utils.getPackageManagerExecCommand)();
             resultLines.push(colors.cyan(`    Usage:`));
             resultLines.push('');
-            resultLines.push(colors.cyan(`        ${packageManagerCommand} playwright show-trace ${quotePathIfNeeded(relativePath)}`));
+            if (!bazelTestlogsMatch) {
+              resultLines.push(colors.cyan(`        npx playwright show-trace ${relativePath}`));
+            } else {
+              resultLines.push(colors.cyan(`        bazel run //${bazelTestlogsMatch[1]}:${bazelTestlogsMatch[2]}_show_trace -- ${bazelTestlogsMatch[3]}`));
+            }              
             resultLines.push('');
           }
         } else {
diff --git a/lib/transform/transform.js b/lib/transform/transform.js
index a1ec4ef884a12157e4a509788ed35c0d280b4831..0e25514d334a30f6ebf1344ab91fa70963da01e2 100644
--- a/lib/transform/transform.js
+++ b/lib/transform/transform.js
@@ -82,6 +82,11 @@ function loadAndValidateTsconfigsForFile(file) {
 const pathSeparator = process.platform === 'win32' ? ';' : ':';
 const builtins = new Set(_module.default.builtinModules);
 function resolveHook(filename, specifier) {
+  // NOTE(calebmer): We transform our own code through Bazel. Add an option to
+  // disable all Playwright code transformation. Since Playwright code
+  // transformation slows things down and messes with source maps.
+  if (process.env.PLAYWRIGHT_DISABLE_TRANSFORMS) return;  
+
   if (specifier.startsWith('node:') || builtins.has(specifier)) return;
   if (!shouldTransform(filename)) return;
   if (isRelativeSpecifier(specifier)) return (0, _util.resolveImportSpecifierExtension)(_path.default.resolve(_path.default.dirname(filename), specifier));
@@ -143,6 +148,11 @@ function setTransformData(pluginName, value) {
   transformData.set(pluginName, value);
 }
 function transformHook(originalCode, filename, moduleUrl) {
+  // NOTE(calebmer): We transform our own code through Bazel. Add an option to
+  // disable all Playwright code transformation. Since Playwright code
+  // transformation slows things down and messes with source maps.
+  if (process.env.PLAYWRIGHT_DISABLE_TRANSFORMS) return originalCode;  
+
   const isTypeScript = filename.endsWith('.ts') || filename.endsWith('.tsx') || filename.endsWith('.mts') || filename.endsWith('.cts');
   const hasPreprocessor = process.env.PW_TEST_SOURCE_TRANSFORM && process.env.PW_TEST_SOURCE_TRANSFORM_SCOPE && process.env.PW_TEST_SOURCE_TRANSFORM_SCOPE.split(pathSeparator).some(f => filename.startsWith(f));
   const pluginsPrologue = _transformConfig.babelPlugins;
@@ -201,6 +211,11 @@ async function requireOrImport(file) {
   }
 }
 function installTransform() {
+  // NOTE(calebmer): We transform our own code through Bazel. Add an option to
+  // disable all Playwright code transformation. Since Playwright code
+  // transformation slows things down and messes with source maps.
+  if (process.env.PLAYWRIGHT_DISABLE_TRANSFORMS) return () => {};  
+
   (0, _compilationCache.installSourceMapSupportIfNeeded)();
   let reverted = false;
   const originalResolveFilename = _module.default._resolveFilename;
