diff --git a/dist/index.cjs b/dist/index.cjs
index 475bbd8642fc713f7bf3a9e20899f4e1cf923bfa..2d3c7374040a99f2ace46dc6cfe056dd59468ce1 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -386,6 +386,39 @@ var NodeSelection = function (_Selection2) {
     value: function getBookmark() {
       return new NodeBookmark(this.anchor);
     }
+  }, {
+    key: "replace",
+    value: function replace(tr) {
+      var content = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : prosemirrorModel.Slice.empty;
+
+      // NOTE(calebmer): When deleting a `file` node that's the only `file` in a
+      // `fileRow` then instead of replacing the `file` with one that has a null `id`
+      // instead delete the whole `fileRow`. We chose to put this here so that every
+      // time `tr.deleteSelection()` is called this logic runs. We could manually
+      // write this logic everywhere `tr.deleteSelection()` is called but sometimes
+      // `tr.deleteSelection()` is called at a framework level in `prosemirror-view`.
+      // Not just in keymap backspace/delete handlers.
+      if (
+        !content.size &&
+        this.node.type.name === "file" &&
+        (this.$anchor.parent.type.name === "fileRow" ||
+          this.$anchor.parent.type.name === "fileFloat" ||
+          this.$anchor.parent.type.name === "fileRowTable") &&
+        this.$anchor.parent.childCount === 1
+      ) {
+        let mapFrom = tr.steps.length;
+        let mapping = tr.mapping.slice(mapFrom);
+        tr.replaceRange(
+          mapping.map(this.$anchor.before()),
+          mapping.map(this.$anchor.after()),
+          content,
+        );
+        selectionToInsertionEnd(tr, mapFrom, 1);
+        return;
+      }
+
+      _get(_getPrototypeOf(NodeSelection.prototype), "replace", this).call(this, tr, content);
+    }
   }], [{
     key: "fromJSON",
     value: function fromJSON(doc, json) {
diff --git a/dist/index.js b/dist/index.js
index 5dc1684937ed1d8de749629ac5989183339df41b..74fe3112ec93b204aa04fe64e4b404abe3628879 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -381,6 +381,35 @@ class NodeSelection extends Selection {
     static isSelectable(node) {
         return !node.isText && node.type.spec.selectable !== false;
     }
+    replace(tr, content = Slice.empty) {
+        // NOTE(calebmer): When deleting a `file` node that's the only `file` in a
+        // `fileRow` then instead of replacing the `file` with one that has a null `id`
+        // instead delete the whole `fileRow`. We chose to put this here so that every
+        // time `tr.deleteSelection()` is called this logic runs. We could manually
+        // write this logic everywhere `tr.deleteSelection()` is called but sometimes
+        // `tr.deleteSelection()` is called at a framework level in `prosemirror-view`.
+        // Not just in keymap backspace/delete handlers.
+        if (
+            !content.size &&
+            this.node.type.name === "file" &&
+            (this.$anchor.parent.type.name === "fileRow" ||
+                this.$anchor.parent.type.name === "fileFloat" ||
+                this.$anchor.parent.type.name === "fileRowTable") &&
+            this.$anchor.parent.childCount === 1
+        ) {
+            let mapFrom = tr.steps.length;
+            let mapping = tr.mapping.slice(mapFrom);
+            tr.replaceRange(
+                mapping.map(this.$anchor.before()),
+                mapping.map(this.$anchor.after()),
+                content,
+            );
+            selectionToInsertionEnd(tr, mapFrom, 1);
+            return;
+        }
+
+        super.replace(tr, content);
+    }
 }
 NodeSelection.prototype.visible = false;
 Selection.jsonID("node", NodeSelection);
diff --git a/src/selection.ts b/src/selection.ts
index c0ff60847f94c5ac8f6cb654780cf9e81bf372d8..030fd81532890f5a9a0f69a486a372dde816e949 100644
--- a/src/selection.ts
+++ b/src/selection.ts
@@ -373,6 +373,36 @@ export class NodeSelection extends Selection {
   static isSelectable(node: Node) {
     return !node.isText && node.type.spec.selectable !== false
   }
+
+  replace(tr: Transaction, content = Slice.empty) {
+    // NOTE(calebmer): When deleting a `file` node that's the only `file` in a
+    // `fileRow` then instead of replacing the `file` with one that has a null `id`
+    // instead delete the whole `fileRow`. We chose to put this here so that every
+    // time `tr.deleteSelection()` is called this logic runs. We could manually
+    // write this logic everywhere `tr.deleteSelection()` is called but sometimes
+    // `tr.deleteSelection()` is called at a framework level in `prosemirror-view`.
+    // Not just in keymap backspace/delete handlers.
+    if (
+      !content.size &&
+      this.node.type.name === "file" &&
+      (this.$anchor.parent.type.name === "fileRow" ||
+        this.$anchor.parent.type.name === "fileFloat" ||
+        this.$anchor.parent.type.name === "fileRowTable") &&
+      this.$anchor.parent.childCount === 1
+    ) {
+      let mapFrom = tr.steps.length
+      let mapping = tr.mapping.slice(mapFrom)
+      tr.replaceRange(
+        mapping.map(this.$anchor.before()),
+        mapping.map(this.$anchor.after()),
+        content,
+      )
+      selectionToInsertionEnd(tr, mapFrom, 1)
+      return;
+    }
+
+    super.replace(tr, content)
+  }
 }
 
 NodeSelection.prototype.visible = false
