diff --git a/dist/index.cjs b/dist/index.cjs
index 6f4e0fcfa3f07ec49e8dc392fe9a1d0e5c3adefd..47d1ba098a159b1de6cdccf845c7f68ec8a05e44 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -181,8 +181,32 @@ function scrollRectIntoView(view, rect, startDOM) {
     var bounding = atTop ? windowRect(doc) : clientRect(elt);
     var moveX = 0,
         moveY = 0;
-    if (rect.top < bounding.top + getSide(scrollThreshold, "top")) moveY = -(bounding.top - rect.top + getSide(scrollMargin, "top"));else if (rect.bottom > bounding.bottom - getSide(scrollThreshold, "bottom")) moveY = rect.bottom - bounding.bottom + getSide(scrollMargin, "bottom");
-    if (rect.left < bounding.left + getSide(scrollThreshold, "left")) moveX = -(bounding.left - rect.left + getSide(scrollMargin, "left"));else if (rect.right > bounding.right - getSide(scrollThreshold, "right")) moveX = rect.right - bounding.right + getSide(scrollMargin, "right");
+    // NOTE(calebmer): Oddly, the implementation of `scrollRectIntoView()` tries to
+    // scroll elements even when they're not scrollable! Check if the element is
+    // scrollable before computing `moveX` and `moveY`. Also bound `moveX` and
+    // `moveY` by the min/max values it can move.
+    //
+    // Admittedly, the implementation was probably fine without putting these bounds
+    // in place but these changes do lead to cleaner scroll event debugger logging.
+    var eltStyle = getComputedStyle(elt);
+    var overflowX = eltStyle.overflowX;
+    var overflowY = eltStyle.overflowY;
+    if (overflowY === "auto" || overflowY === "scroll") {
+      if (rect.top < bounding.top + getSide(scrollThreshold, "top")) moveY = -(bounding.top - rect.top + getSide(scrollMargin, "top"));else if (rect.bottom > bounding.bottom - getSide(scrollThreshold, "bottom")) moveY = rect.bottom - bounding.bottom + getSide(scrollMargin, "bottom");
+
+      if (moveY) {
+        moveY = Math.min(moveY, elt.scrollHeight - elt.clientHeight - elt.scrollTop)
+        moveY = Math.max(moveY, -elt.scrollTop)
+      }
+    }
+    if (overflowX === "auto" || overflowX === "scroll") {
+      if (rect.left < bounding.left + getSide(scrollThreshold, "left")) moveX = -(bounding.left - rect.left + getSide(scrollMargin, "left"));else if (rect.right > bounding.right - getSide(scrollThreshold, "right")) moveX = rect.right - bounding.right + getSide(scrollMargin, "right");
+
+      if (moveX) {
+        moveX = Math.min(moveX, elt.scrollWidth - elt.clientWidth - elt.scrollLeft)
+        moveX = Math.max(moveX, -elt.scrollLeft)
+      }
+    }
 
     if (moveX || moveY) {
       if (atTop) {
@@ -203,7 +227,7 @@ function scrollRectIntoView(view, rect, startDOM) {
       }
     }
 
-    if (atTop) break;
+    if (atTop || /^(fixed|sticky)$/.test(getComputedStyle(parent).position)) break;
   }
 }
 
@@ -3888,9 +3912,33 @@ handlers.focus = function (view) {
     view.dom.classList.add("ProseMirror-focused");
     view.domObserver.start();
     view.focused = true;
-    setTimeout(function () {
-      if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection())) selectionToDOM(view);
-    }, 20);
+
+    if (view.shouldUseDOMSelectionOnFocus) {
+      // NOTE(calebmer): When the user clicks into the editor, provide a
+      // configuration option to use the new selection from the browser instead
+      // of restoring the selection from state.
+      var newSel = selectionFromDOM(view);
+      if (newSel && !view.state.selection.eq(newSel)) {
+        view.dispatch(view.state.tr.setSelection(newSel));
+      }
+    } else {
+      // NOTE(calebmer): When focus is returned to the view, always serialize our
+      // selection from state back to the DOM. Otherwise we're seeing Chrome
+      // sometimes reset the selection.
+      //
+      // To reproduce the bug which caused us to add this in Chrome:
+      //
+      // 1. Highlight some text
+      // 2. Open the link input
+      // 3. Hit escape to close the link input
+      //
+      // Without this, Chrome restores focus to the beginning of the editor.
+      selectionToDOM(view);
+
+      setTimeout(function () {
+        if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection())) selectionToDOM(view);
+      }, 20);
+    }
   }
 };
 
@@ -5392,6 +5440,8 @@ var EditorView = function () {
     this.domObserver.start();
     initInput(this);
     this.updatePluginViews();
+
+    this.shouldUseDOMSelectionOnFocus = props.shouldUseDOMSelectionOnFocus || false;
   }
 
   _createClass(EditorView, [{
@@ -5792,3 +5842,6 @@ exports.EditorView = EditorView;
 exports.__endComposition = __endComposition;
 exports.__parseFromClipboard = __parseFromClipboard;
 exports.__serializeForClipboard = __serializeForClipboard;
+
+// NOTE(calebmer): Exported so we can build our own write to clipboard logic.
+exports.serializeForClipboard = serializeForClipboard;
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 52ee8840c754562f9ae4d9891f61682035b092e7..1f85b45afccc6fc0091ca8dbffdd2c5eec2c0e60 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -831,6 +831,17 @@ interface DirectEditorProps extends EditorProps {
     the view instance as its `this` binding.
     */
     dispatchTransaction?: (tr: Transaction) => void;
+
+    // NOTE(calebmer): When the user clicks into the editor, provide a
+    // configuration option to use the new selection from the browser instead
+    // of restoring the selection from state.
+    shouldUseDOMSelectionOnFocus?: boolean;
 }
 
 export { DOMEventMap, Decoration, DecorationAttrs, DecorationSet, DecorationSource, DirectEditorProps, EditorProps, EditorView, MarkViewConstructor, NodeView, NodeViewConstructor };
+
+// NOTE(calebmer): Exported so we can build our own write to clipboard logic.
+export function serializeForClipboard(view: EditorView, slice: Slice): {
+    dom: HTMLDivElement;
+    text: string;
+};
diff --git a/dist/index.js b/dist/index.js
index bb65a77310b75d96a0f548d5fc2e36ff54fc8d3a..6a70e4bea94983c67ef46006776d62d2e3238555 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -141,14 +141,36 @@ function scrollRectIntoView(view, rect, startDOM) {
         let atTop = elt == doc.body;
         let bounding = atTop ? windowRect(doc) : clientRect(elt);
         let moveX = 0, moveY = 0;
-        if (rect.top < bounding.top + getSide(scrollThreshold, "top"))
-            moveY = -(bounding.top - rect.top + getSide(scrollMargin, "top"));
-        else if (rect.bottom > bounding.bottom - getSide(scrollThreshold, "bottom"))
-            moveY = rect.bottom - bounding.bottom + getSide(scrollMargin, "bottom");
-        if (rect.left < bounding.left + getSide(scrollThreshold, "left"))
-            moveX = -(bounding.left - rect.left + getSide(scrollMargin, "left"));
-        else if (rect.right > bounding.right - getSide(scrollThreshold, "right"))
-            moveX = rect.right - bounding.right + getSide(scrollMargin, "right");
+        // NOTE(calebmer): Oddly, the implementation of `scrollRectIntoView()` tries to
+        // scroll elements even when they're not scrollable! Check if the element is
+        // scrollable before computing `moveX` and `moveY`. Also bound `moveX` and
+        // `moveY` by the min/max values it can move.
+        //
+        // Admittedly, the implementation was probably fine without putting these bounds
+        // in place but these changes do lead to cleaner scroll event debugger logging.
+        let {overflowX, overflowY} = getComputedStyle(elt)
+        if (overflowY === "auto" || overflowY === "scroll") {
+            if (rect.top < bounding.top + getSide(scrollThreshold, "top"))
+                moveY = -(bounding.top - rect.top + getSide(scrollMargin, "top"));
+            else if (rect.bottom > bounding.bottom - getSide(scrollThreshold, "bottom"))
+                moveY = rect.bottom - bounding.bottom + getSide(scrollMargin, "bottom");
+
+            if (moveY) {
+                moveY = Math.min(moveY, elt.scrollHeight - elt.clientHeight - elt.scrollTop)
+                moveY = Math.max(moveY, -elt.scrollTop)
+            }
+        }
+        if (overflowX === "auto" || overflowX === "scroll") {
+            if (rect.left < bounding.left + getSide(scrollThreshold, "left"))
+                moveX = -(bounding.left - rect.left + getSide(scrollMargin, "left"));
+            else if (rect.right > bounding.right - getSide(scrollThreshold, "right"))
+                moveX = rect.right - bounding.right + getSide(scrollMargin, "right");
+
+            if (moveX) {
+                moveX = Math.min(moveX, elt.scrollWidth - elt.clientWidth - elt.scrollLeft)
+                moveX = Math.max(moveX, -elt.scrollLeft)
+            }
+        }
         if (moveX || moveY) {
             if (atTop) {
                 doc.defaultView.scrollBy(moveX, moveY);
@@ -163,7 +185,7 @@ function scrollRectIntoView(view, rect, startDOM) {
                 rect = { left: rect.left - dX, top: rect.top - dY, right: rect.right - dX, bottom: rect.bottom - dY };
             }
         }
-        if (atTop)
+        if (atTop || /^(fixed|sticky)$/.test(getComputedStyle(parent).position))
             break;
     }
 }
@@ -3460,10 +3482,34 @@ handlers.focus = view => {
         view.dom.classList.add("ProseMirror-focused");
         view.domObserver.start();
         view.focused = true;
-        setTimeout(() => {
-            if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
-                selectionToDOM(view);
-        }, 20);
+
+        if (view.shouldUseDOMSelectionOnFocus) {
+            // NOTE(calebmer): When the user clicks into the editor, provide a
+            // configuration option to use the new selection from the browser instead
+            // of restoring the selection from state.
+            let newSel = selectionFromDOM(view);
+            if (newSel && !view.state.selection.eq(newSel)) {
+              view.dispatch(view.state.tr.setSelection(newSel));
+            }
+        } else {      
+            // NOTE(calebmer): When focus is returned to the view, always serialize our
+            // selection from state back to the DOM. Otherwise we're seeing Chrome
+            // sometimes reset the selection.
+            //
+            // To reproduce the bug which caused us to add this in Chrome:
+            //
+            // 1. Highlight some text
+            // 2. Open the link input
+            // 3. Hit escape to close the link input
+            //
+            // Without this, Chrome restores focus to the beginning of the editor.
+            selectionToDOM(view);
+
+            setTimeout(() => {
+                if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
+                    selectionToDOM(view);
+            }, 20);
+        }
     }
 };
 handlers.blur = (view, _event) => {
@@ -4879,6 +4925,8 @@ class EditorView {
         this.domObserver.start();
         initInput(this);
         this.updatePluginViews();
+
+        this.shouldUseDOMSelectionOnFocus = props.shouldUseDOMSelectionOnFocus || false;
     }
     /**
     Holds `true` when a
@@ -5304,3 +5352,6 @@ function checkStateComponent(plugin) {
 }
 
 export { Decoration, DecorationSet, EditorView, __endComposition, __parseFromClipboard, __serializeForClipboard };
+
+// NOTE(calebmer): Exported so we can build our own write to clipboard logic.
+export { serializeForClipboard };
diff --git a/src/domcoords.ts b/src/domcoords.ts
index 4b8e0f3291116e3f1c0e3136a13c0d6626de42f3..3cb6ef9dbc24da546b8c8317c0005a0a11d10deb 100644
--- a/src/domcoords.ts
+++ b/src/domcoords.ts
@@ -35,14 +35,36 @@ export function scrollRectIntoView(view: EditorView, rect: Rect, startDOM: Node)
     let atTop = elt == doc.body
     let bounding = atTop ? windowRect(doc) : clientRect(elt as HTMLElement)
     let moveX = 0, moveY = 0
-    if (rect.top < bounding.top + getSide(scrollThreshold, "top"))
-      moveY = -(bounding.top - rect.top + getSide(scrollMargin, "top"))
-    else if (rect.bottom > bounding.bottom - getSide(scrollThreshold, "bottom"))
-      moveY = rect.bottom - bounding.bottom + getSide(scrollMargin, "bottom")
-    if (rect.left < bounding.left + getSide(scrollThreshold, "left"))
-      moveX = -(bounding.left - rect.left + getSide(scrollMargin, "left"))
-    else if (rect.right > bounding.right - getSide(scrollThreshold, "right"))
-      moveX = rect.right - bounding.right + getSide(scrollMargin, "right")
+    // NOTE(calebmer): Oddly, the implementation of `scrollRectIntoView()` tries to
+    // scroll elements even when they're not scrollable! Check if the element is
+    // scrollable before computing `moveX` and `moveY`. Also bound `moveX` and
+    // `moveY` by the min/max values it can move.
+    //
+    // Admittedly, the implementation was probably fine without putting these bounds
+    // in place but these changes do lead to cleaner scroll event debugger logging.
+    let {overflowX, overflowY} = getComputedStyle(elt)
+    if (overflowY === "auto" || overflowY === "scroll") {
+      if (rect.top < bounding.top + getSide(scrollThreshold, "top"))
+        moveY = -(bounding.top - rect.top + getSide(scrollMargin, "top"))
+      else if (rect.bottom > bounding.bottom - getSide(scrollThreshold, "bottom"))
+        moveY = rect.bottom - bounding.bottom + getSide(scrollMargin, "bottom")
+
+      if (moveY) {
+        moveY = Math.min(moveY, elt.scrollHeight - elt.clientHeight - elt.scrollTop)
+        moveY = Math.max(moveY, -elt.scrollTop)
+      }
+    }
+    if (overflowX === "auto" || overflowX === "scroll") {
+      if (rect.left < bounding.left + getSide(scrollThreshold, "left"))
+        moveX = -(bounding.left - rect.left + getSide(scrollMargin, "left"))
+      else if (rect.right > bounding.right - getSide(scrollThreshold, "right"))
+        moveX = rect.right - bounding.right + getSide(scrollMargin, "right")
+
+      if (moveX) {
+        moveX = Math.min(moveX, elt.scrollWidth - elt.clientWidth - elt.scrollLeft)
+        moveX = Math.max(moveX, -elt.scrollLeft)
+      }
+    }
     if (moveX || moveY) {
       if (atTop) {
         doc.defaultView!.scrollBy(moveX, moveY)
@@ -54,7 +76,7 @@ export function scrollRectIntoView(view: EditorView, rect: Rect, startDOM: Node)
         rect = {left: rect.left - dX, top: rect.top - dY, right: rect.right - dX, bottom: rect.bottom - dY}
       }
     }
-    if (atTop) break
+    if (atTop || /^(fixed|sticky)$/.test(getComputedStyle(parent as HTMLElement).position)) break
   }
 }
 
diff --git a/src/index.ts b/src/index.ts
index f28002967d6ff175430efdd8c2389924211c35ce..e6262cf2d5f958031a1b1ef07ef499410685a99e 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -25,6 +25,9 @@ export const __parseFromClipboard = parseFromClipboard
 /// @internal
 export const __endComposition = endComposition
 
+// NOTE(calebmer): Exported so we can build our own write to clipboard logic.
+export {serializeForClipboard};
+
 /// An editor view manages the DOM structure that represents an
 /// editable document. Its state and behavior are determined by its
 /// [props](#view.DirectEditorProps).
@@ -57,6 +60,8 @@ export class EditorView {
   /// [space is eaten issue](https://github.com/ProseMirror/prosemirror/issues/651)
   /// @internal
   requiresGeckoHackNode: boolean = false
+  /// @internal
+  shouldUseDOMSelectionOnFocus: boolean
 
   /// The view's current [state](#state.EditorState).
   public state: EditorState
@@ -90,6 +95,8 @@ export class EditorView {
     this.domObserver.start()
     initInput(this)
     this.updatePluginViews()
+
+    this.shouldUseDOMSelectionOnFocus = props.shouldUseDOMSelectionOnFocus || false
   }
 
   /// An editable DOM node containing the document. (You probably
@@ -734,4 +741,9 @@ export interface DirectEditorProps extends EditorProps {
   /// [applied](#state.EditorState.apply). The callback will be bound to have
   /// the view instance as its `this` binding.
   dispatchTransaction?: (tr: Transaction) => void
+
+  // NOTE(calebmer): When the user clicks into the editor, provide a
+  // configuration option to use the new selection from the browser instead
+  // of restoring the selection from state.
+  shouldUseDOMSelectionOnFocus?: boolean
 }
diff --git a/src/input.ts b/src/input.ts
index d5d66205262a462451dc19cb9213dab20e2ce02a..6e1ceb7002f6998b83c617a0eb48380383365eab 100644
--- a/src/input.ts
+++ b/src/input.ts
@@ -714,10 +714,34 @@ handlers.focus = view => {
     view.dom.classList.add("ProseMirror-focused")
     view.domObserver.start()
     view.focused = true
-    setTimeout(() => {
-      if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
-        selectionToDOM(view)
-    }, 20)
+
+    if (view.shouldUseDOMSelectionOnFocus) {
+      // NOTE(calebmer): When the user clicks into the editor, provide a
+      // configuration option to use the new selection from the browser instead
+      // of restoring the selection from state.
+      let newSel = selectionFromDOM(view)
+      if (newSel && !view.state.selection.eq(newSel)) {
+        view.dispatch(view.state.tr.setSelection(newSel))
+      }
+    } else {
+      // NOTE(calebmer): When focus is returned to the view, always serialize our
+      // selection from state back to the DOM. Otherwise we're seeing Chrome
+      // sometimes reset the selection.
+      //
+      // To reproduce the bug which caused us to add this in Chrome:
+      //
+      // 1. Highlight some text
+      // 2. Open the link input
+      // 3. Hit escape to close the link input
+      //
+      // Without this, Chrome restores focus to the beginning of the editor.
+      selectionToDOM(view)
+
+      setTimeout(() => {
+        if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
+          selectionToDOM(view)
+      }, 20)
+    }
   }
 }
 
