diff --git a/dist/index.cjs b/dist/index.cjs
index 475bbd8642fc713f7bf3a9e20899f4e1cf923bfa..ce40af2c03a0e9fc7580aff82f90f77504dd5b25 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -36,7 +36,7 @@ var prosemirrorTransform = require('prosemirror-transform');
 
 var classesById = Object.create(null);
 
-var Selection = function () {
+var Selection = (function () {
   function Selection($anchor, $head, ranges) {
     _classCallCheck(this, Selection);
 
@@ -191,7 +191,7 @@ var Selection = function () {
   }]);
 
   return Selection;
-}();
+})();
 
 Selection.prototype.visible = true;
 
@@ -211,7 +211,7 @@ function checkTextSelection($pos) {
   }
 }
 
-var TextSelection = function (_Selection) {
+var TextSelection = (function (_Selection) {
   _inherits(TextSelection, _Selection);
 
   var _super = _createSuper(TextSelection);
@@ -308,11 +308,11 @@ var TextSelection = function (_Selection) {
   }]);
 
   return TextSelection;
-}(Selection);
+})(Selection);
 
 Selection.jsonID("text", TextSelection);
 
-var TextBookmark = function () {
+var TextBookmark = (function () {
   function TextBookmark(anchor, head) {
     _classCallCheck(this, TextBookmark);
 
@@ -333,9 +333,9 @@ var TextBookmark = function () {
   }]);
 
   return TextBookmark;
-}();
+})();
 
-var NodeSelection = function (_Selection2) {
+var NodeSelection = (function (_Selection2) {
   _inherits(NodeSelection, _Selection2);
 
   var _super2 = _createSuper(NodeSelection);
@@ -386,6 +386,35 @@ var NodeSelection = function (_Selection2) {
     value: function getBookmark() {
       return new NodeBookmark(this.anchor);
     }
+  }, {
+    key: "replace",
+    value: function replace(tr) {
+      var content = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : prosemirrorModel.Slice.empty;
+
+      // NOTE(calebmer): When deleting a `file` node that's the only `file` in a
+      // `fileRow` then instead of replacing the `file` with one that has a null `id`
+      // instead delete the whole `fileRow`. We chose to put this here so that every
+      // time `tr.deleteSelection()` is called this logic runs. We could manually
+      // write this logic everywhere `tr.deleteSelection()` is called but sometimes
+      // `tr.deleteSelection()` is called at a framework level in `prosemirror-view`.
+      // Not just in keymap backspace/delete handlers.
+      if (
+        !content.size &&
+        this.node.type.name === "file" &&
+        (this.$anchor.parent.type.name === "fileRow" ||
+          this.$anchor.parent.type.name === "fileFloat" ||
+          this.$anchor.parent.type.name === "fileTable") &&
+        this.$anchor.parent.childCount === 1
+      ) {
+        const mapFrom = tr.steps.length;
+        const mapping = tr.mapping.slice(mapFrom);
+        tr.replaceRange(mapping.map(this.$anchor.before()), mapping.map(this.$anchor.after()), content);
+        selectionToInsertionEnd(tr, mapFrom, 1);
+        return;
+      }
+
+      _get(_getPrototypeOf(NodeSelection.prototype), "replace", this).call(this, tr, content);
+    }
   }], [{
     key: "fromJSON",
     value: function fromJSON(doc, json) {
@@ -405,12 +434,12 @@ var NodeSelection = function (_Selection2) {
   }]);
 
   return NodeSelection;
-}(Selection);
+})(Selection);
 
 NodeSelection.prototype.visible = false;
 Selection.jsonID("node", NodeSelection);
 
-var NodeBookmark = function () {
+var NodeBookmark = (function () {
   function NodeBookmark(anchor) {
     _classCallCheck(this, NodeBookmark);
 
@@ -437,9 +466,9 @@ var NodeBookmark = function () {
   }]);
 
   return NodeBookmark;
-}();
+})();
 
-var AllSelection = function (_Selection3) {
+var AllSelection = (function (_Selection3) {
   _inherits(AllSelection, _Selection3);
 
   var _super3 = _createSuper(AllSelection);
@@ -493,7 +522,7 @@ var AllSelection = function (_Selection3) {
   }]);
 
   return AllSelection;
-}(Selection);
+})(Selection);
 
 Selection.jsonID("all", AllSelection);
 var AllBookmark = {
@@ -542,7 +571,7 @@ var UPDATED_SEL = 1,
     UPDATED_MARKS = 2,
     UPDATED_SCROLL = 4;
 
-var Transaction = function (_prosemirrorTransform) {
+var Transaction = (function (_prosemirrorTransform) {
   _inherits(Transaction, _prosemirrorTransform);
 
   var _super4 = _createSuper(Transaction);
@@ -708,7 +737,7 @@ var Transaction = function (_prosemirrorTransform) {
   }]);
 
   return Transaction;
-}(prosemirrorTransform.Transform);
+})(prosemirrorTransform.Transform);
 
 function bind(f, self) {
   return !self || !f ? f : f.bind(self);
@@ -771,7 +800,7 @@ var Configuration = _createClass(function Configuration(schema, plugins) {
   });
 });
 
-var EditorState = function () {
+var EditorState = (function () {
   function EditorState(config) {
     _classCallCheck(this, EditorState);
 
@@ -959,7 +988,7 @@ var EditorState = function () {
   }]);
 
   return EditorState;
-}();
+})();
 
 function bindProps(obj, self, target) {
   for (var prop in obj) {
@@ -971,7 +1000,7 @@ function bindProps(obj, self, target) {
   return target;
 }
 
-var Plugin = function () {
+var Plugin = (function () {
   function Plugin(spec) {
     _classCallCheck(this, Plugin);
 
@@ -989,7 +1018,7 @@ var Plugin = function () {
   }]);
 
   return Plugin;
-}();
+})();
 
 var keys = Object.create(null);
 
@@ -999,7 +1028,7 @@ function createKey(name) {
   return name + "$";
 }
 
-var PluginKey = function () {
+var PluginKey = (function () {
   function PluginKey() {
     var name = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "key";
 
@@ -1021,7 +1050,7 @@ var PluginKey = function () {
   }]);
 
   return PluginKey;
-}();
+})();
 
 exports.AllSelection = AllSelection;
 exports.EditorState = EditorState;
diff --git a/dist/index.js b/dist/index.js
index 4e31ea63142d7cd3fe570b499c8cb160023531af..bd9a0ec252390f4343119efd38c1384de54d2ad4 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -381,6 +381,29 @@ class NodeSelection extends Selection {
     static isSelectable(node) {
         return !node.isText && node.type.spec.selectable !== false;
     }
+    replace(tr, content = Slice.empty) {
+        // NOTE(calebmer): When deleting a `file` node that's the only `file` in a
+        // `fileRow` then instead of replacing the `file` with one that has a null `id`
+        // instead delete the whole `fileRow`. We chose to put this here so that every
+        // time `tr.deleteSelection()` is called this logic runs. We could manually
+        // write this logic everywhere `tr.deleteSelection()` is called but sometimes
+        // `tr.deleteSelection()` is called at a framework level in `prosemirror-view`.
+        // Not just in keymap backspace/delete handlers.
+        if (
+            !content.size &&
+            this.node.type.name === "file" &&
+            (this.$anchor.parent.type.name === "fileRow" || this.$anchor.parent.type.name === "fileFloat" || this.$anchor.parent.type.name === "fileTable") &&
+            this.$anchor.parent.childCount === 1
+        ) {
+            let mapFrom = tr.steps.length;
+            let mapping = tr.mapping.slice(mapFrom);
+            tr.replaceRange(mapping.map(this.$anchor.before()), mapping.map(this.$anchor.after()), content);
+            selectionToInsertionEnd(tr, mapFrom, 1);
+            return;
+        }
+
+        super.replace(tr, content);
+    }
 }
 NodeSelection.prototype.visible = false;
 Selection.jsonID("node", NodeSelection);
diff --git a/src/selection.ts b/src/selection.ts
index c0ff60847f94c5ac8f6cb654780cf9e81bf372d8..d34159ea239f4fd8c02970066697c4e56c49a239 100644
--- a/src/selection.ts
+++ b/src/selection.ts
@@ -373,6 +373,30 @@ export class NodeSelection extends Selection {
   static isSelectable(node: Node) {
     return !node.isText && node.type.spec.selectable !== false
   }
+
+  replace(tr: Transaction, content = Slice.empty) {
+    // NOTE(calebmer): When deleting a `file` node that's the only `file` in a
+    // `fileRow` then instead of replacing the `file` with one that has a null `id`
+    // instead delete the whole `fileRow`. We chose to put this here so that every
+    // time `tr.deleteSelection()` is called this logic runs. We could manually
+    // write this logic everywhere `tr.deleteSelection()` is called but sometimes
+    // `tr.deleteSelection()` is called at a framework level in `prosemirror-view`.
+    // Not just in keymap backspace/delete handlers.
+    if (
+      !content.size &&
+      this.node.type.name === "file" &&
+      (this.$anchor.parent.type.name === "fileRow" || this.$anchor.parent.type.name === "fileFloat" || this.$anchor.parent.type.name === "fileTable") &&
+      this.$anchor.parent.childCount === 1
+    ) {
+      let mapFrom = tr.steps.length
+      let mapping = tr.mapping.slice(mapFrom)
+      tr.replaceRange(mapping.map(this.$anchor.before()), mapping.map(this.$anchor.after()), content)
+      selectionToInsertionEnd(tr, mapFrom, 1)
+      return
+    }
+
+    super.replace(tr, content)
+  }
 }
 
 NodeSelection.prototype.visible = false
