diff --git a/dist/index.cjs b/dist/index.cjs
index 6f4e0fcfa3f07ec49e8dc392fe9a1d0e5c3adefd..b79f5d52aae28784152efa5e9e3e04bd902102a9 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -3437,8 +3437,30 @@ function defaultTripleClick(view, inside, event) {
   for (var i = $pos.depth + 1; i > 0; i--) {
     var node = i > $pos.depth ? $pos.nodeAfter : $pos.node(i);
     var nodePos = $pos.before(i);
-    if (node.inlineContent) updateSelection(view, prosemirrorState.TextSelection.create(doc, nodePos + 1, nodePos + 1 + node.content.size), "pointer");else if (prosemirrorState.NodeSelection.isSelectable(node)) updateSelection(view, prosemirrorState.NodeSelection.create(doc, nodePos), "pointer");else continue;
-    return true;
+    if (node.inlineContent) {
+      // NOTE(calebmer): By calling `event.preventDefault()` after a triple click it
+      // stops Chrome from allowing a drag selection to continue. To test, try triple
+      // clicking a paragraph in a document then dragging down. In Chrome the
+      // selection won't extend to where you're dragging if `event.preventDefault()`
+      // is called. Given the platform default triple click works well for regular
+      // paragraphs, allow the user to override when the default triple click
+      // behavior kicks in.
+      if (view.someProp("shouldDefaultTripleClickNotPreventDefault", f => f(node))) {
+        return false;
+      } else {
+        updateSelection(view, prosemirrorState.TextSelection.create(doc, nodePos + 1, nodePos + 1 + node.content.size), "pointer");
+        return true;
+      }
+    } else if (prosemirrorState.NodeSelection.isSelectable(node)) {
+      if (view.someProp("shouldDefaultTripleClickNotPreventDefault", f => f(node))) {
+        return false;
+      } else {
+        updateSelection(view, prosemirrorState.NodeSelection.create(doc, nodePos), "pointer");
+        return true;
+      }
+    } else {
+      continue;
+    }
   }
 }
 
@@ -3888,9 +3910,33 @@ handlers.focus = function (view) {
     view.dom.classList.add("ProseMirror-focused");
     view.domObserver.start();
     view.focused = true;
-    setTimeout(function () {
-      if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection())) selectionToDOM(view);
-    }, 20);
+
+    if (view.shouldUseDOMSelectionOnFocus) {
+      // NOTE(calebmer): When the user clicks into the editor, provide a
+      // configuration option to use the new selection from the browser instead
+      // of restoring the selection from state.
+      var newSel = selectionFromDOM(view);
+      if (newSel && !view.state.selection.eq(newSel)) {
+        view.dispatch(view.state.tr.setSelection(newSel));
+      }
+    } else {
+      // NOTE(calebmer): When focus is returned to the view, always serialize our
+      // selection from state back to the DOM. Otherwise we're seeing Chrome
+      // sometimes reset the selection.
+      //
+      // To reproduce the bug which caused us to add this in Chrome:
+      //
+      // 1. Highlight some text
+      // 2. Open the link input
+      // 3. Hit escape to close the link input
+      //
+      // Without this, Chrome restores focus to the beginning of the editor.
+      selectionToDOM(view);
+
+      setTimeout(function () {
+        if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection())) selectionToDOM(view);
+      }, 20);
+    }
   }
 };
 
@@ -5392,6 +5438,8 @@ var EditorView = function () {
     this.domObserver.start();
     initInput(this);
     this.updatePluginViews();
+
+    this.shouldUseDOMSelectionOnFocus = props.shouldUseDOMSelectionOnFocus || false;
   }
 
   _createClass(EditorView, [{
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 52ee8840c754562f9ae4d9891f61682035b092e7..7dd3c99c6d69fd3b54f6be547f8379fe7a415c32 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -638,6 +638,14 @@ interface EditorProps<P = any> {
     Called when the editor is triple-clicked, after `handleTripleClickOn`.
     */
     handleTripleClick?: (this: P, view: EditorView, pos: number, event: MouseEvent) => boolean | void;
+    // NOTE(calebmer): By calling `event.preventDefault()` after a triple click it
+    // stops Chrome from allowing a drag selection to continue. To test, try triple
+    // clicking a paragraph in a document then dragging down. In Chrome the
+    // selection won't extend to where you're dragging if `event.preventDefault()`
+    // is called. Given the platform default triple click works well for regular
+    // paragraphs, allow the user to override when the default triple click
+    // behavior kicks in.
+    shouldDefaultTripleClickNotPreventDefault?: (this: P, node: Node) => boolean | void;
     /**
     Can be used to override the behavior of pasting. `slice` is the
     pasted content parsed by the editor, but you can directly access
@@ -831,6 +839,11 @@ interface DirectEditorProps extends EditorProps {
     the view instance as its `this` binding.
     */
     dispatchTransaction?: (tr: Transaction) => void;
+
+    // NOTE(calebmer): When the user clicks into the editor, provide a
+    // configuration option to use the new selection from the browser instead
+    // of restoring the selection from state.
+    shouldUseDOMSelectionOnFocus?: boolean;
 }
 
 export { DOMEventMap, Decoration, DecorationAttrs, DecorationSet, DecorationSource, DirectEditorProps, EditorProps, EditorView, MarkViewConstructor, NodeView, NodeViewConstructor };
diff --git a/dist/index.js b/dist/index.js
index bb65a77310b75d96a0f548d5fc2e36ff54fc8d3a..57ccce2264259c76cdb714ef77e044fc2428ed8d 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3000,13 +3000,30 @@ function defaultTripleClick(view, inside, event) {
     for (let i = $pos.depth + 1; i > 0; i--) {
         let node = i > $pos.depth ? $pos.nodeAfter : $pos.node(i);
         let nodePos = $pos.before(i);
-        if (node.inlineContent)
-            updateSelection(view, TextSelection.create(doc, nodePos + 1, nodePos + 1 + node.content.size), "pointer");
-        else if (NodeSelection.isSelectable(node))
-            updateSelection(view, NodeSelection.create(doc, nodePos), "pointer");
-        else
+        if (node.inlineContent) {
+            // NOTE(calebmer): By calling `event.preventDefault()` after a triple click it
+            // stops Chrome from allowing a drag selection to continue. To test, try triple
+            // clicking a paragraph in a document then dragging down. In Chrome the
+            // selection won't extend to where you're dragging if `event.preventDefault()`
+            // is called. Given the platform default triple click works well for regular
+            // paragraphs, allow the user to override when the default triple click
+            // behavior kicks in.
+            if (view.someProp("shouldDefaultTripleClickNotPreventDefault", f => f(node))) {
+                return false;
+            } else {
+                updateSelection(view, TextSelection.create(doc, nodePos + 1, nodePos + 1 + node.content.size), "pointer");
+                return true;
+            }
+        } else if (NodeSelection.isSelectable(node)) {
+            if (view.someProp("shouldDefaultTripleClickNotPreventDefault", f => f(node))) {
+                return false;
+            } else {
+                updateSelection(view, NodeSelection.create(doc, nodePos), "pointer");
+                return true;
+            }
+        } else {
             continue;
-        return true;
+        }
     }
 }
 function forceDOMFlush(view) {
@@ -3460,10 +3477,34 @@ handlers.focus = view => {
         view.dom.classList.add("ProseMirror-focused");
         view.domObserver.start();
         view.focused = true;
-        setTimeout(() => {
-            if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
-                selectionToDOM(view);
-        }, 20);
+
+        if (view.shouldUseDOMSelectionOnFocus) {
+            // NOTE(calebmer): When the user clicks into the editor, provide a
+            // configuration option to use the new selection from the browser instead
+            // of restoring the selection from state.
+            let newSel = selectionFromDOM(view);
+            if (newSel && !view.state.selection.eq(newSel)) {
+              view.dispatch(view.state.tr.setSelection(newSel));
+            }
+        } else {      
+            // NOTE(calebmer): When focus is returned to the view, always serialize our
+            // selection from state back to the DOM. Otherwise we're seeing Chrome
+            // sometimes reset the selection.
+            //
+            // To reproduce the bug which caused us to add this in Chrome:
+            //
+            // 1. Highlight some text
+            // 2. Open the link input
+            // 3. Hit escape to close the link input
+            //
+            // Without this, Chrome restores focus to the beginning of the editor.
+            selectionToDOM(view);
+
+            setTimeout(() => {
+                if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
+                    selectionToDOM(view);
+            }, 20);
+        }
     }
 };
 handlers.blur = (view, _event) => {
@@ -4879,6 +4920,8 @@ class EditorView {
         this.domObserver.start();
         initInput(this);
         this.updatePluginViews();
+
+        this.shouldUseDOMSelectionOnFocus = props.shouldUseDOMSelectionOnFocus || false;
     }
     /**
     Holds `true` when a
diff --git a/src/index.ts b/src/index.ts
index f28002967d6ff175430efdd8c2389924211c35ce..fafe2d9ca42f63ba42bad3897c9660c71db13e09 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -57,6 +57,8 @@ export class EditorView {
   /// [space is eaten issue](https://github.com/ProseMirror/prosemirror/issues/651)
   /// @internal
   requiresGeckoHackNode: boolean = false
+  /// @internal
+  shouldUseDOMSelectionOnFocus: boolean
 
   /// The view's current [state](#state.EditorState).
   public state: EditorState
@@ -90,6 +92,8 @@ export class EditorView {
     this.domObserver.start()
     initInput(this)
     this.updatePluginViews()
+
+    this.shouldUseDOMSelectionOnFocus = props.shouldUseDOMSelectionOnFocus || false
   }
 
   /// An editable DOM node containing the document. (You probably
@@ -581,6 +585,15 @@ export interface EditorProps<P = any> {
   /// Called for each node around a triple click.
   handleTripleClickOn?: (this: P, view: EditorView, pos: number, node: Node, nodePos: number, event: MouseEvent, direct: boolean) => boolean | void
 
+  // NOTE(calebmer): By calling `event.preventDefault()` after a triple click it
+  // stops Chrome from allowing a drag selection to continue. To test, try triple
+  // clicking a paragraph in a document then dragging down. In Chrome the
+  // selection won't extend to where you're dragging if `event.preventDefault()`
+  // is called. Given the platform default triple click works well for regular
+  // paragraphs, allow the user to override when the default triple click
+  // behavior kicks in.
+  shouldDefaultTripleClickNotPreventDefault?: (this: P, node: Node) => boolean | void;
+
   /// Called when the editor is triple-clicked, after `handleTripleClickOn`.
   handleTripleClick?: (this: P, view: EditorView, pos: number, event: MouseEvent) => boolean | void
 
@@ -734,4 +747,9 @@ export interface DirectEditorProps extends EditorProps {
   /// [applied](#state.EditorState.apply). The callback will be bound to have
   /// the view instance as its `this` binding.
   dispatchTransaction?: (tr: Transaction) => void
+
+  // NOTE(calebmer): When the user clicks into the editor, provide a
+  // configuration option to use the new selection from the browser instead
+  // of restoring the selection from state.
+  shouldUseDOMSelectionOnFocus?: boolean
 }
diff --git a/src/input.ts b/src/input.ts
index d5d66205262a462451dc19cb9213dab20e2ce02a..5a01593d0a0ec6d4fad0e06e57fd50e16178c060 100644
--- a/src/input.ts
+++ b/src/input.ts
@@ -252,13 +252,30 @@ function defaultTripleClick(view: EditorView, inside: number, event: MouseEvent)
   for (let i = $pos.depth + 1; i > 0; i--) {
     let node = i > $pos.depth ? $pos.nodeAfter! : $pos.node(i)
     let nodePos = $pos.before(i)
-    if (node.inlineContent)
-      updateSelection(view, TextSelection.create(doc, nodePos + 1, nodePos + 1 + node.content.size), "pointer")
-    else if (NodeSelection.isSelectable(node))
-      updateSelection(view, NodeSelection.create(doc, nodePos), "pointer")
-    else
+    if (node.inlineContent) {
+      // NOTE(calebmer): By calling `event.preventDefault()` after a triple click it
+      // stops Chrome from allowing a drag selection to continue. To test, try triple
+      // clicking a paragraph in a document then dragging down. In Chrome the
+      // selection won't extend to where you're dragging if `event.preventDefault()`
+      // is called. Given the platform default triple click works well for regular
+      // paragraphs, allow the user to override when the default triple click
+      // behavior kicks in.
+      if (view.someProp("shouldDefaultTripleClickNotPreventDefault", f => f(node))) {
+        return false;
+      } else {
+        updateSelection(view, TextSelection.create(doc, nodePos + 1, nodePos + 1 + node.content.size), "pointer")
+        return true
+      }
+    } else if (NodeSelection.isSelectable(node)) {
+      if (view.someProp("shouldDefaultTripleClickNotPreventDefault", f => f(node))) {
+        return false;
+      } else {
+        updateSelection(view, NodeSelection.create(doc, nodePos), "pointer")
+        return true
+      }
+    } else {
       continue
-    return true
+    }
   }
 }
 
@@ -714,10 +731,34 @@ handlers.focus = view => {
     view.dom.classList.add("ProseMirror-focused")
     view.domObserver.start()
     view.focused = true
-    setTimeout(() => {
-      if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
-        selectionToDOM(view)
-    }, 20)
+
+    if (view.shouldUseDOMSelectionOnFocus) {
+      // NOTE(calebmer): When the user clicks into the editor, provide a
+      // configuration option to use the new selection from the browser instead
+      // of restoring the selection from state.
+      let newSel = selectionFromDOM(view)
+      if (newSel && !view.state.selection.eq(newSel)) {
+        view.dispatch(view.state.tr.setSelection(newSel))
+      }
+    } else {
+      // NOTE(calebmer): When focus is returned to the view, always serialize our
+      // selection from state back to the DOM. Otherwise we're seeing Chrome
+      // sometimes reset the selection.
+      //
+      // To reproduce the bug which caused us to add this in Chrome:
+      //
+      // 1. Highlight some text
+      // 2. Open the link input
+      // 3. Hit escape to close the link input
+      //
+      // Without this, Chrome restores focus to the beginning of the editor.
+      selectionToDOM(view)
+
+      setTimeout(() => {
+        if (view.docView && view.hasFocus() && !view.domObserver.currentSelection.eq(view.domSelection()))
+          selectionToDOM(view)
+      }, 20)
+    }
   }
 }
 