diff --git a/installer/sources.py b/installer/sources.py
index 84c914f..e81ef8d 100644
--- a/installer/sources.py
+++ b/installer/sources.py
@@ -32,10 +32,37 @@ class WheelSource:
         self.distribution = distribution
         self.version = version

+    # NOTE(calebmer): Patch from:
+    # https://github.com/pypa/installer/pull/137
+    #
+    # Which fixes:
+    # https://github.com/pypa/installer/issues/134
     @property
     def dist_info_dir(self):
         """Name of the dist-info directory."""
-        return f"{self.distribution}-{self.version}.dist-info"
+        if not hasattr(self, "_dist_info_dir"):
+            top_level_directories = {
+                path.split("/", 1)[0] for path in self._zipfile.namelist()
+            }
+            dist_infos = [
+                name for name in top_level_directories if name.endswith(".dist-info")
+            ]
+
+            assert (
+                len(dist_infos) == 1
+            ), "Wheel doesn't contain exactly one .dist-info directory"
+            dist_info_dir = dist_infos[0]
+
+            # NAME-VER.dist-info
+            di_dname = dist_info_dir.rsplit("-", 2)[0]
+            norm_di_dname = installer.utils.normalize_distribution_name(di_dname)
+            norm_file_dname = installer.utils.normalize_distribution_name(self.distribution)
+            assert (
+                norm_di_dname == norm_file_dname
+            ), "Wheel .dist-info directory doesn't match wheel filename"
+
+            self._dist_info_dir = dist_info_dir
+        return self._dist_info_dir

     @property
     def data_dir(self):
diff --git a/installer/utils.py b/installer/utils.py
index 0ccce12..bc22a1d 100644
--- a/installer/utils.py
+++ b/installer/utils.py
@@ -33,6 +33,7 @@ AllSchemes = Tuple[Scheme, ...]

 __all__ = [
     "parse_metadata_file",
+    "normalize_distribution_name",
     "parse_wheel_filename",
     "copyfileobj_with_hashing",
     "get_launcher_kind",
@@ -92,6 +93,13 @@ def parse_metadata_file(contents: str) -> Message:
     return feed_parser.close()


+def normalize_distribution_name(name: str) -> str:
+    """Normalize a project name according to PEP-503.
+    :param name: The project name to normalize
+    """
+    return re.sub(r"[-_.]+", "-", name).lower()
+
+
 def parse_wheel_filename(filename: str) -> WheelFilename:
     """Parse a wheel filename, into it's various components.

