diff --git a/dist/server.d.ts b/dist/server.d.ts
index 1e7ef5f5a015e23cd68415c6579b8a039b354021..9192b0569d19a24d9cf1df7f7cdda1a28ee28d93 100644
--- a/dist/server.d.ts
+++ b/dist/server.d.ts
@@ -18,6 +18,10 @@ export interface StaticRouterProviderProps {
     router: RemixRouter;
     hydrate?: boolean;
     nonce?: string;
+    // NOTE(calebmer): We need nested router support for implementing peeks.
+    // Taking full responsibility for any nested router issues. Acknowledge this
+    // is not officially supported by calling the prop "dangerous".
+    dangerouslyAllowNesting?: boolean;
 }
 /**
  * A Data Router that may not navigate to any other location. This is useful
diff --git a/dist/server.js b/dist/server.js
index c8f4a78fc223125965e19179dde3bc40d46778aa..5ee5317d803e22ee855c195c1304b55636a26870 100644
--- a/dist/server.js
+++ b/dist/server.js
@@ -67,7 +67,8 @@ function StaticRouterProvider({
   context,
   router: router$1,
   hydrate = true,
-  nonce
+  nonce,
+  dangerouslyAllowNesting
 }) {
   !(router$1 && context) ? process.env.NODE_ENV !== "production" ? router.UNSAFE_invariant(false, "You must provide `router` and `context` to <StaticRouterProvider>") : router.UNSAFE_invariant(false) : void 0;
   let dataRouterContext = {
@@ -111,6 +112,7 @@ function StaticRouterProvider({
     navigationType: state.historyAction,
     navigator: dataRouterContext.navigator,
     static: dataRouterContext.static,
+    dangerouslyAllowNesting: dangerouslyAllowNesting,
     future: {
       v7_relativeSplatPath: router$1.future.v7_relativeSplatPath
     }
diff --git a/dist/server.mjs b/dist/server.mjs
index 600b61c55e4eec47629d7cb67d9a37c89d54e207..e3191cbff6793d1b61d8b51d5ed47fc402691d7a 100644
--- a/dist/server.mjs
+++ b/dist/server.mjs
@@ -43,7 +43,8 @@ function StaticRouterProvider({
   context,
   router,
   hydrate = true,
-  nonce
+  nonce,
+  dangerouslyAllowNesting
 }) {
   !(router && context) ? process.env.NODE_ENV !== "production" ? UNSAFE_invariant(false, "You must provide `router` and `context` to <StaticRouterProvider>") : UNSAFE_invariant(false) : void 0;
   let dataRouterContext = {
@@ -87,6 +88,7 @@ function StaticRouterProvider({
     navigationType: state.historyAction,
     navigator: dataRouterContext.navigator,
     static: dataRouterContext.static,
+    dangerouslyAllowNesting: dangerouslyAllowNesting,
     future: {
       v7_relativeSplatPath: router.future.v7_relativeSplatPath
     }
diff --git a/server.d.ts b/server.d.ts
index 1e7ef5f5a015e23cd68415c6579b8a039b354021..9192b0569d19a24d9cf1df7f7cdda1a28ee28d93 100644
--- a/server.d.ts
+++ b/server.d.ts
@@ -18,6 +18,10 @@ export interface StaticRouterProviderProps {
     router: RemixRouter;
     hydrate?: boolean;
     nonce?: string;
+    // NOTE(calebmer): We need nested router support for implementing peeks.
+    // Taking full responsibility for any nested router issues. Acknowledge this
+    // is not officially supported by calling the prop "dangerous".
+    dangerouslyAllowNesting?: boolean;
 }
 /**
  * A Data Router that may not navigate to any other location. This is useful
diff --git a/server.js b/server.js
index c8f4a78fc223125965e19179dde3bc40d46778aa..5ee5317d803e22ee855c195c1304b55636a26870 100644
--- a/server.js
+++ b/server.js
@@ -67,7 +67,8 @@ function StaticRouterProvider({
   context,
   router: router$1,
   hydrate = true,
-  nonce
+  nonce,
+  dangerouslyAllowNesting
 }) {
   !(router$1 && context) ? process.env.NODE_ENV !== "production" ? router.UNSAFE_invariant(false, "You must provide `router` and `context` to <StaticRouterProvider>") : router.UNSAFE_invariant(false) : void 0;
   let dataRouterContext = {
@@ -111,6 +112,7 @@ function StaticRouterProvider({
     navigationType: state.historyAction,
     navigator: dataRouterContext.navigator,
     static: dataRouterContext.static,
+    dangerouslyAllowNesting: dangerouslyAllowNesting,
     future: {
       v7_relativeSplatPath: router$1.future.v7_relativeSplatPath
     }
diff --git a/server.mjs b/server.mjs
index 600b61c55e4eec47629d7cb67d9a37c89d54e207..e3191cbff6793d1b61d8b51d5ed47fc402691d7a 100644
--- a/server.mjs
+++ b/server.mjs
@@ -43,7 +43,8 @@ function StaticRouterProvider({
   context,
   router,
   hydrate = true,
-  nonce
+  nonce,
+  dangerouslyAllowNesting
 }) {
   !(router && context) ? process.env.NODE_ENV !== "production" ? UNSAFE_invariant(false, "You must provide `router` and `context` to <StaticRouterProvider>") : UNSAFE_invariant(false) : void 0;
   let dataRouterContext = {
@@ -87,6 +88,7 @@ function StaticRouterProvider({
     navigationType: state.historyAction,
     navigator: dataRouterContext.navigator,
     static: dataRouterContext.static,
+    dangerouslyAllowNesting: dangerouslyAllowNesting,
     future: {
       v7_relativeSplatPath: router.future.v7_relativeSplatPath
     }
