diff --git a/python/pip_install/extract_wheels/bazel.py b/python/pip_install/extract_wheels/bazel.py
index 013e4a2..2f99536 100644
--- a/python/pip_install/extract_wheels/bazel.py
+++ b/python/pip_install/extract_wheels/bazel.py
@@ -370,7 +370,15 @@ def extract_wheel(
     # Packages may create dependency cycles when specifying optional-dependencies / 'extras'.
     # Example: github.com/google/etils/blob/a0b71032095db14acf6b33516bca6d885fe09e35/pyproject.toml#L32.
     self_edge_dep = set([whl.name])
-    whl_deps = sorted(whl.dependencies(extras_requested) - self_edge_dep)
+    # NOTE(calebmer): `localstack` and `localstack-ext` have a circular
+    # dependency. This breaks Bazel. Arbitrarily break the cycle until the
+    # LocalStack team figures out a better way to manage this.
+    # https://github.com/localstack/localstack/issues/5634
+    #
+    # We patch `rules_python` since the Python rules don't provide a way to
+    # patch individual Python packages directly.
+    localstack_ext_dep = set(["localstack"]) if whl.name == "localstack-ext" else set()
+    whl_deps = sorted(whl.dependencies(extras_requested) - self_edge_dep - localstack_ext_dep)

     if incremental:
         sanitised_dependencies = [
