diff --git a/lib/libvips.js b/lib/libvips.js
index c9927e7a934a660f8543f3e6718bc89a49f32e22..68ee29739e24d65be42352093a4ca6077a5f078c 100644
--- a/lib/libvips.js
+++ b/lib/libvips.js
@@ -123,18 +123,33 @@ const yarnLocator = () => {
 const spawnRebuild = () =>
   spawnSync(`node-gyp rebuild --directory=src ${isEmscripten() ? '--nodedir=emscripten' : ''}`, {
     ...spawnSyncOptions,
-    stdio: 'inherit'
+    stdio: 'inherit',
+    env: {
+      ...process.env,
+      // NOTE(calebmer): Needed to fix "ld: library 'intl' not found" error. `intl`
+      // is provided by `gettext`. So `gettext` needs to be on the library path for
+      // the linker to find it.
+      CPATH: process.env.BAZEL_BINDIR ? `${process.env.JS_BINARY__EXECROOT}/${process.env.BAZEL_BINDIR}/external/gettext/gettext/include` : undefined,
+      LIBRARY_PATH: process.env.BAZEL_BINDIR ? `${process.env.JS_BINARY__EXECROOT}/${process.env.BAZEL_BINDIR}/external/gettext/gettext/lib` : undefined,
+    },
   }).status;
 
 const globalLibvipsVersion = () => {
   if (process.platform !== 'win32') {
-    const globalLibvipsVersion = spawnSync('pkg-config --modversion vips-cpp', {
+    // NOTE(calebmer): We provide a hermetic `pkg-config` tool to `sharp` via the
+    // `PKG_CONFIG` environment variable.
+    const pkgConfig = process.env.PKG_CONFIG || 'pkg-config';
+    const result = spawnSync(pkgConfig + ' --modversion vips-cpp', {
       ...spawnSyncOptions,
       env: {
         ...process.env,
         PKG_CONFIG_PATH: pkgConfigPath()
       }
-    }).stdout;
+    });
+    if (result.status !== 0) {
+      log('pkg-config lookup failed, exit code: ' + result.status + ', stderr: ' + result.stderr.trim());
+    }
+    const globalLibvipsVersion = result.stdout;
     /* istanbul ignore next */
     return (globalLibvipsVersion || '').trim();
   } else {
diff --git a/src/binding.gyp b/src/binding.gyp
index 33607f9aaf8230af07d486480c9db75df99f4317..eec33a573f4731a554cda98b2cb60ca677fccded 100644
--- a/src/binding.gyp
+++ b/src/binding.gyp
@@ -111,8 +111,8 @@
     'conditions': [
       ['use_global_libvips == "true"', {
         # Use pkg-config for include and lib
-        'include_dirs': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --cflags-only-I vips-cpp vips glib-2.0 | sed s\/-I//g)'],
-        'libraries': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --libs vips-cpp)'],
+        'include_dirs': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" $(eval "echo ${PKG_CONFIG:-pkg-config}") --cflags-only-I vips-cpp vips glib-2.0 | sed s\/-I//g)'],
+        'libraries': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" $(eval "echo ${PKG_CONFIG:-pkg-config}") --libs vips-cpp)'],
         'defines': [
           'SHARP_USE_GLOBAL_LIBVIPS'
         ],
@@ -120,7 +120,7 @@
           ['OS == "linux"', {
             'defines': [
               # Inspect libvips-cpp.so to determine which C++11 ABI version was used and set _GLIBCXX_USE_CXX11_ABI accordingly. This is quite horrible.
-              '_GLIBCXX_USE_CXX11_ABI=<!(if readelf -Ws "$(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --variable libdir vips-cpp)/libvips-cpp.so" | c++filt | grep -qF __cxx11;then echo "1";else echo "0";fi)'
+              '_GLIBCXX_USE_CXX11_ABI=<!(if readelf -Ws "$(PKG_CONFIG_PATH="<(pkg_config_path)" $(eval "echo ${PKG_CONFIG:-pkg-config}") --variable libdir vips-cpp)/libvips-cpp.so" | c++filt | grep -qF __cxx11;then echo "1";else echo "0";fi)'
             ]
           }]
         ]
@@ -199,7 +199,7 @@
                 '-sWASM_BIGINT'
               ],
               'libraries': [
-                '<!@(PKG_CONFIG_PATH="<!(node -p "require(\'@img/sharp-libvips-dev-wasm32/lib\')")/pkgconfig" pkg-config --static --libs vips-cpp)'
+                '<!@(PKG_CONFIG_PATH="<!(node -p "require(\'@img/sharp-libvips-dev-wasm32/lib\')")/pkgconfig" $(eval "echo ${PKG_CONFIG:-pkg-config}") --static --libs vips-cpp)'
               ],
             }
           }]
