diff --git a/dist/node/chunks/dep-stQc5rCc.js b/dist/node/chunks/dep-stQc5rCc.js
index 47f3abd8895308e1e7a194f9c192be879d2c2fcf..fdd95d57a9db92e16d7807e7a2954973427f4fee 100644
--- a/dist/node/chunks/dep-stQc5rCc.js
+++ b/dist/node/chunks/dep-stQc5rCc.js
@@ -64482,19 +64482,53 @@ async function _createServer(inlineConfig = {}, options) {
         await handleFileAddUnlink(file, server, isUnlink);
         await onHMRUpdate(file, true);
     };
-    watcher.on('change', async (file) => {
+    const onFileChange = async (file) => {
         file = normalizePath$3(file);
         await container.watchChange(file, { event: 'update' });
         // invalidate module graph cache on file change
         moduleGraph.onFileChange(file);
         await onHMRUpdate(file, false);
+    };
+    let pendingUnlinkTimeoutIdByFile = new Map();
+    watcher.on('change', async (file) => {
+        const pendingUnlinkTimeoutId = pendingUnlinkTimeoutIdByFile.get(file);
+        if (pendingUnlinkTimeoutId !== undefined) {
+            clearTimeout(pendingUnlinkTimeoutId);
+            pendingUnlinkTimeoutIdByFile.delete(file);
+            onFileAddUnlink(file, true);
+        }
+
+        await onFileChange(file);
     });
     getFsUtils(config).initWatcher?.(watcher);
     watcher.on('add', (file) => {
-        onFileAddUnlink(file, false);
+        const pendingUnlinkTimeoutId = pendingUnlinkTimeoutIdByFile.get(file);
+        if (pendingUnlinkTimeoutId !== undefined) {
+            clearTimeout(pendingUnlinkTimeoutId);
+            pendingUnlinkTimeoutIdByFile.delete(file);
+            onFileChange(file);
+        } else {
+            onFileAddUnlink(file, false);
+        }
     });
     watcher.on('unlink', (file) => {
-        onFileAddUnlink(file, true);
+        const pendingUnlinkTimeoutId = pendingUnlinkTimeoutIdByFile.get(file);
+        if (pendingUnlinkTimeoutId !== undefined) {
+            clearTimeout(pendingUnlinkTimeoutId);
+            pendingUnlinkTimeoutIdByFile.delete(file);
+            onFileAddUnlink(file, true);
+        }
+
+        // NOTE(calebmer): To update a file Bazel deletes the file then adds it back.
+        // So when we detect an `unlink` event, wait a bit to see if we get an `add`
+        // event for the same path. If we get an `add` event then instead of Vite
+        // dispatching a delete then add event, we'll have Vite dispatch one
+        // update event.
+        const timeoutId = setTimeout(() => {
+            pendingUnlinkTimeoutIdByFile.delete(file);
+            onFileAddUnlink(file, true);
+        }, 500);
+        pendingUnlinkTimeoutIdByFile.set(file, timeoutId);
     });
     hot.on('vite:invalidate', async ({ path, message }) => {
         const mod = moduleGraph.urlToModuleMap.get(path);
