diff --git a/dist/node/chunks/dep-BzOvws4Y.js b/dist/node/chunks/dep-BzOvws4Y.js
index 184597a2d7b1acc23556e16264e6677457023e81..f2abb9129aa1663c24f5aefc773d481546faaa97 100644
--- a/dist/node/chunks/dep-BzOvws4Y.js
+++ b/dist/node/chunks/dep-BzOvws4Y.js
@@ -49958,14 +49958,53 @@ async function initDevSsrDepsOptimizer(config, server) {
   return await creatingDevSsrOptimizer;
 }
 async function createDepsOptimizer(config, server) {
-  const { logger } = config;
   const ssr = false;
+  const options = getDepOptimizationConfig(config, ssr);
+  // NOTE(calebmer): We prebuild optimized dependencies with Bazel so Vite doesn't
+  // have to build them at runtime. This way optimized dependencies get all the
+  // usual Bazel benefits. Like being able to build them in parallel with other
+  // assets and having the ability to remotely cache optimized dependencies.
+  if (options.hasBazelPrebuild) {
+    const depsCacheDir = getDepsCacheDir(config, ssr);
+    const metadataPath = path$n.join(depsCacheDir, METADATA_FILENAME);
+    const metadata = parseDepsOptimizerMetadata(
+      await fsp.readFile(metadataPath, "utf-8"),
+      depsCacheDir
+    );
+
+    const depsOptimizer = {
+      metadata,
+      registerMissingImport: (id) => {
+        const optimized = metadata.optimized[id];
+        if (optimized) {
+          return optimized;
+        }
+        const chunk = metadata.chunks[id];
+        if (chunk) {
+          return chunk;
+        }
+        throw new Error(`Missing import "${id}" not found in prebuilt optimized dependencies`);
+      },
+      run: () => {
+        // noop...
+      },
+      isOptimizedDepFile: createIsOptimizedDepFile(config),
+      isOptimizedDepUrl: createIsOptimizedDepUrl(config),
+      getOptimizedDepId: (depInfo) => `${depInfo.file}?v=${depInfo.browserHash}`,
+      close: () => {
+        // noop...
+      },
+      options
+    };
+    depsOptimizerMap.set(config, depsOptimizer);
+    return;
+  }
+  const { logger } = config;
   const sessionTimestamp = Date.now().toString();
   const cachedMetadata = await loadCachedDepOptimizationMetadata(config, ssr);
   let debounceProcessingHandle;
   let closed = false;
   let metadata = cachedMetadata || initDepsOptimizerMetadata(config, ssr, sessionTimestamp);
-  const options = getDepOptimizationConfig(config, ssr);
   const { noDiscovery, holdUntilCrawlEnd } = options;
   const depsOptimizer = {
     metadata,
@@ -50415,6 +50454,39 @@ ${e.stack}`),
   }
 }
 async function createDevSsrDepsOptimizer(config) {
+  // NOTE(calebmer): We prebuild optimized dependencies with Bazel so Vite doesn't
+  // have to build them at runtime. This way optimized dependencies get all the
+  // usual Bazel benefits. Like being able to build them in parallel with other
+  // assets and having the ability to remotely cache optimized dependencies.
+  if (config.ssr.optimizeDeps.hasBazelPrebuild) {
+    const depsCacheDir = getDepsCacheDir(config, true);
+    const metadataPath = path$n.join(depsCacheDir, METADATA_FILENAME);
+    const metadata = parseDepsOptimizerMetadata(
+      await fsp.readFile(metadataPath, "utf-8"),
+      depsCacheDir
+    );
+
+    const depsOptimizer = {
+      metadata,
+      isOptimizedDepFile: createIsOptimizedDepFile(config),
+      isOptimizedDepUrl: createIsOptimizedDepUrl(config),
+      getOptimizedDepId: (depInfo) => `${depInfo.file}?v=${depInfo.browserHash}`,
+      registerMissingImport: () => {
+        throw new Error(
+          "Vite Internal Error: registerMissingImport is not supported in dev SSR"
+        );
+      },
+      // noop, there is no scanning during dev SSR
+      // the optimizer blocks the server start
+      run: () => {
+      },
+      close: async () => {
+      },
+      options: config.ssr.optimizeDeps
+    };
+    devSsrDepsOptimizerMap.set(config, depsOptimizer);
+    return;
+  }
   const metadata = await optimizeServerSsrDeps(config);
   const depsOptimizer = {
     metadata,
@@ -51766,6 +51838,14 @@ async function loadAndTransform(id, url, server, options, timestamp, mod, resolv
     }
     if (options.ssr || isFileServingAllowed(file, server)) {
       try {
+        // NOTE(calebmer): While Bazel is actively building it removes and adds files
+        // at will. Vite shouldn't observe this since it may read the file system after
+        // Bazel has removed a file but before Bazel has added it back. So wait for
+        // Bazel builds to finish before reading from the file system.
+        if (server.config.waitForBazelBuild) {
+          await server.config.waitForBazelBuild();
+        }
+
         code = await fsp.readFile(file, "utf-8");
         debugLoad?.(`${timeFrom(loadStart)} [fs] ${prettyUrl}`);
       } catch (e) {
@@ -62687,7 +62767,7 @@ async function _createServer(inlineConfig = {}, options) {
     setClientErrorHandler(httpServer, config.logger);
   }
   const watchEnabled = serverConfig.watch !== null;
-  const watcher = watchEnabled ? chokidar.watch(
+  let watcher = watchEnabled ? chokidar.watch(
     // config file dependencies and env file might be outside of root
     [
       root,
@@ -62699,6 +62779,99 @@ async function _createServer(inlineConfig = {}, options) {
     ],
     resolvedWatchOptions
   ) : createNoopWatcher(resolvedWatchOptions);
+  // NOTE(calebmer): While Bazel is actively building it removes and adds files
+  // at will. Vite shouldn't observe this since it may read the file system after
+  // Bazel has removed a file but before Bazel has added it back. So wait for
+  // Bazel builds to finish before reading from the file system.
+  if (config.waitForBazelBuild) {
+    const originalWatcher = watcher;
+    const emitter = new EventEmitter$4();
+
+    originalWatcher.on("ready", (...args) => {
+      emitter.emit("ready", ...args);
+    });
+
+    originalWatcher.on("raw", (...args) => {
+      emitter.emit("raw", ...args);
+    });
+
+    originalWatcher.on("error", (...args) => {
+      emitter.emit("error", ...args);
+    });
+
+    let pendingEvents = null;
+
+    function preparePendingEvents() {
+      if (pendingEvents !== null) return;
+      pendingEvents = [];
+
+      config.waitForBazelBuild().then(
+        () => {
+          const events = pendingEvents;
+          pendingEvents = null;
+          for (const event of events) {
+            emitter.emit(event[0], ...event[1]);
+            emitter.emit("all", event[0], ...event[1]);
+          }
+        },
+        error => {
+          pendingEvents = null;
+          emitter.emit("error", error);
+        },
+      );
+    }
+
+    originalWatcher.on("add", (...args) => {
+      preparePendingEvents();
+
+      let eventName = "add";
+
+      // If there was an unlink event for the same file then remove the unlink event
+      // and push a `change` event (instead of an `add` event). This way we treat
+      // `unlink`/`add` events while waiting for the Bazel lock as a single
+      // `change` event.
+      for (let i = pendingEvents.length - 1; i >= 0; i--) {
+        const pendingEvent = pendingEvents[i];
+        if (pendingEvent[0] === "unlink" && pendingEvent[1][0] === args[0]) {
+            eventName = "change";
+            pendingEvents.splice(i, 1);
+            break;
+        }
+      }
+
+      pendingEvents.push([eventName, args]);
+    });
+
+    originalWatcher.on("addDir", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["addDir", args]);
+    });
+
+    originalWatcher.on("change", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["change", args]);
+    });
+
+    originalWatcher.on("unlink", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["unlink", args]);
+    });
+
+    originalWatcher.on("unlinkDir", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["unlinkDir", args]);
+    });
+
+    watcher = {
+      add: (...args) => originalWatcher.add(...args),
+      unwatch: (...args) => originalWatcher.unwatch(...args),
+      getWatched: (...args) => originalWatcher.getWatched(...args),
+      close: (...args) => originalWatcher.close(...args),
+      ref: (...args) => originalWatcher.ref(...args),
+      unref: (...args) => originalWatcher.unref(...args),
+      on: (...args) => { emitter.on(...args); return watcher },
+    };
+  }
   const moduleGraph = new ModuleGraph(
     (url, ssr) => container.resolveId(url, void 0, { ssr })
   );
@@ -66618,7 +66791,7 @@ function getDepOptimizationConfig(config, ssr) {
 }
 function isDepsOptimizerEnabled(config, ssr) {
   const optimizeDeps = getDepOptimizationConfig(config, ssr);
-  return !(optimizeDeps.noDiscovery && !optimizeDeps.include?.length);
+  return !(optimizeDeps.noDiscovery && !optimizeDeps.include?.length) || !!optimizeDeps.hasBazelPrebuild;
 }
 function optimizeDepsDisabledBackwardCompatibility(resolved, optimizeDeps, optimizeDepsPath = "") {
   const optimizeDepsDisabled = optimizeDeps.disabled;
@@ -66650,3 +66823,8 @@ function optimizeDepsDisabledBackwardCompatibility(resolved, optimizeDeps, optim
 }
 
 export { colors$1 as A, getDefaultExportFromCjs as B, commonjsGlobal as C, index$1 as D, index as E, build$1 as F, preview$1 as G, arraify as a, build as b, createServer as c, defineConfig as d, preprocessCSS as e, formatPostcssSourceMap as f, buildErrorMessage as g, fetchModule as h, isInNodeModules$1 as i, mergeAlias as j, createFilter as k, loadConfigFromFile as l, mergeConfig as m, normalizePath$3 as n, optimizeDeps as o, preview as p, rollupVersion as q, resolveConfig as r, sortUserPlugins as s, transformWithEsbuild as t, send as u, createLogger as v, searchForWorkspaceRoot as w, isFileServingAllowed as x, loadEnv as y, resolveEnvPrefix as z };
+
+// NOTE(calebmer): We build optimized deps with Bazel instead of Vite so we get
+// Bazel's usual benefits like hermeticity, dependency tracking, and remote
+// caching.
+export { prepareEsbuildOptimizerRun, esbuildOutputFromId, flattenId, needsInterop, stringifyDepsOptimizerMetadata, tryNodeResolve };
diff --git a/dist/node/index.d.ts b/dist/node/index.d.ts
index 8592398c31e06c937468f1458650bf1194e18f8e..d78bf8a08f9eaf50cef16cceb2a975356b37dc02 100644
--- a/dist/node/index.d.ts
+++ b/dist/node/index.d.ts
@@ -2132,7 +2132,7 @@ declare namespace Terser {
     hoist_vars?: boolean
     ie8?: boolean
     if_return?: boolean
-    inline?: boolean | InlineFunctions
+    ?: boolean | InlineFunctions
     join_vars?: boolean
     keep_classnames?: boolean | RegExp
     keep_fargs?: boolean
@@ -2670,6 +2670,11 @@ interface DepOptimizationConfig {
      * @experimental
      */
     holdUntilCrawlEnd?: boolean;
+    // NOTE(calebmer): We prebuild optimized dependencies with Bazel so Vite doesn't
+    // have to build them at runtime. This way optimized dependencies get all the
+    // usual Bazel benefits. Like being able to build them in parallel with other
+    // assets and having the ability to remotely cache optimized dependencies.
+    hasBazelPrebuild?: boolean;
 }
 type DepOptimizationOptions = DepOptimizationConfig & {
     /**
@@ -3411,6 +3416,11 @@ interface ResolvedWorkerOptions {
 interface InlineConfig extends UserConfig {
     configFile?: string | false;
     envFile?: false;
+    // NOTE(calebmer): While Bazel is actively building it removes and adds files
+    // at will. Vite shouldn't observe this since it may read the file system after
+    // Bazel has removed a file but before Bazel has added it back. So wait for
+    // Bazel builds to finish before reading from the file system.
+    waitForBazelBuild?: () => Promise<void>;
 }
 type ResolvedConfig = Readonly<Omit<UserConfig, 'plugins' | 'css' | 'assetsInclude' | 'optimizeDeps' | 'worker' | 'build'> & {
     configFile: string | undefined;
diff --git a/dist/node/index.js b/dist/node/index.js
index c62e12e7f52daea4543114b149807e13b1f6c972..62a3020db3ef805a718eba0a48561d0b9c32db13 100644
--- a/dist/node/index.js
+++ b/dist/node/index.js
@@ -258,3 +258,8 @@ async function createViteRuntime(server, options = {}) {
 }
 
 export { ServerHMRConnector, createViteRuntime, isCSSRequest, splitVendorChunk, splitVendorChunkPlugin };
+
+// NOTE(calebmer): We build optimized deps with Bazel instead of Vite so we get
+// Bazel's usual benefits like hermeticity, dependency tracking, and remote
+// caching.
+export { prepareEsbuildOptimizerRun, esbuildOutputFromId, flattenId, needsInterop, stringifyDepsOptimizerMetadata, tryNodeResolve } from './chunks/dep-BzOvws4Y.js';
