diff --git a/dist/node/chunks/dep-BzOvws4Y.js b/dist/node/chunks/dep-BzOvws4Y.js
index 184597a2d7b1acc23556e16264e6677457023e81..de45580787220f3d5708ef3619f5740e07fdf85e 100644
--- a/dist/node/chunks/dep-BzOvws4Y.js
+++ b/dist/node/chunks/dep-BzOvws4Y.js
@@ -62675,6 +62675,23 @@ async function _createServer(inlineConfig = {}, options) {
     emptyOutDir
   );
   const middlewares = connect$1();
+  // NOTE(calebmer): Acquire a Bazel lock before processing a request. This way
+  // we won't read files during a build when files could be temporarily removed.
+  if (inlineConfig.acquireBazelLock) {
+    middlewares.use((req, res, next) => {
+      inlineConfig.acquireBazelLock().then(
+        release => {
+          const end = res.end;
+          res.end = (...args) => {
+            release();
+            return end.call(res, ...args);
+          };
+          next();
+        },
+        next,
+      );
+    });
+  }
   const httpServer = middlewareMode ? null : await resolveHttpServer(serverConfig, middlewares, httpsOptions);
   const ws = createWebSocketServer(httpServer, config, httpsOptions);
   const hot = createHMRBroadcaster().addChannel(ws).addChannel(createServerHMRChannel());
@@ -62687,7 +62704,7 @@ async function _createServer(inlineConfig = {}, options) {
     setClientErrorHandler(httpServer, config.logger);
   }
   const watchEnabled = serverConfig.watch !== null;
-  const watcher = watchEnabled ? chokidar.watch(
+  let watcher = watchEnabled ? chokidar.watch(
     // config file dependencies and env file might be outside of root
     [
       root,
@@ -62699,6 +62716,102 @@ async function _createServer(inlineConfig = {}, options) {
     ],
     resolvedWatchOptions
   ) : createNoopWatcher(resolvedWatchOptions);
+  // NOTE(calebmer): If we have the acquire Bazel lock function then every time
+  // we get a file watcher event, wait for the Bazel lock before letting Vite
+  // process the event. Because if Bazel is currently building it will be
+  // removing, updating, adding many files. We want to process all these updates
+  // atomically at once instead of processing them piecemeal during the build
+  // when Bazel's output directory isn't in a stable state.
+  if (inlineConfig.acquireBazelLock) {
+    const originalWatcher = watcher;
+    const emitter = new EventEmitter$4();
+
+    originalWatcher.on("ready", (...args) => {
+      emitter.emit("ready", ...args);
+    });
+
+    originalWatcher.on("raw", (...args) => {
+      emitter.emit("raw", ...args);
+    });
+
+    originalWatcher.on("error", (...args) => {
+      emitter.emit("error", ...args);
+    });
+
+    let pendingEvents = null;
+
+    function preparePendingEvents() {
+      if (pendingEvents === null) {
+        pendingEvents = [];
+
+        inlineConfig.acquireBazelLock().then(
+          release => {
+            const events = pendingEvents;
+            pendingEvents = null;
+            for (const event of events) {
+              emitter.emit(event[0], ...event[1]);
+              emitter.emit("all", event[0], ...event[1]);
+            }
+            release();
+          },
+          error => {
+            pendingEvents = null;
+            emitter.emit("error", error);
+          },
+        );
+      }
+    }
+
+    originalWatcher.on("add", (...args) => {
+      preparePendingEvents();
+
+      let eventName = "add";
+
+      // If there was an unlink event for the same file then remove the unlink event and push a
+      // `change` event (instead of an `add` event). This way we treat `unlink`/`add` events
+      // while waiting for the Bazel lock as a single `change` event.
+      for (let i = pendingEvents.length - 1; i >= 0; i--) {
+        const pendingEvent = pendingEvents[i];
+        if (pendingEvent[0] === "unlink" && pendingEvent[1][0] === args[0]) {
+            eventName = "change";
+            pendingEvents.splice(i, 1);
+            break;
+        }
+      }
+
+      pendingEvents.push([eventName, args]);
+    });
+
+    originalWatcher.on("addDir", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["addDir", args]);
+    });
+
+    originalWatcher.on("change", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["change", args]);
+    });
+
+    originalWatcher.on("unlink", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["unlink", args]);
+    });
+
+    originalWatcher.on("unlinkDir", (...args) => {
+      preparePendingEvents();
+      pendingEvents.push(["unlinkDir", args]);
+    });
+
+    watcher = {
+      add: (...args) => originalWatcher.add(...args),
+      unwatch: (...args) => originalWatcher.unwatch(...args),
+      getWatched: (...args) => originalWatcher.getWatched(...args),
+      close: (...args) => originalWatcher.close(...args),
+      ref: (...args) => originalWatcher.ref(...args),
+      unref: (...args) => originalWatcher.unref(...args),
+      on: (...args) => { emitter.on(...args); return watcher },
+    };
+  }
   const moduleGraph = new ModuleGraph(
     (url, ssr) => container.resolveId(url, void 0, { ssr })
   );
diff --git a/dist/node/index.d.ts b/dist/node/index.d.ts
index 8592398c31e06c937468f1458650bf1194e18f8e..4ffaf2d08d9b079b6fa385571cda5dc7c1c997ea 100644
--- a/dist/node/index.d.ts
+++ b/dist/node/index.d.ts
@@ -3411,6 +3411,12 @@ interface ResolvedWorkerOptions {
 interface InlineConfig extends UserConfig {
     configFile?: string | false;
     envFile?: false;
+    // NOTE(calebmer): Vite should request a Bazel lock before doing any file
+    // system processing. That way we can make sure the file system is in a good
+    // state. While Bazel is building it will temporarily remove files, if Vite
+    // observes this it will show a "file not found" error which is confusing
+    // and wrong.
+    acquireBazelLock?: () => Promise<() => void>;
 }
 type ResolvedConfig = Readonly<Omit<UserConfig, 'plugins' | 'css' | 'assetsInclude' | 'optimizeDeps' | 'worker' | 'build'> & {
     configFile: string | undefined;
