diff --git a/wrangler-dist/cli.js b/wrangler-dist/cli.js
index ab3a304d1718640e830015b0a2f62268bb80a815..143582c4ab5a6c318a0174f0b6b096c48cd1f73f 100644
--- a/wrangler-dist/cli.js
+++ b/wrangler-dist/cli.js
@@ -139838,6 +139838,16 @@ async function forEachFile(baseDir, fn2) {
       } else if (entry.isFile()) {
         returnValues.push(await fn2(pathname));
       }
+      // NOTE(calebmer): Bazel creates a lot of symlinks. Make sure this
+      // function supports traversing symlinks.
+      else if (entry.isSymbolicLink()) {
+        const stats = await import_promises10.default.stat(pathname);
+        if (stats.isDirectory()) {
+          searchPaths.push(pathname);
+        } else if (stats.isFile()) {
+          returnValues.push(await fn2(pathname));
+        }
+      }
     }
   }
   return returnValues;