diff --git a/wrangler-dist/cli.js b/wrangler-dist/cli.js
index 717ca2fb4e02822f8bda411d9070974df2dcdd7e..ddaee0323b413dbd077dd50dbb9fd7d7ac883d5c 100644
--- a/wrangler-dist/cli.js
+++ b/wrangler-dist/cli.js
@@ -211543,6 +211543,13 @@ async function forEachFile(baseDir$1, fn3) {
       const pathname = path31__namespace.default.join(cwd2, entry.name);
       if (entry.isDirectory()) searchPaths.push(pathname);
       else if (entry.isFile()) returnValues.push(await fn3(pathname));
+      // NOTE(calebmer): Bazel creates a lot of symlinks. Make sure this
+      // function supports traversing symlinks.
+      else if (entry.isSymbolicLink()) {
+        const stats = await fs$1__namespace.default.stat(pathname);
+        if (stats.isDirectory()) searchPaths.push(pathname);
+        else if (stats.isFile()) returnValues.push(await fn3(pathname));
+      }
     }
   }
   return returnValues;
