diff --git a/src/sites.ts b/src/sites.ts
index 0ec3b2fb0804ff1f05f2b33f5b74c7d07366df15..2ba88079266b188a9db40c2b9ea65e3443c570e3 100644
--- a/src/sites.ts
+++ b/src/sites.ts
@@ -32,11 +32,12 @@ async function* getFilesInFolder(dirPath: string): AsyncIterable<string> {
 		if (file.name.startsWith(".") && !HIDDEN_FILES_TO_INCLUDE.has(file.name)) {
 			continue;
 		}
-		// TODO: follow symlinks??
-		if (file.isDirectory()) {
-			yield* await getFilesInFolder(path.join(dirPath, file.name));
+		// NOTE(calebmer): Support directory symlinks.
+		const filePath = path.join(dirPath, file.name);
+		if (file.isDirectory() || (file.isSymbolicLink() && (await stat(filePath)).isDirectory())) {
+			yield* await getFilesInFolder(filePath);
 		} else {
-			yield path.join(dirPath, file.name);
+			yield filePath;
 		}
 	}
 }
diff --git a/wrangler-dist/cli.js b/wrangler-dist/cli.js
index 998e045998ce7202e46ae03fa4999da7b57eb5a1..c1a09e05ce837f1c46e715691f41a51cf659900e 100644
--- a/wrangler-dist/cli.js
+++ b/wrangler-dist/cli.js
@@ -146255,10 +146255,12 @@ async function* getFilesInFolder(dirPath) {
     if (file.name.startsWith(".") && !HIDDEN_FILES_TO_INCLUDE.has(file.name)) {
       continue;
     }
-    if (file.isDirectory()) {
-      yield* await getFilesInFolder(path32.join(dirPath, file.name));
+    // NOTE(calebmer): Support directory symlinks.
+    const filePath = path32.join(dirPath, file.name);
+    if (file.isDirectory() || (file.isSymbolicLink() && (await (0, import_promises12.stat)(filePath)).isDirectory())) {
+      yield* await getFilesInFolder(filePath);
     } else {
-      yield path32.join(dirPath, file.name);
+      yield filePath;
     }
   }
 }