diff --git a/dist/y-prosemirror.cjs b/dist/y-prosemirror.cjs
index 52a54c8511a726df3e1a94861939f0fb75418afb..15d6db2f05f414fe92c894cdb4489540cffa1db0 100644
--- a/dist/y-prosemirror.cjs
+++ b/dist/y-prosemirror.cjs
@@ -128,6 +128,8 @@ const getUserColor = (colorMapping, colors, user) => {
   return /** @type {ColorDef} */ (colorMapping.get(user))
 };
 
+let scheduledRerenderCallbacks = null;
+
 /**
  * This plugin listens to changes in prosemirror view and keeps yXmlState and view in sync.
  *
@@ -143,7 +145,7 @@ const ySyncPlugin = (yXmlFragment, {
   onFirstRender = () => {}
 } = {}) => {
   let changedInitialContent = false;
-  let rerenderTimeout;
+  let hasScheduledRerender = false;
   const plugin = new prosemirrorState.Plugin({
     props: {
       editable: (state) => {
@@ -226,15 +228,36 @@ const ySyncPlugin = (yXmlFragment, {
     },
     view: (view) => {
       const binding = new ProsemirrorBinding(yXmlFragment, view);
-      if (rerenderTimeout != null) {
-        rerenderTimeout.destroy();
+      // NOTE(calebmer): While profiling task grid view scrolling, I found
+      // creating a timeout for each individual editor ends up taking a
+      // non-trivial amount of time (6% of time in a render in one test).
+      // So batch rerenders together in a single timeout call.
+      if (!hasScheduledRerender) {
+        hasScheduledRerender = true;
+        
+        if (!scheduledRerenderCallbacks) {
+          scheduledRerenderCallbacks = [];
+
+          eventloop__namespace.timeout(0, () => {
+            const callbacks = scheduledRerenderCallbacks
+            scheduledRerenderCallbacks = null
+
+            for (const callback of callbacks) {
+              callback();
+            }
+          });
+        }
+
+        scheduledRerenderCallbacks.push(() => {
+          if (!hasScheduledRerender) return;
+          hasScheduledRerender = false;
+
+          // Make sure this is called in a separate context
+          binding._forceRerender();
+          view.dispatch(view.state.tr.setMeta(ySyncPluginKey, { binding }));
+          onFirstRender();
+        });
       }
-      // Make sure this is called in a separate context
-      rerenderTimeout = eventloop__namespace.timeout(0, () => {
-        binding._forceRerender();
-        view.dispatch(view.state.tr.setMeta(ySyncPluginKey, { binding }));
-        onFirstRender();
-      });
       return {
         update: () => {
           const pluginState = plugin.getState(view.state);
@@ -271,7 +294,7 @@ const ySyncPlugin = (yXmlFragment, {
           }
         },
         destroy: () => {
-          rerenderTimeout.destroy();
+          hasScheduledRerender = false;
           binding.destroy();
         }
       }
@@ -451,13 +474,35 @@ class ProsemirrorBinding {
   _forceRerender () {
     this.mapping = new Map();
     this.mux(() => {
-      const fragmentContent = this.type.toArray().map((t) =>
-        createNodeFromYElement(
-          /** @type {Y.XmlElement} */ (t),
-          this.prosemirrorView.state.schema,
-          this.mapping
-        )
-      ).filter((n) => n !== null);
+      // NOTE(calebmer): ProseMirror can also have inline text content at the
+      // root. Not just elements. This code is copied from the implementation
+      // of `createNodeFromYElement()`.
+      const fragmentContent = []
+      this.type.toArray().forEach((type) => {
+        if (type.constructor === Y.XmlElement) {
+          const n = createNodeFromYElement(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (n !== null) {
+            fragmentContent.push(n)
+          }
+        } else {
+          const ns = createTextNodesFromYText(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (ns !== null) {
+            ns.forEach((textchild) => {
+              if (textchild !== null) {
+                fragmentContent.push(textchild)
+              }
+            })
+          }    
+        }
+      })
       // @ts-ignore
       const tr = this._tr.replace(
         0,
@@ -577,13 +622,35 @@ class ProsemirrorBinding {
       );
       transaction.changed.forEach(delType);
       transaction.changedParentTypes.forEach(delType);
-      const fragmentContent = this.type.toArray().map((t) =>
-        createNodeIfNotExists(
-          /** @type {Y.XmlElement | Y.XmlHook} */ (t),
-          this.prosemirrorView.state.schema,
-          this.mapping
-        )
-      ).filter((n) => n !== null);
+      // NOTE(calebmer): ProseMirror can also have inline text content at the
+      // root. Not just elements. This code is copied from the implementation
+      // of `createNodeFromYElement()`.
+      const fragmentContent = []
+      this.type.toArray().forEach((type) => {
+        if (type.constructor === Y.XmlElement) {
+          const n = createNodeIfNotExists(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (n !== null) {
+            fragmentContent.push(n)
+          }
+        } else {
+          const ns = createTextNodesFromYText(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (ns !== null) {
+            ns.forEach((textchild) => {
+              if (textchild !== null) {
+                fragmentContent.push(textchild)
+              }
+            })
+          }    
+        }
+      })
       // @ts-ignore
       let tr = this._tr.replace(
         0,
@@ -1553,6 +1620,8 @@ function yXmlFragmentToProsemirrorJSON (xmlFragment) {
 
   return {
     type: 'doc',
+    // NOTE(calebmer): We need a `flat()` call here for inline content at the
+    // top level just like we need a flat on line 1547.
     content: items.map(serialize)
   }
 }
diff --git a/package.json b/package.json
index 6767536abd7947ea0e02c8f8ecc9ec190d362123..044c83561cbdc0e5bc2bdde22dc462d6f1db6f63 100644
--- a/package.json
+++ b/package.json
@@ -23,7 +23,8 @@
   "exports": {
     ".": {
       "import": "./src/y-prosemirror.js",
-      "require": "./dist/y-prosemirror.cjs"
+      "require": "./dist/y-prosemirror.cjs",
+      "types": "./dist/src/y-prosemirror.d.ts"
     }
   },
   "files": [
diff --git a/src/lib.js b/src/lib.js
index 964f8affdad7519e6681064fbcc42d804de31317..fb5bd7457374579e7f67431f3385b26942745236 100644
--- a/src/lib.js
+++ b/src/lib.js
@@ -372,6 +372,8 @@ export function yXmlFragmentToProsemirrorJSON (xmlFragment) {
 
   return {
     type: 'doc',
-    content: items.map(serialize)
+    // NOTE(calebmer): We need a `flat()` call here for inline content at the
+    // top level just like we need a flat on line 366.
+    content: items.map(serialize).flat()
   }
 }
diff --git a/src/plugins/sync-plugin.js b/src/plugins/sync-plugin.js
index e4d5163b4932228fae5ede4e40fd7671daa54aca..98f35b60d634281117d176de453a199ecabc45df 100644
--- a/src/plugins/sync-plugin.js
+++ b/src/plugins/sync-plugin.js
@@ -75,6 +75,8 @@ const getUserColor = (colorMapping, colors, user) => {
   return /** @type {ColorDef} */ (colorMapping.get(user))
 }
 
+let scheduledRerenderCallbacks = null
+
 /**
  * This plugin listens to changes in prosemirror view and keeps yXmlState and view in sync.
  *
@@ -90,7 +92,7 @@ export const ySyncPlugin = (yXmlFragment, {
   onFirstRender = () => {}
 } = {}) => {
   let changedInitialContent = false
-  let rerenderTimeout
+  let hasScheduledRerender = false
   const plugin = new Plugin({
     props: {
       editable: (state) => {
@@ -173,15 +175,36 @@ export const ySyncPlugin = (yXmlFragment, {
     },
     view: (view) => {
       const binding = new ProsemirrorBinding(yXmlFragment, view)
-      if (rerenderTimeout != null) {
-        rerenderTimeout.destroy()
+      // NOTE(calebmer): While profiling task grid view scrolling, I found
+      // creating a timeout for each individual editor ends up taking a
+      // non-trivial amount of time (6% of time in a render in one test).
+      // So batch rerenders together in a single timeout call.
+      if (!hasScheduledRerender) {
+        hasScheduledRerender = true
+        
+        if (!scheduledRerenderCallbacks) {
+          scheduledRerenderCallbacks = []
+
+          eventloop.timeout(0, () => {
+            const callbacks = scheduledRerenderCallbacks
+            scheduledRerenderCallbacks = null
+
+            for (const callback of callbacks) {
+              callback()
+            }
+          })
+        }
+
+        scheduledRerenderCallbacks.push(() => {
+          if (!hasScheduledRerender) return
+          hasScheduledRerender = false
+
+          // Make sure this is called in a separate context
+          binding._forceRerender()
+          view.dispatch(view.state.tr.setMeta(ySyncPluginKey, { binding }))
+          onFirstRender()
+        })
       }
-      // Make sure this is called in a separate context
-      rerenderTimeout = eventloop.timeout(0, () => {
-        binding._forceRerender()
-        view.dispatch(view.state.tr.setMeta(ySyncPluginKey, { binding }))
-        onFirstRender()
-      })
       return {
         update: () => {
           const pluginState = plugin.getState(view.state)
@@ -218,7 +241,7 @@ export const ySyncPlugin = (yXmlFragment, {
           }
         },
         destroy: () => {
-          rerenderTimeout.destroy()
+          hasScheduledRerender = false
           binding.destroy()
         }
       }
@@ -398,13 +421,35 @@ export class ProsemirrorBinding {
   _forceRerender () {
     this.mapping = new Map()
     this.mux(() => {
-      const fragmentContent = this.type.toArray().map((t) =>
-        createNodeFromYElement(
-          /** @type {Y.XmlElement} */ (t),
-          this.prosemirrorView.state.schema,
-          this.mapping
-        )
-      ).filter((n) => n !== null)
+      // NOTE(calebmer): ProseMirror can also have inline text content at the
+      // root. Not just elements. This code is copied from the implementation
+      // of `createNodeFromYElement()`.
+      const fragmentContent = []
+      this.type.toArray().forEach((type) => {
+        if (type.constructor === Y.XmlElement) {
+          const n = createNodeFromYElement(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (n !== null) {
+            fragmentContent.push(n)
+          }
+        } else {
+          const ns = createTextNodesFromYText(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (ns !== null) {
+            ns.forEach((textchild) => {
+              if (textchild !== null) {
+                fragmentContent.push(textchild)
+              }
+            })
+          }    
+        }
+      })
       // @ts-ignore
       const tr = this._tr.replace(
         0,
@@ -524,13 +569,35 @@ export class ProsemirrorBinding {
       )
       transaction.changed.forEach(delType)
       transaction.changedParentTypes.forEach(delType)
-      const fragmentContent = this.type.toArray().map((t) =>
-        createNodeIfNotExists(
-          /** @type {Y.XmlElement | Y.XmlHook} */ (t),
-          this.prosemirrorView.state.schema,
-          this.mapping
-        )
-      ).filter((n) => n !== null)
+      // NOTE(calebmer): ProseMirror can also have inline text content at the
+      // root. Not just elements. This code is copied from the implementation
+      // of `createNodeFromYElement()`.
+      const fragmentContent = []
+      this.type.toArray().forEach((type) => {
+        if (type.constructor === Y.XmlElement) {
+          const n = createNodeIfNotExists(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (n !== null) {
+            fragmentContent.push(n)
+          }
+        } else {
+          const ns = createTextNodesFromYText(
+            type,
+            this.prosemirrorView.state.schema,
+            this.mapping
+          )
+          if (ns !== null) {
+            ns.forEach((textchild) => {
+              if (textchild !== null) {
+                fragmentContent.push(textchild)
+              }
+            })
+          }    
+        }
+      })
       // @ts-ignore
       let tr = this._tr.replace(
         0,