diff --git a/dist/y-prosemirror.cjs b/dist/y-prosemirror.cjs
index 52a54c8511a726df3e1a94861939f0fb75418afb..e795f82ac696cabbea756d44acec3e5ca7731b69 100644
--- a/dist/y-prosemirror.cjs
+++ b/dist/y-prosemirror.cjs
@@ -1553,7 +1553,9 @@ function yXmlFragmentToProsemirrorJSON (xmlFragment) {
 
   return {
     type: 'doc',
-    content: items.map(serialize)
+    // NOTE(calebmer): We need a `flat()` call here for inline content at the
+    // top level just like we need a flat on line 1547.
+    content: items.map(serialize).flat()
   }
 }
 
diff --git a/package.json b/package.json
index 6767536abd7947ea0e02c8f8ecc9ec190d362123..044c83561cbdc0e5bc2bdde22dc462d6f1db6f63 100644
--- a/package.json
+++ b/package.json
@@ -23,7 +23,8 @@
   "exports": {
     ".": {
       "import": "./src/y-prosemirror.js",
-      "require": "./dist/y-prosemirror.cjs"
+      "require": "./dist/y-prosemirror.cjs",
+      "types": "./dist/src/y-prosemirror.d.ts"
     }
   },
   "files": [
diff --git a/src/lib.js b/src/lib.js
index 964f8affdad7519e6681064fbcc42d804de31317..c8d7919cc5e58166a645a712bf5b08dce43f2105 100644
--- a/src/lib.js
+++ b/src/lib.js
@@ -372,6 +372,8 @@ export function yXmlFragmentToProsemirrorJSON (xmlFragment) {
 
   return {
     type: 'doc',
-    content: items.map(serialize)
+    // NOTE(calebmer): We need a `flat()` call here for inline content at the
+    // top level just like we need a flat on line 1547.
+    content: items.map(serialize).flat()
   }
 }
