diff --git a/dist/config/routesConvention.js b/dist/config/routesConvention.js
index 94179914f536e0fcd62a6d99996379e9db8eb98b..0c501f2ef0a2638e4e6887f3a1eaa6b8875bc303 100644
--- a/dist/config/routesConvention.js
+++ b/dist/config/routesConvention.js
@@ -60,17 +60,37 @@ function isRouteModuleFile(filename) {
 function defineConventionalRoutes(appDir, ignoredFilePatterns) {
   let files = {}; // First, find all route modules in app/routes
 
-  visitFiles(path__namespace.join(appDir, "routes"), file => {
+  visitFiles(path__namespace.join(appDir, "routes"), (file, otherFilenames) => {
     if (ignoredFilePatterns && ignoredFilePatterns.some(pattern => minimatch__default["default"](file, pattern))) {
       return;
     }
 
+    // NOTE(calebmer): When executing in a Bazel output directory we may have
+    // `.js` and `.js.map` files next to `.tsx` files. In that case we want to
+    // ignore both the `.js` and `.js.map` files.
+    //
+    // If we see a `.js` file we check if there is a higher priority extension
+    // in the directory. If there is we skip the file. If we see a `.js.map`
+    // file we check that the file path without `.map` is a valid route file.
     if (isRouteModuleFile(file)) {
+      const filename = path__namespace.basename(file);
+      const ext = path__namespace.extname(file);
+      const extPriority = routeModuleExts.indexOf(ext);
+      for (let priority = 0; priority < extPriority; priority++) {
+        if (otherFilenames.includes(filename.slice(0, -ext.length) + routeModuleExts[priority])) {
+          return;
+        }
+      }
+
       let routeId = routes.createRouteId(path__namespace.join("routes", file));
       files[routeId] = path__namespace.join("routes", file);
       return;
     }
 
+    if (file.endsWith('.map') && isRouteModuleFile(file.slice(0, -4))) {
+      return;
+    }
+
     throw new Error(`Invalid route module file: ${path__namespace.join(appDir, "routes", file)}`);
   });
   let routeIds = Object.keys(files).sort(byLongestFirst);
@@ -204,14 +224,15 @@ function byLongestFirst(a, b) {
 }
 
 function visitFiles(dir, visitor, baseDir = dir) {
-  for (let filename of fs__namespace.readdirSync(dir)) {
+  let filenames = fs__namespace.readdirSync(dir);
+  for (let filename of filenames) {
     let file = path__namespace.resolve(dir, filename);
     let stat = fs__namespace.lstatSync(file);
 
     if (stat.isDirectory()) {
       visitFiles(file, visitor, baseDir);
     } else if (stat.isFile()) {
-      visitor(path__namespace.relative(baseDir, file));
+      visitor(path__namespace.relative(baseDir, file), filenames);
     }
   }
 }