diff --git a/ts/private/ts_project.bzl b/ts/private/ts_project.bzl
index 367bba0..4fdc98c 100644
--- a/ts/private/ts_project.bzl
+++ b/ts/private/ts_project.bzl
@@ -366,6 +366,10 @@ See https://github.com/aspect-build/rules_ts/issues/361 for more details.
             # https://bazel.build/extending/rules#validations_output_group
             # "hold the otherwise unused outputs of validation actions"
             _validation = validation_outs,
+            # NOTE(calebmer): We need access to the source files of
+            # `ts_project()` rules to run a repository wide TypeScript
+            # check test.
+            srcs = srcs_inputs,
         ),
         coverage_common.instrumented_files_info(
             ctx,
