diff --git a/esbuild/private/esbuild.bzl b/esbuild/private/esbuild.bzl
index 10d6616..32db947 100644
--- a/esbuild/private/esbuild.bzl
+++ b/esbuild/private/esbuild.bzl
@@ -209,6 +209,11 @@ See https://esbuild.github.io/api/#target for more details
         for more information.
         """,
     ),
+    # NOTE(calebmer): Added so we can configure the environment when esbuild runs.
+    # Notably we want to be able to set `NODE_ENV`.
+    "env": attr.string_dict(
+        doc = """Environment variables of the action.""",
+    ),
 }
 
 def _bin_relative_path(ctx, file):
@@ -308,7 +313,7 @@ def _esbuild_impl(ctx):
 
         args.update({"outfile": _bin_relative_path(ctx, js_out)})
 
-    env = {
+    env = ctx.attr.env | {
         "BAZEL_BINDIR": ctx.bin_dir.path,
         "ESBUILD_BINARY_PATH": "../../../" + esbuild_toolinfo.target_tool_path,
     }
diff --git a/esbuild/private/plugins/bazel-sandbox.js b/esbuild/private/plugins/bazel-sandbox.js
index c58b668..f61cc2c 100644
--- a/esbuild/private/plugins/bazel-sandbox.js
+++ b/esbuild/private/plugins/bazel-sandbox.js
@@ -1,5 +1,6 @@
 const path = require('path')
 const process = require('process')
+const fs = require('fs/promises')
 
 const bindir = process.env.BAZEL_BINDIR
 const execroot = process.env.JS_BINARY__EXECROOT
@@ -12,7 +13,7 @@ function bazelSandboxPlugin() {
     name: 'bazel-sandbox',
     setup(build) {
       build.onResolve(
-        { filter: /./ },
+        { filter: /./, namespace: 'file' },
         async ({ path: importPath, ...otherOptions }) => {
           // NB: these lines are to prevent infinite recursion when we call `build.resolve`.
           if (otherOptions.pluginData) {
@@ -32,7 +33,30 @@ function bazelSandboxPlugin() {
 }
 
 async function resolveInExecroot(build, importPath, otherOptions) {
-  const result = await build.resolve(importPath, otherOptions)
+  let result;
+
+  // NOTE(calebmer): Relative imports are easy to resolve ourselves without
+  // calling `build.resolve()`. This dramatically improves performance since
+  // `build.resolve()` requires an IPC call to esbuild's go code. It takes our
+  // `//app:app_client_optimize_deps` rule from building in ~40s to building
+  // in ~4.
+  if (importPath.startsWith(".")) {
+    let resolvedPath = path.join(otherOptions.resolveDir, importPath);
+    try {
+      if ((await fs.lstat(resolvedPath)).isSymbolicLink()) {
+        resolvedPath = await fs.readlink(path.join(otherOptions.resolveDir, importPath));
+      }
+      result = {path: resolvedPath};
+    } catch (error) {
+      if (error.code !== "ENOENT") {
+        throw error;
+      }
+    }
+  }
+
+  if (!result) {
+    result = await build.resolve(importPath, otherOptions);
+  }
 
   if (result.errors && result.errors.length) {
     // There was an error resolving, just return the error as-is.
diff --git a/esbuild/private/versions.bzl b/esbuild/private/versions.bzl
index 7d2250f..281884e 100644
--- a/esbuild/private/versions.bzl
+++ b/esbuild/private/versions.bzl
@@ -5,6 +5,14 @@ RULES_ESBUILD_VERSION = "$Format:%(describe:tags=true)$"  # Replaced during publ
 # This assignment is automatically updated by /scripts/mirror_release.sh.
 # Versions should be descending order so TOOL_VERSIONS.keys()[0] is the latest version.
 TOOL_VERSIONS = {
+    "0.21.5": {
+        "npm": "sha512-mg3OPMV4hXywwpoDxu3Qda5xCKQi+vCTZq8S9J/EpkhB2HzKXq4SNFZE3+NK93JYxc8VMSep+lOUSC/RVKaBqw==",
+        "darwin-x64": "sha512-se/JjF8NlmKVG4kNIuyWMV/22ZaerB+qaSi5MdrXtd6R08kvs2qCN4C09miupktDitvh8jRFflwGFBQcxZRjbw==",
+        "darwin-arm64": "sha512-DwqXqZyuk5AiWWf3UfLiRDJ5EDd49zg6O9wclZ7kUMv2WRFr4HKjXp/5t8JZ11QbQfUS6/cRCKGwYhtNAY88kQ==",
+        "linux-x64": "sha512-1rYdTpyv03iycF1+BhzrzQJCdOuAOtaqHTWJZCWvijKD2N5Xu0TtVC8/+1faWqcP9iBCWOmjmhoH94dH82BxPQ==",
+        "linux-arm64": "sha512-ibKvmyYzKsBeX8d8I7MH/TMfWDXBF3db4qM6sy+7re0YXya+K1cem3on9XgdT2EQGMu4hQyZhan7TeQ8XkGp4Q==",
+        "win32-x64": "sha512-tQd/1efJuzPC6rCFwEvLtci/xNFcTZknmXs98FYDfGE4wP9ClFV98nyKrzJKVPMhdDnjzLhdUyMX4PsQAPjwIw=="
+    },
     "0.21.1": {
         "npm": "sha512-GPqx+FX7mdqulCeQ4TsGZQ3djBJkx5k7zBGtqt9ycVlWNg8llJ4RO9n2vciu8BN2zAEs6lPbPl0asZsAh7oWzg==",
         "darwin-x64": "sha512-D3h3wBQmeS/vp93O4B+SWsXB8HvRDwMyhTNhBd8yMbh5wN/2pPWRW5o/hM3EKgk9bdKd9594lMGoTCTiglQGRQ==",
