diff --git a/ts/private/ts_project.bzl b/ts/private/ts_project.bzl
index 367bba0..4fdc98c 100644
--- a/ts/private/ts_project.bzl
+++ b/ts/private/ts_project.bzl
@@ -366,6 +366,10 @@ See https://github.com/aspect-build/rules_ts/issues/361 for more details.
             # https://bazel.build/extending/rules#validations_output_group
             # "hold the otherwise unused outputs of validation actions"
             _validation = validation_outs,
+            # NOTE(calebmer): We need access to the source files of
+            # `ts_project()` rules to run a repository wide TypeScript
+            # check test.
+            srcs = srcs_inputs,
         ),
         coverage_common.instrumented_files_info(
             ctx,
diff --git a/ts/private/versions.bzl b/ts/private/versions.bzl
index dbfe905..105152d 100644
--- a/ts/private/versions.bzl
+++ b/ts/private/versions.bzl
@@ -168,4 +168,7 @@ TOOL_VERSIONS = {
     "5.5.3": "sha512-/hreyEujaB0w76zKo6717l3L0o/qEUtRgdvUBvlkhoWeOVMjMuHNHk0BRBzikzuGDqNmPQbg5ifMEqsHLiIUcQ==",
     "5.5.4": "sha512-Mtq29sKDAEYP7aljRgtPOpTvOfbwRWlS6dPRzwjdE+C0R4brX/GUyhHSecbHMFLNBLcJIPt9nl9yG5TZ1weH+Q==",
     "5.6.2": "sha512-NW8ByodCSNCwZeghjN3o+JX5OFH0Ojg6sadjEKY4huZ52TqbJTJnDo5+Tw98lSy63NZvi4n+ez5m2u5d4PkZyw==",
+    # NOTE(calebmer): Allow using the latest version of TypeScript without
+    # updating `rules_ts`.
+    "5.9.2": "sha512-CWBzXQrc/qOkhidw1OzBTQuYRbfyxDXJMVJ1XNwUHGROVmuaeiEm3OslpZ1RV96d7SKKjZKrSJu3+t/xlw3R9A==",
 }
