diff --git a/swc/private/swc.bzl b/swc/private/swc.bzl
index 3331bdd..ab19d57 100644
--- a/swc/private/swc.bzl
+++ b/swc/private/swc.bzl
@@ -203,10 +203,32 @@ def _calculate_source_file(ctx, src):
     return paths.join(s, src_pkg, src.basename)
 
 def _swc_action(ctx, swc_binary, **kwargs):
+    execution_requirements = kwargs.pop("execution_requirements", default = {})
+
+    execution_requirements.update({
+        # NOTE(calebmer): We want SWC to correctly transform our imports using the `baseUrl`
+        # and `paths` configuration options. This requires the input file and output file to
+        # be in the same directory. `copy_file_to_bin_action()` is executed with
+        # `execution_requirements` including `"no-sandbox": "1"`. This means our input file
+        # is linked from `execroot` when we create a sandbox however we want to output in
+        # the sandbox. So execute SWC outside of a sandbox too so our input and output have
+        # the same base path.
+        #
+        # Not sandboxing means we have to trust SWC to be hermetic. This is a dangerous
+        # assumption. SWC could look at `package.json` or `.swcrc` files it is not supposed
+        # to. Generally SWC is a small, focused, tool so we trust it to behave hermetically.
+        #
+        # NOTE(calebmer, 2024-11-15): We might not need this patch anymore. Looks like
+        # `rules_swc` was updated to include `"no-sandbox": "1"` by default in some
+        # scenarios.
+        "no-sandbox": "1",
+    })
+
     ctx.actions.run(
         mnemonic = "SWCCompile",
         progress_message = "Compiling %{label} [swc %{input}]",
         executable = swc_binary,
+        execution_requirements = execution_requirements,
         **kwargs
     )
 
