diff --git a/js/private/coverage/merger.bzl b/js/private/coverage/merger.bzl
index b41acfa6..86864854 100644
--- a/js/private/coverage/merger.bzl
+++ b/js/private/coverage/merger.bzl
@@ -31,10 +31,13 @@ def _coverage_merger_impl(ctx):
         # TODO(3.0): drop support for deprecated toolchain attributes
         node_path = _deprecated_target_tool_path_to_short_path(nodeinfo.target_tool_path)
 
-    # The '_' avoids collisions with another file matching the label name.
-    # For example, test and test/my.spec.ts. This naming scheme is borrowed from rules_go:
-    # https://github.com/bazelbuild/rules_go/blob/f3cc8a2d670c7ccd5f45434ab226b25a76d44de1/go/private/context.bzl#L144
-    bash_launcher = ctx.actions.declare_file("{}_/{}".format(ctx.label.name, ctx.label.name))
+    # NOTE(calebmer): Revert [breaking change which removes `.sh` extension][1]
+    # from binary executable file. Instead `rules_js` chose to use a weird
+    # `{label}_/{label}` format. We don't find the argument for making this change
+    # compelling and we're worried about how this will break our code.
+    #
+    # [1]: https://github.com/aspect-build/rules_js/pull/1586
+    bash_launcher = ctx.actions.declare_file("{}.sh".format(ctx.label.name))
     ctx.actions.expand_template(
         template = ctx.file._launcher_template,
         output = bash_launcher,
diff --git a/js/private/js_binary.bzl b/js/private/js_binary.bzl
index 8aa233e4..381263cd 100644
--- a/js/private/js_binary.bzl
+++ b/js/private/js_binary.bzl
@@ -486,10 +486,13 @@ def _bash_launcher(ctx, nodeinfo, entry_point_path, log_prefix_rule_set, log_pre
         "{{workspace_name}}": ctx.workspace_name,
     }
 
-    # The '_' avoids collisions with another file matching the label name.
-    # For example, test and test/my.spec.ts. This naming scheme is borrowed from rules_go:
-    # https://github.com/bazelbuild/rules_go/blob/f3cc8a2d670c7ccd5f45434ab226b25a76d44de1/go/private/context.bzl#L144
-    launcher = ctx.actions.declare_file("{}_/{}".format(ctx.label.name, ctx.label.name))
+    # NOTE(calebmer): Revert [breaking change which removes `.sh` extension][1]
+    # from binary executable file. Instead `rules_js` chose to use a weird
+    # `{label}_/{label}` format. We don't find the argument for making this change
+    # compelling and we're worried about how this will break our code.
+    #
+    # [1]: https://github.com/aspect-build/rules_js/pull/1586
+    launcher = ctx.actions.declare_file("{}.sh".format(ctx.label.name))
     ctx.actions.expand_template(
         template = ctx.file._launcher_template,
         output = launcher,
diff --git a/js/private/js_image_layer.bzl b/js/private/js_image_layer.bzl
index cc451794..2d1b4cf6 100644
--- a/js/private/js_image_layer.bzl
+++ b/js/private/js_image_layer.bzl
@@ -267,7 +267,14 @@ source {real_binary_path}
 
 def _write_laucher(ctx, real_binary_path):
     "Creates a call-through shell entrypoint which sets BAZEL_BINDIR to '.' then immediately invokes the original entrypoint."
-    launcher = ctx.actions.declare_file("%s_launcher" % ctx.label.name)
+
+    # NOTE(calebmer): Revert [breaking change which removes `.sh` extension][1]
+    # from binary executable file. Instead `rules_js` chose to use a weird
+    # `{label}_/{label}` format. We don't find the argument for making this change
+    # compelling and we're worried about how this will break our code.
+    #
+    # [1]: https://github.com/aspect-build/rules_js/pull/1586
+    launcher = ctx.actions.declare_file("%s_launcher.sh" % ctx.label.name)
     ctx.actions.write(
         output = launcher,
         content = _LAUNCHER_TMPL.format(real_binary_path = real_binary_path),
diff --git a/js/private/js_run_binary.bzl b/js/private/js_run_binary.bzl
index 95f58d17..71515024 100644
--- a/js/private/js_run_binary.bzl
+++ b/js/private/js_run_binary.bzl
@@ -401,7 +401,9 @@ See https://github.com/aspect-build/rules_js/tree/main/docs#using-binaries-publi
     _run_binary(
         name = name,
         tool = tool,
-        env = dicts.add(fixed_env, env),
+        # NOTE(calebmer): Use dictionary union syntax which supports merging
+        # dictionaries that contain `select({})`.
+        env = fixed_env | env,
         srcs = srcs + extra_srcs,
         outs = outs + extra_outs,
         out_dirs = out_dirs,
diff --git a/npm/private/npm_import.bzl b/npm/private/npm_import.bzl
index 641a3125..87cdc56c 100644
--- a/npm/private/npm_import.bzl
+++ b/npm/private/npm_import.bzl
@@ -110,6 +110,23 @@ def npm_imported_package_store(name):
     )
 """
 
+# NOTE(calebmer): (This comment refers to the
+# `{lifecycle_hooks_env} | select({})` change we patch in below).
+#
+# Some npm packages like `utf-8-validate` and `bufferutil` have native C code
+# that's compiled in a lifecycle hook (e.g. `postinstall`) by
+# `node-gyp rebuild`. `aspect_rules_js` is smart enough to set
+# `npm_config_arch=x64` (used by `node-gyp-build`) which makes sure to pass the
+# right arguments to the C compiler for cross compilation. But it's not smart
+# enough to configure the right cross-compiler `CC` and `CXX` tools. So we get
+# an error as we try to use options for the x86_64 C compiler with the host's
+# arm64 C compiler.
+#
+# We install x86_64 cross compilers on our arm64 Linux GitHub runner with
+# `apt-get install crossbuild-essential-amd64`. This gives us a cross compiler
+# at `/usr/bin/x86_64-linux-gnu-gcc`. Next, we need to set the `CC` and `CXX`
+# environment variables so `node-gyp` in a lifecycle hook can use the right C
+# compiler. We do that here.
 _LINK_JS_PACKAGE_LIFECYCLE_TMPL = """\
     lc_deps = {lc_deps}
 
@@ -169,7 +186,17 @@ _LINK_JS_PACKAGE_LIFECYCLE_TMPL = """\
         execution_requirements = {lifecycle_hooks_execution_requirements},
         mnemonic = "NpmLifecycleHook",
         progress_message = "Running lifecycle hooks on npm package {package}@{version}",
-        env = {lifecycle_hooks_env},
+        env = {lifecycle_hooks_env} | select({{
+            "@aspect_rules_js//platforms/os:linux_host_and_cpu_arm64": {{
+                "CC": "/usr/bin/aarch64-linux-gnu-gcc",
+                "CXX": "/usr/bin/aarch64-linux-gnu-g++",
+            }},
+            "@aspect_rules_js//platforms/os:linux_host_and_cpu_x86_64": {{
+                "CC": "/usr/bin/x86_64-linux-gnu-gcc",
+                "CXX": "/usr/bin/x86_64-linux-gnu-g++",
+            }},
+            "//conditions:default": {{}},
+        }}),
         use_default_shell_env = {use_default_shell_env},
     )
 
diff --git a/platforms/os/BUILD.bazel b/platforms/os/BUILD.bazel
index 9cc3129b..874795a2 100644
--- a/platforms/os/BUILD.bazel
+++ b/platforms/os/BUILD.bazel
@@ -1,3 +1,6 @@
+load("@host_platform//:constraints.bzl", "HOST_CONSTRAINTS")
+load("@bazel_skylib//lib:selects.bzl", "selects")
+
 alias(
     name = "osx",
     actual = "@platforms//os:osx",
@@ -15,3 +18,43 @@ alias(
     actual = "@platforms//os:windows",
     visibility = ["//visibility:public"],
 )
+
+# NOTE(calebmer): Constraint that checks:
+#
+# - Host:
+#     - OS: Linux
+# - Target:
+#     - OS: Linux
+#     - CPU: arm64
+#
+# If our host is MacOS then `HOST_CONSTRAINTS` will contain
+# `@platforms//os:osx`. We'll never match both `@platforms//os:linux`
+# AND `@platforms//os:osx`.
+selects.config_setting_group(
+    name = "linux_host_and_cpu_arm64",
+    match_all = [
+        ":linux",
+        "//platforms/cpu:arm64",
+    ] + [constraint for constraint in HOST_CONSTRAINTS if constraint.startswith("@platforms//os:")],
+    visibility = ["//visibility:public"],
+)
+
+# NOTE(calebmer): Constraint that checks:
+#
+# - Host:
+#     - OS: Linux
+# - Target:
+#     - OS: Linux
+#     - CPU: x86_64
+#
+# If our host is MacOS then `HOST_CONSTRAINTS` will contain
+# `@platforms//os:osx`. We'll never match both `@platforms//os:linux`
+# AND `@platforms//os:osx`.
+selects.config_setting_group(
+    name = "linux_host_and_cpu_x86_64",
+    match_all = [
+        ":linux",
+        "//platforms/cpu:x86_64",
+    ] + [constraint for constraint in HOST_CONSTRAINTS if constraint.startswith("@platforms//os:")],
+    visibility = ["//visibility:public"],
+)
