diff --color -ur a/girepository/cmph/meson.build b/girepository/cmph/meson.build
--- a/girepository/cmph/meson.build	2023-03-22 18:43:34
+++ b/girepository/cmph/meson.build	2024-08-27 22:48:47
@@ -58,20 +58,5 @@
   include_directories: include_directories('.'),
 )
 
-if cc.get_id() != 'msvc'
-  custom_c_args = cc.get_supported_arguments([
-    '-Wno-old-style-definition',
-    '-Wno-type-limits',
-  ])
-endif
-
-cmph_test = executable('cmph-bdz-test', '../cmph-bdz-test.c',
-  dependencies: [
-    cmph_dep,
-    glib_dep,
-    gobject_dep,
-  ],
-  c_args: custom_c_args,
-)
-
-test('cmph-bdz-test', cmph_test)
+# NOTE(calebmer): Don't build tests. Speeds things up and means we don't have
+# to address some build errors in the test directory.
diff --color -ur a/girepository/meson.build b/girepository/meson.build
--- a/girepository/meson.build	2023-03-22 18:43:34
+++ b/girepository/meson.build	2024-08-27 22:48:07
@@ -191,11 +191,8 @@
   include_directories: include_directories('.'),
 )
 
-gthash_test = executable('gthash-test', 'gthash-test.c',
-  dependencies: girepo_gthash_dep,
-)
-
-test('gthash-test', gthash_test)
+# NOTE(calebmer): Don't build tests. Speeds things up and means we don't have
+# to address some build errors in the test directory.
 
 if giounix_dep.found() or giowin_dep.found()
   executable('gi-dump-types', 'gi-dump-types.c',
diff --color -ur a/meson.build b/meson.build
--- a/meson.build	2023-03-22 18:43:34
+++ b/meson.build	2024-08-27 22:44:40
@@ -237,11 +237,8 @@
 endif
 subdir('docs')
 
-# The tests will also run, which is not possible if they
-# were built for a different architecture.
-if not meson.is_cross_build()
-  subdir('tests')
-endif
+# NOTE(calebmer): Don't build tests. Speeds things up and means we don't have
+# to address some build errors in the test directory.
 
 install_data('Makefile.introspection', install_dir: join_paths(get_option('datadir'), 'gobject-introspection-1.0'))
 install_data('m4/introspection.m4', install_dir: join_paths(get_option('datadir'), 'aclocal'))
