diff --git a/foreign_cc/private/framework.bzl b/foreign_cc/private/framework.bzl
index 892467e..b4138db 100644
--- a/foreign_cc/private/framework.bzl
+++ b/foreign_cc/private/framework.bzl
@@ -460,6 +460,44 @@ def cc_external_rule_impl(ctx, attrs):
     ] + _print_env() + _copy_deps_and_tools(inputs) + [
         "cd $$BUILD_TMPDIR$$",
     ] + attrs.create_configure_script(ConfigureParameters(ctx = ctx, attrs = attrs, inputs = inputs)) + postfix_script + [
+        # NOTE(calebmer): Looks like by default MacOS dynamic libraries are built with
+        # an absolute path for `LC_ID_DYLIB` unless `-install_name` is set. The problem
+        # is the absolute path at build time will be in a Bazel sandbox directory! If a
+        # program is dynamically linked to this shared library, it will try to load the
+        # library from `LC_ID_DYLIB` which fails because the original build directory
+        # is gone.
+        #
+        # So on MacOS we use `install_name_tool {file} -id {id}` at the end of the
+        # build to change `LC_ID_DYLIB` to a relative path based on `@loader_path`. We
+        # make some assumptions about the build directory layout created by
+        # `rules_foreign_cc`. Namely that libraries live under the path
+        # `external/{workspace_name}/{target_name}/lib/{lib_name}`. Which means we can
+        # set `LC_ID_DYLIB` to
+        # `@loader_path/../../../{workspace_name}/{target_name}/lib/{lib_name}` and
+        # it'll point to the right location.
+        #
+        # Since this runs at the end of the build, the build tool may have built some
+        # of its dynamic libraries and binaries with links to other libraries generated
+        # by the same build. When the build tool performs this link `LC_ID_DYLIB` will
+        # be the sandbox absolute path. So this script also runs
+        # `install_name_tool {file} -change {old} {new}` for every other shared library
+        # to make sure any `LC_LOAD_DYLIB` references to other shared libraries from
+        # this build are replaced with relative paths.
+        #
+        # The `{old}` paths we change are:
+        #
+        # 1. Paths to dynamic libraries in the same directory as our file
+        #    (installation path).
+        # 2. Paths to dynamic libraries in the `build_tmpdir` for our build.
+        #
+        # This script line noops for all other platforms.
+        "##macos_install_name_tool_lib## {}".format(file.path)
+        for file in outputs.libraries.shared_libraries
+    ] + [
+        # Make sure `LC_LOAD_DYLIB` references in built binaries are updated too.
+        "##macos_install_name_tool_bin## {}".format(file.path)
+        for file in outputs.out_binary_files
+    ] + [
         # replace references to the root directory when building ($BUILD_TMPDIR)
         # and the root where the dependencies were installed ($EXT_BUILD_DEPS)
         # for the results which are in $INSTALLDIR (with placeholder)
diff --git a/foreign_cc/private/framework/toolchains/commands.bzl b/foreign_cc/private/framework/toolchains/commands.bzl
index e4f1073..650ec20 100644
--- a/foreign_cc/private/framework/toolchains/commands.bzl
+++ b/foreign_cc/private/framework/toolchains/commands.bzl
@@ -203,6 +203,32 @@ PLATFORM_COMMANDS = {
             "Replace the target symlink with resolved file it points to if `file` is a symlink"
         ),
     ),
+    "macos_install_name_tool_lib": _command_info(
+        arguments = [
+            _argument_info(
+                name = "file",
+                data_type = type(""),
+                doc = "Target file",
+            ),
+        ],
+        doc = (
+            "Run `install_name_tool` on MacOS to replace `LC_ID_DYLIB` and `LC_LOAD_DYLIB` with relative " +
+            "paths instead of an absolute path to the sandbox"
+        ),
+    ),
+    "macos_install_name_tool_bin": _command_info(
+        arguments = [
+            _argument_info(
+                name = "file",
+                data_type = type(""),
+                doc = "Target file",
+            ),
+        ],
+        doc = (
+            "Run `install_name_tool` on MacOS to replace `LC_LOAD_DYLIB` with relative paths instead of an " +
+            "absolute path to the sandbox"
+        ),
+    ),
     "rm_rf": _command_info(
         arguments = [_argument_info(name = "path", data_type = type(""), doc = "Path to directory")],
         doc = "Recursively removes the given path",
diff --git a/foreign_cc/private/framework/toolchains/freebsd_commands.bzl b/foreign_cc/private/framework/toolchains/freebsd_commands.bzl
index 9fb552f..2433011 100644
--- a/foreign_cc/private/framework/toolchains/freebsd_commands.bzl
+++ b/foreign_cc/private/framework/toolchains/freebsd_commands.bzl
@@ -271,6 +271,12 @@ if [[ -L "{file}" ]]; then
 fi
 """.format(file = file)
 
+def macos_install_name_tool_lib(file):
+    return "# noop macos_install_name_tool_lib \"{file}\"".format(file = file)
+
+def macos_install_name_tool_bin(file):
+    return "# noop macos_install_name_tool_bin \"{file}\"".format(file = file)
+
 commands = struct(
     assert_script_errors = assert_script_errors,
     cat = cat,
@@ -296,6 +302,8 @@ commands = struct(
     replace_in_files = replace_in_files,
     replace_sandbox_paths = replace_sandbox_paths,
     replace_symlink = replace_symlink,
+    macos_install_name_tool_lib = macos_install_name_tool_lib,
+    macos_install_name_tool_bin = macos_install_name_tool_bin,
     rm_rf = rm_rf,
     script_extension = script_extension,
     script_prelude = script_prelude,
diff --git a/foreign_cc/private/framework/toolchains/linux_commands.bzl b/foreign_cc/private/framework/toolchains/linux_commands.bzl
index ba265eb..97bcf09 100644
--- a/foreign_cc/private/framework/toolchains/linux_commands.bzl
+++ b/foreign_cc/private/framework/toolchains/linux_commands.bzl
@@ -253,6 +253,12 @@ if [[ -L "{file}" ]]; then
 fi
 """.format(file = file)
 
+def macos_install_name_tool_lib(file):
+    return "# noop macos_install_name_tool_lib \"{file}\"".format(file = file)
+
+def macos_install_name_tool_bin(file):
+    return "# noop macos_install_name_tool_bin \"{file}\"".format(file = file)
+
 commands = struct(
     assert_script_errors = assert_script_errors,
     cat = cat,
@@ -278,6 +284,8 @@ commands = struct(
     replace_in_files = replace_in_files,
     replace_sandbox_paths = replace_sandbox_paths,
     replace_symlink = replace_symlink,
+    macos_install_name_tool_lib = macos_install_name_tool_lib,
+    macos_install_name_tool_bin = macos_install_name_tool_bin,
     rm_rf = rm_rf,
     script_extension = script_extension,
     script_prelude = script_prelude,
diff --git a/foreign_cc/private/framework/toolchains/macos_commands.bzl b/foreign_cc/private/framework/toolchains/macos_commands.bzl
index ed04f24..90ad807 100644
--- a/foreign_cc/private/framework/toolchains/macos_commands.bzl
+++ b/foreign_cc/private/framework/toolchains/macos_commands.bzl
@@ -266,6 +266,44 @@ if [[ -L "{file}" ]]; then
 fi
 """.format(file = file)
 
+def macos_install_name_tool_lib(file):
+    return """\
+install_name_tool_file="$EXT_BUILD_ROOT/{file}"
+install_name_tool_dirname1="$(dirname $install_name_tool_file)"
+install_name_tool_dirname2="$(dirname $install_name_tool_dirname1)"
+install_name_tool_dirname3="$(dirname $install_name_tool_dirname2)"
+install_name_tool_args="$install_name_tool_file -id @loader_path/../../../$(basename "$install_name_tool_dirname3")/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file")"
+for install_name_tool_file in "$install_name_tool_dirname1"/*; do
+    if [[ -f "$install_name_tool_file" ]]; then
+        install_name_tool_dirname1="$(dirname $install_name_tool_file)"
+        install_name_tool_dirname2="$(dirname $install_name_tool_dirname1)"
+        install_name_tool_dirname3="$(dirname $install_name_tool_dirname2)"
+        install_name_tool_args="$install_name_tool_args -change $install_name_tool_file @loader_path/../../../$(basename "$install_name_tool_dirname3")/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file")"
+        install_name_tool_args="$install_name_tool_args -change $install_name_tool_dirname3/$(basename "$install_name_tool_dirname2").build_tmpdir/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file") @loader_path/../../../$(basename "$install_name_tool_dirname3")/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file")"
+    fi
+done
+install_name_tool $install_name_tool_args
+""".format(file = file)
+
+def macos_install_name_tool_bin(file):
+    return """\
+install_name_tool_bin_file="$EXT_BUILD_ROOT/{file}"
+install_name_tool_args=""
+for install_name_tool_file in "$(dirname $(dirname $install_name_tool_bin_file))"/lib/*; do
+    if [[ -f "$install_name_tool_file" ]]; then
+        install_name_tool_dirname1="$(dirname $install_name_tool_file)"
+        install_name_tool_dirname2="$(dirname $install_name_tool_dirname1)"
+        install_name_tool_dirname3="$(dirname $install_name_tool_dirname2)"
+        install_name_tool_args="$install_name_tool_args -change $install_name_tool_file @loader_path/../../../$(basename "$install_name_tool_dirname3")/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file")"
+        install_name_tool_args="$install_name_tool_args -change $install_name_tool_dirname3/$(basename "$install_name_tool_dirname2").build_tmpdir/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file") @loader_path/../../../$(basename "$install_name_tool_dirname3")/$(basename "$install_name_tool_dirname2")/$(basename "$install_name_tool_dirname1")/$(basename "$install_name_tool_file")"
+    fi
+done
+if [ ! -z "$install_name_tool_args" ]; then
+  # Ignore error in case `install_name_tool_bin_file` is not a Mach-O file.
+  install_name_tool "$install_name_tool_bin_file" $install_name_tool_args || true
+fi
+""".format(file = file)
+
 commands = struct(
     assert_script_errors = assert_script_errors,
     cat = cat,
@@ -291,6 +329,8 @@ commands = struct(
     replace_in_files = replace_in_files,
     replace_sandbox_paths = replace_sandbox_paths,
     replace_symlink = replace_symlink,
+    macos_install_name_tool_lib = macos_install_name_tool_lib,
+    macos_install_name_tool_bin = macos_install_name_tool_bin,
     rm_rf = rm_rf,
     script_extension = script_extension,
     script_prelude = script_prelude,
diff --git a/foreign_cc/private/framework/toolchains/windows_commands.bzl b/foreign_cc/private/framework/toolchains/windows_commands.bzl
index f74cd94..e3f74c3 100644
--- a/foreign_cc/private/framework/toolchains/windows_commands.bzl
+++ b/foreign_cc/private/framework/toolchains/windows_commands.bzl
@@ -288,6 +288,9 @@ if [[ -L "{file}" ]]; then
 fi
 """.format(file = file)
 
+def macos_install_name_tool_lib(file):
+    return "# noop macos_install_name_tool_lib \"{file}\"".format(file = file)
+
 commands = struct(
     assert_script_errors = assert_script_errors,
     cat = cat,
@@ -313,6 +316,8 @@ commands = struct(
     replace_in_files = replace_in_files,
     replace_sandbox_paths = replace_sandbox_paths,
     replace_symlink = replace_symlink,
+    macos_install_name_tool_lib = macos_install_name_tool_lib,
+    macos_install_name_tool_bin = macos_install_name_tool_bin,
     rm_rf = rm_rf,
     script_extension = script_extension,
     script_prelude = script_prelude,
