diff --git a/nodejs/repositories.bzl b/nodejs/repositories.bzl
index 4534fb42..75ff9a82 100644
--- a/nodejs/repositories.bzl
+++ b/nodejs/repositories.bzl
@@ -337,7 +337,10 @@ filegroup(
         node_bin_export = "\n  \"%s\"," % node_bin,
         npm_bin_export = "\n  \"%s\"," % npm_bin,
         npx_bin_export = "\n  \"%s\"," % npx_bin,
-        npm_files_glob = "glob([\"bin/nodejs/**\"]) + ",
+        # NOTE(calebmer): Exclude dot files like `.DS_Store` added by MacOS. We
+        # observed remote caching issues where MacOS can't reuse a remote cache
+        # because of `.DS_Store` files in this rule.
+        npm_files_glob = "glob([\"bin/nodejs/**\"], exclude = [\"**/.*\"]) + ",
         node_bin_label = node_bin_label,
         npm_bin_label = npm_bin_label,
         npx_bin_label = npx_bin_label,
